<!DOCTYPE HTML>
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta http-equiv=Content-Type content="text/html; charset=us-ascii">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="DiyabcRandomForestUserManual21-01-2021_fichiers/filelist.xml">
<link rel=Edit-Time-Data
href="DiyabcRandomForestUserManual21-01-2021_fichiers/editdata.mso">
<!--[if !mso]>
<style>
v\:* {behavior:url(#default#VML);}
o\:* {behavior:url(#default#VML);}
w\:* {behavior:url(#default#VML);}
.shape {behavior:url(#default#VML);}
</style>
<![endif]-->
<title>USER MANUAL for DIYABC Random Forest v1.0</title>
<!--[if gte mso 9]><xml>
 <o:DocumentProperties>
  <o:Author>Arnaud Estoup</o:Author>
  <o:LastAuthor>Fran&#231;ois-David Collin</o:LastAuthor>
  <o:Revision>2</o:Revision>
  <o:TotalTime>8</o:TotalTime>
  <o:LastPrinted>2021-01-21T16:10:00Z</o:LastPrinted>
  <o:Created>2021-01-21T16:44:00Z</o:Created>
  <o:LastSaved>2021-01-21T16:44:00Z</o:LastSaved>
  <o:Pages>62</o:Pages>
  <o:Words>20733</o:Words>
  <o:Characters>114033</o:Characters>
  <o:Company>Universit&#233; de Montpellier</o:Company>
  <o:Paragraphs>en-US</o:Paragraphs>
  <o:Lines>950</o:Lines>
  <o:Paragraphs>268</o:Paragraphs>
  <o:CharactersWithSpaces>134498</o:CharactersWithSpaces>
  <o:Version>16.00</o:Version>
 </o:DocumentProperties>
 <o:CustomDocumentProperties>
  <o:AppVersion dt:dt="string">14.0000</o:AppVersion>
  <o:Company dt:dt="string">Universit&#233; de Montpellier</o:Company>
  <o:DocSecurity dt:dt="float">0</o:DocSecurity>
  <o:HyperlinksChanged dt:dt="boolean">0</o:HyperlinksChanged>
  <o:LinksUpToDate dt:dt="boolean">0</o:LinksUpToDate>
  <o:ScaleCrop dt:dt="boolean">0</o:ScaleCrop>
  <o:ShareDoc dt:dt="boolean">0</o:ShareDoc>
 </o:CustomDocumentProperties>
</xml><![endif]-->
<link rel=dataStoreItem
href="DiyabcRandomForestUserManual21-01-2021_fichiers/item0001.xml"
target="DiyabcRandomForestUserManual21-01-2021_fichiers/props002.xml">
<link rel=themeData
href="DiyabcRandomForestUserManual21-01-2021_fichiers/themedata.thmx">
<link rel=colorSchemeMapping
href="DiyabcRandomForestUserManual21-01-2021_fichiers/colorschememapping.xml">
<!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:ActiveWritingStyle Lang="FR" VendorID="64" DLLVersion="6" NLCheck="1">0</w:ActiveWritingStyle>
  <w:ActiveWritingStyle Lang="EN-US" VendorID="64" DLLVersion="6" NLCheck="1">0</w:ActiveWritingStyle>
  <w:ActiveWritingStyle Lang="EN-US" VendorID="64" DLLVersion="4096" NLCheck="1">0</w:ActiveWritingStyle>
  <w:ActiveWritingStyle Lang="FR" VendorID="64" DLLVersion="4096" NLCheck="1">0</w:ActiveWritingStyle>
  <w:SpellingState>Clean</w:SpellingState>
  <w:GrammarState>Clean</w:GrammarState>
  <w:TrackMoves>false</w:TrackMoves>
  <w:TrackFormatting/>
  <w:HyphenationZone>21</w:HyphenationZone>
  <w:PunctuationKerning/>
  <w:ValidateAgainstSchemas/>
  <w:SaveIfXMLInvalid>false</w:SaveIfXMLInvalid>
  <w:IgnoreMixedContent>false</w:IgnoreMixedContent>
  <w:AlwaysShowPlaceholderText>false</w:AlwaysShowPlaceholderText>
  <w:DoNotPromoteQF/>
  <w:LidThemeOther>EN-US</w:LidThemeOther>
  <w:Compatibility>
   <w:BreakWrappedTables/>
   <w:SnapToGridInCell/>
   <w:WrapTextWithPunct/>
   <w:UseAsianBreakRules/>
   <w:UseWord2010TableStyleRules/>
   <w:DontGrowAutofit/>
   <w:SplitPgBreakAndParaMark/>
   <w:EnableOpenTypeKerning/>
   <w:DontFlipMirrorIndents/>
   <w:OverrideTableStyleHps/>
  </w:Compatibility>
  <w:DoNotOptimizeForBrowser/>
  <m:mathPr>
   <m:mathFont m:val="Cambria Math"/>
   <m:brkBin m:val="before"/>
   <m:brkBinSub m:val="&#45;-"/>
   <m:smallFrac m:val="off"/>
   <m:dispDef/>
   <m:lMargin m:val="0"/>
   <m:rMargin m:val="0"/>
   <m:defJc m:val="centerGroup"/>
   <m:wrapIndent m:val="1440"/>
   <m:intLim m:val="subSup"/>
   <m:naryLim m:val="undOvr"/>
  </m:mathPr></w:WordDocument>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:LatentStyles DefLockedState="false" DefUnhideWhenUsed="false"
  DefSemiHidden="false" DefQFormat="false" LatentStyleCount="375">
  <w:LsdException Locked="false" QFormat="true" Name="Normal"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 1"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 2"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="heading 6"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="index 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="index 4"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="index 5"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="index 6"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="index 7"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="index 8"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="index 9"/>
  <w:LsdException Locked="false" Priority="39" SemiHidden="true"
   UnhideWhenUsed="true" Name="toc 1"/>
  <w:LsdException Locked="false" Priority="39" SemiHidden="true"
   UnhideWhenUsed="true" Name="toc 2"/>
  <w:LsdException Locked="false" Priority="39" SemiHidden="true"
   UnhideWhenUsed="true" Name="toc 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="toc 4"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="toc 5"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="toc 6"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="toc 7"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="toc 8"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="toc 9"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Normal Indent"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="footnote text"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true"
   UnhideWhenUsed="true" QFormat="true" Name="annotation text"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="header"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true"
   UnhideWhenUsed="true" Name="footer"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="index heading"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="caption"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="table of figures"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="envelope address"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="envelope return"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="footnote reference"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true"
   UnhideWhenUsed="true" Name="annotation reference"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="line number"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="page number"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="endnote reference"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="endnote text"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="table of authorities"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="macro"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="toa heading"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List Bullet"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List Number"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List 4"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List 5"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List Bullet 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List Bullet 4"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List Number 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List Number 4"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List Number 5"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Closing"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Signature"/>
  <w:LsdException Locked="false" Priority="1" SemiHidden="true"
   UnhideWhenUsed="true" Name="Default Paragraph Font"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="Body Text"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Body Text Indent"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List Continue"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List Continue 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List Continue 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List Continue 4"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="List Continue 5"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Message Header"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Salutation"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Date"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Body Text First Indent"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Body Text First Indent 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Body Text Indent 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Block Text"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true"
   UnhideWhenUsed="true" Name="Hyperlink"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="FollowedHyperlink"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Document Map"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Plain Text"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="E-mail Signature"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="HTML Top of Form"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="HTML Bottom of Form"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Normal (Web)"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="HTML Acronym"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="HTML Address"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="HTML Cite"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="HTML Code"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="HTML Definition"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="HTML Keyboard"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="HTML Preformatted"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="HTML Sample"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="HTML Typewriter"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="HTML Variable"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Normal Table"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="annotation subject"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true"
   UnhideWhenUsed="true" Name="No List"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Outline List 1"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Outline List 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Outline List 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Simple 1"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Simple 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Simple 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Classic 1"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Classic 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Classic 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Classic 4"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Colorful 1"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Colorful 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Colorful 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Columns 1"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Columns 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Columns 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Columns 4"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Columns 5"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Grid 1"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Grid 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Grid 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Grid 4"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Grid 5"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Grid 6"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Grid 7"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Grid 8"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table List 1"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table List 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table List 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table List 4"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table List 5"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table List 6"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table List 7"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table List 8"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table 3D effects 1"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table 3D effects 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table 3D effects 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Contemporary"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Elegant"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Professional"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Subtle 1"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Subtle 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Web 1"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Web 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Web 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Balloon Text"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Grid"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Table Theme"/>
  <w:LsdException Locked="false" SemiHidden="true" Name="Placeholder Text"/>
  <w:LsdException Locked="false" SemiHidden="true" Name="Revision"/>
  <w:LsdException Locked="false" QFormat="true" Name="List Paragraph"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   Name="Bibliography"/>
  <w:LsdException Locked="false" Priority="39" SemiHidden="true"
   UnhideWhenUsed="true" QFormat="true" Name="TOC Heading"/>
  <w:LsdException Locked="false" Priority="41" Name="Plain Table 1"/>
  <w:LsdException Locked="false" Priority="42" Name="Plain Table 2"/>
  <w:LsdException Locked="false" Priority="43" Name="Plain Table 3"/>
  <w:LsdException Locked="false" Priority="44" Name="Plain Table 4"/>
  <w:LsdException Locked="false" Priority="45" Name="Plain Table 5"/>
  <w:LsdException Locked="false" Priority="40" Name="Grid Table Light"/>
  <w:LsdException Locked="false" Priority="46" Name="Grid Table 1 Light"/>
  <w:LsdException Locked="false" Priority="47" Name="Grid Table 2"/>
  <w:LsdException Locked="false" Priority="48" Name="Grid Table 3"/>
  <w:LsdException Locked="false" Priority="49" Name="Grid Table 4"/>
  <w:LsdException Locked="false" Priority="50" Name="Grid Table 5 Dark"/>
  <w:LsdException Locked="false" Priority="51" Name="Grid Table 6 Colorful"/>
  <w:LsdException Locked="false" Priority="52" Name="Grid Table 7 Colorful"/>
  <w:LsdException Locked="false" Priority="46"
   Name="Grid Table 1 Light Accent 1"/>
  <w:LsdException Locked="false" Priority="47" Name="Grid Table 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="48" Name="Grid Table 3 Accent 1"/>
  <w:LsdException Locked="false" Priority="49" Name="Grid Table 4 Accent 1"/>
  <w:LsdException Locked="false" Priority="50" Name="Grid Table 5 Dark Accent 1"/>
  <w:LsdException Locked="false" Priority="51"
   Name="Grid Table 6 Colorful Accent 1"/>
  <w:LsdException Locked="false" Priority="52"
   Name="Grid Table 7 Colorful Accent 1"/>
  <w:LsdException Locked="false" Priority="46"
   Name="Grid Table 1 Light Accent 2"/>
  <w:LsdException Locked="false" Priority="47" Name="Grid Table 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="48" Name="Grid Table 3 Accent 2"/>
  <w:LsdException Locked="false" Priority="49" Name="Grid Table 4 Accent 2"/>
  <w:LsdException Locked="false" Priority="50" Name="Grid Table 5 Dark Accent 2"/>
  <w:LsdException Locked="false" Priority="51"
   Name="Grid Table 6 Colorful Accent 2"/>
  <w:LsdException Locked="false" Priority="52"
   Name="Grid Table 7 Colorful Accent 2"/>
  <w:LsdException Locked="false" Priority="46"
   Name="Grid Table 1 Light Accent 3"/>
  <w:LsdException Locked="false" Priority="47" Name="Grid Table 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="48" Name="Grid Table 3 Accent 3"/>
  <w:LsdException Locked="false" Priority="49" Name="Grid Table 4 Accent 3"/>
  <w:LsdException Locked="false" Priority="50" Name="Grid Table 5 Dark Accent 3"/>
  <w:LsdException Locked="false" Priority="51"
   Name="Grid Table 6 Colorful Accent 3"/>
  <w:LsdException Locked="false" Priority="52"
   Name="Grid Table 7 Colorful Accent 3"/>
  <w:LsdException Locked="false" Priority="46"
   Name="Grid Table 1 Light Accent 4"/>
  <w:LsdException Locked="false" Priority="47" Name="Grid Table 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="48" Name="Grid Table 3 Accent 4"/>
  <w:LsdException Locked="false" Priority="49" Name="Grid Table 4 Accent 4"/>
  <w:LsdException Locked="false" Priority="50" Name="Grid Table 5 Dark Accent 4"/>
  <w:LsdException Locked="false" Priority="51"
   Name="Grid Table 6 Colorful Accent 4"/>
  <w:LsdException Locked="false" Priority="52"
   Name="Grid Table 7 Colorful Accent 4"/>
  <w:LsdException Locked="false" Priority="46"
   Name="Grid Table 1 Light Accent 5"/>
  <w:LsdException Locked="false" Priority="47" Name="Grid Table 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="48" Name="Grid Table 3 Accent 5"/>
  <w:LsdException Locked="false" Priority="49" Name="Grid Table 4 Accent 5"/>
  <w:LsdException Locked="false" Priority="50" Name="Grid Table 5 Dark Accent 5"/>
  <w:LsdException Locked="false" Priority="51"
   Name="Grid Table 6 Colorful Accent 5"/>
  <w:LsdException Locked="false" Priority="52"
   Name="Grid Table 7 Colorful Accent 5"/>
  <w:LsdException Locked="false" Priority="46"
   Name="Grid Table 1 Light Accent 6"/>
  <w:LsdException Locked="false" Priority="47" Name="Grid Table 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="48" Name="Grid Table 3 Accent 6"/>
  <w:LsdException Locked="false" Priority="49" Name="Grid Table 4 Accent 6"/>
  <w:LsdException Locked="false" Priority="50" Name="Grid Table 5 Dark Accent 6"/>
  <w:LsdException Locked="false" Priority="51"
   Name="Grid Table 6 Colorful Accent 6"/>
  <w:LsdException Locked="false" Priority="52"
   Name="Grid Table 7 Colorful Accent 6"/>
  <w:LsdException Locked="false" Priority="46" Name="List Table 1 Light"/>
  <w:LsdException Locked="false" Priority="47" Name="List Table 2"/>
  <w:LsdException Locked="false" Priority="48" Name="List Table 3"/>
  <w:LsdException Locked="false" Priority="49" Name="List Table 4"/>
  <w:LsdException Locked="false" Priority="50" Name="List Table 5 Dark"/>
  <w:LsdException Locked="false" Priority="51" Name="List Table 6 Colorful"/>
  <w:LsdException Locked="false" Priority="52" Name="List Table 7 Colorful"/>
  <w:LsdException Locked="false" Priority="46"
   Name="List Table 1 Light Accent 1"/>
  <w:LsdException Locked="false" Priority="47" Name="List Table 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="48" Name="List Table 3 Accent 1"/>
  <w:LsdException Locked="false" Priority="49" Name="List Table 4 Accent 1"/>
  <w:LsdException Locked="false" Priority="50" Name="List Table 5 Dark Accent 1"/>
  <w:LsdException Locked="false" Priority="51"
   Name="List Table 6 Colorful Accent 1"/>
  <w:LsdException Locked="false" Priority="52"
   Name="List Table 7 Colorful Accent 1"/>
  <w:LsdException Locked="false" Priority="46"
   Name="List Table 1 Light Accent 2"/>
  <w:LsdException Locked="false" Priority="47" Name="List Table 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="48" Name="List Table 3 Accent 2"/>
  <w:LsdException Locked="false" Priority="49" Name="List Table 4 Accent 2"/>
  <w:LsdException Locked="false" Priority="50" Name="List Table 5 Dark Accent 2"/>
  <w:LsdException Locked="false" Priority="51"
   Name="List Table 6 Colorful Accent 2"/>
  <w:LsdException Locked="false" Priority="52"
   Name="List Table 7 Colorful Accent 2"/>
  <w:LsdException Locked="false" Priority="46"
   Name="List Table 1 Light Accent 3"/>
  <w:LsdException Locked="false" Priority="47" Name="List Table 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="48" Name="List Table 3 Accent 3"/>
  <w:LsdException Locked="false" Priority="49" Name="List Table 4 Accent 3"/>
  <w:LsdException Locked="false" Priority="50" Name="List Table 5 Dark Accent 3"/>
  <w:LsdException Locked="false" Priority="51"
   Name="List Table 6 Colorful Accent 3"/>
  <w:LsdException Locked="false" Priority="52"
   Name="List Table 7 Colorful Accent 3"/>
  <w:LsdException Locked="false" Priority="46"
   Name="List Table 1 Light Accent 4"/>
  <w:LsdException Locked="false" Priority="47" Name="List Table 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="48" Name="List Table 3 Accent 4"/>
  <w:LsdException Locked="false" Priority="49" Name="List Table 4 Accent 4"/>
  <w:LsdException Locked="false" Priority="50" Name="List Table 5 Dark Accent 4"/>
  <w:LsdException Locked="false" Priority="51"
   Name="List Table 6 Colorful Accent 4"/>
  <w:LsdException Locked="false" Priority="52"
   Name="List Table 7 Colorful Accent 4"/>
  <w:LsdException Locked="false" Priority="46"
   Name="List Table 1 Light Accent 5"/>
  <w:LsdException Locked="false" Priority="47" Name="List Table 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="48" Name="List Table 3 Accent 5"/>
  <w:LsdException Locked="false" Priority="49" Name="List Table 4 Accent 5"/>
  <w:LsdException Locked="false" Priority="50" Name="List Table 5 Dark Accent 5"/>
  <w:LsdException Locked="false" Priority="51"
   Name="List Table 6 Colorful Accent 5"/>
  <w:LsdException Locked="false" Priority="52"
   Name="List Table 7 Colorful Accent 5"/>
  <w:LsdException Locked="false" Priority="46"
   Name="List Table 1 Light Accent 6"/>
  <w:LsdException Locked="false" Priority="47" Name="List Table 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="48" Name="List Table 3 Accent 6"/>
  <w:LsdException Locked="false" Priority="49" Name="List Table 4 Accent 6"/>
  <w:LsdException Locked="false" Priority="50" Name="List Table 5 Dark Accent 6"/>
  <w:LsdException Locked="false" Priority="51"
   Name="List Table 6 Colorful Accent 6"/>
  <w:LsdException Locked="false" Priority="52"
   Name="List Table 7 Colorful Accent 6"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true"
   UnhideWhenUsed="true" Name="Mention"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true"
   UnhideWhenUsed="true" Name="Smart Hyperlink"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true"
   UnhideWhenUsed="true" Name="Hashtag"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true"
   UnhideWhenUsed="true" Name="Unresolved Mention"/>
 </w:LatentStyles>
</xml><![endif]-->
<style>
<!--
 /* Font Definitions */
 @font-face
	{font-family:Wingdings;
	panose-1:5 0 0 0 0 0 0 0 0 0;
	mso-font-charset:2;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:0 268435456 0 0 -2147483648 0;}
@font-face
	{font-family:"Cambria Math";
	panose-1:2 4 5 3 5 4 6 3 2 4;
	mso-font-charset:0;
	mso-generic-font-family:roman;
	mso-font-pitch:variable;
	mso-font-signature:-536869121 1107305727 33554432 0 415 0;}
@font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4;
	mso-font-charset:0;
	mso-generic-font-family:swiss;
	mso-font-pitch:variable;
	mso-font-signature:-469750017 -1073732485 9 0 511 0;}
@font-face
	{font-family:Cambria;
	panose-1:2 4 5 3 5 4 6 3 2 4;
	mso-font-alt:Cambria;
	mso-font-charset:0;
	mso-generic-font-family:roman;
	mso-font-pitch:variable;
	mso-font-signature:-536869121 1107305727 33554432 0 415 0;}
@font-face
	{font-family:Consolas;
	panose-1:2 11 6 9 2 2 4 3 2 4;
	mso-font-charset:0;
	mso-generic-font-family:modern;
	mso-font-pitch:fixed;
	mso-font-signature:-536869121 64767 1 0 415 0;}
@font-face
	{font-family:Tahoma;
	panose-1:2 11 6 4 3 5 4 4 2 4;
	mso-font-alt:Tahoma;
	mso-font-charset:0;
	mso-generic-font-family:swiss;
	mso-font-pitch:variable;
	mso-font-signature:-520081665 -1073717157 41 0 66047 0;}
@font-face
	{font-family:"Liberation Sans";
	mso-font-alt:Arial;
	mso-font-charset:0;
	mso-generic-font-family:swiss;
	mso-font-pitch:variable;
	mso-font-signature:-536868097 1342208255 33 0 447 0;}
@font-face
	{font-family:"AR PL SungtiL GB";
	mso-font-charset:1;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:"Lohit Devanagari";
	mso-font-alt:"Times New Roman";
	mso-font-charset:1;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:CMR10;
	panose-1:0 0 0 0 0 0 0 0 0 0;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-format:other;
	mso-font-pitch:auto;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:CMMI10;
	panose-1:0 0 0 0 0 0 0 0 0 0;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-format:other;
	mso-font-pitch:auto;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:CMSS10;
	panose-1:0 0 0 0 0 0 0 0 0 0;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-format:other;
	mso-font-pitch:auto;
	mso-font-signature:3 0 0 0 1 0;}
 /* Style Definitions */
 p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
h1
	{mso-style-priority:9;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-link:"Titre 1 Car";
	margin-top:24.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:1;
	font-size:16.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	mso-themecolor:accent1;
	mso-font-kerning:0pt;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
h2
	{mso-style-priority:9;
	mso-style-qformat:yes;
	mso-style-link:"Titre 2 Car";
	margin-top:10.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:2;
	font-size:14.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	mso-themecolor:accent1;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
h3
	{mso-style-priority:9;
	mso-style-qformat:yes;
	mso-style-link:"Titre 3 Car";
	margin-top:10.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:3;
	font-size:12.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	mso-themecolor:accent1;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
h4
	{mso-style-priority:9;
	mso-style-qformat:yes;
	margin-top:10.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:4;
	font-size:12.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	mso-themecolor:accent1;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;
	font-weight:normal;
	mso-bidi-font-weight:bold;
	font-style:italic;
	mso-bidi-font-style:normal;}
h5
	{mso-style-priority:9;
	mso-style-qformat:yes;
	margin-top:10.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:5;
	font-size:12.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	mso-themecolor:accent1;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;
	font-weight:normal;
	mso-bidi-font-style:italic;}
h6
	{mso-style-priority:9;
	mso-style-qformat:yes;
	margin-top:10.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:6;
	font-size:12.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	mso-themecolor:accent1;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;
	font-weight:normal;}
p.MsoHeading7, li.MsoHeading7, div.MsoHeading7
	{mso-style-priority:9;
	mso-style-qformat:yes;
	margin-top:10.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:7;
	font-size:12.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	mso-themecolor:accent1;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoHeading8, li.MsoHeading8, div.MsoHeading8
	{mso-style-priority:9;
	mso-style-qformat:yes;
	margin-top:10.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:8;
	font-size:12.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	mso-themecolor:accent1;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoHeading9, li.MsoHeading9, div.MsoHeading9
	{mso-style-priority:9;
	mso-style-qformat:yes;
	margin-top:10.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:9;
	font-size:12.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	mso-themecolor:accent1;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoToc1, li.MsoToc1, div.MsoToc1
	{mso-style-update:auto;
	mso-style-priority:39;
	mso-style-next:Normal;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:5.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoToc2, li.MsoToc2, div.MsoToc2
	{mso-style-update:auto;
	mso-style-priority:39;
	mso-style-next:Normal;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:5.0pt;
	margin-left:12.0pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoToc3, li.MsoToc3, div.MsoToc3
	{mso-style-update:auto;
	mso-style-priority:39;
	mso-style-next:Normal;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:5.0pt;
	margin-left:24.0pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoFootnoteText, li.MsoFootnoteText, div.MsoFootnoteText
	{mso-style-priority:9;
	mso-style-qformat:yes;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoCommentText, li.MsoCommentText, div.MsoCommentText
	{mso-style-priority:99;
	mso-style-qformat:yes;
	mso-style-link:"Commentaire Car";
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoHeader, li.MsoHeader, div.MsoHeader
	{margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	tab-stops:center 8.0cm right 16.0cm;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoFooter, li.MsoFooter, div.MsoFooter
	{mso-style-priority:99;
	mso-style-link:"Pied de page Car";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	tab-stops:center 8.0cm right 16.0cm;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoCaption, li.MsoCaption, div.MsoCaption
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-link:"L\00E9gende Car";
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:6.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;
	font-style:italic;
	mso-bidi-font-style:normal;}
span.MsoCommentReference
	{mso-style-noshow:yes;
	mso-style-priority:99;
	mso-style-qformat:yes;
	mso-ansi-font-size:8.0pt;
	mso-bidi-font-size:8.0pt;}
p.MsoList, li.MsoList, div.MsoList
	{mso-style-unhide:no;
	mso-style-parent:"Corps de texte";
	margin-top:9.0pt;
	margin-right:0cm;
	margin-bottom:9.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:"Lohit Devanagari";
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoTitle, li.MsoTitle, div.MsoTitle
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	margin-top:24.0pt;
	margin-right:0cm;
	margin-bottom:12.0pt;
	margin-left:0cm;
	text-align:center;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:18.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#345A8A;
	mso-themecolor:accent1;
	mso-themeshade:181;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;
	font-weight:bold;}
p.MsoBodyText, li.MsoBodyText, div.MsoBodyText
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-link:"Corps de texte Car";
	margin-top:9.0pt;
	margin-right:0cm;
	margin-bottom:9.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoSubtitle, li.MsoSubtitle, div.MsoSubtitle
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:Titre;
	margin-top:12.0pt;
	margin-right:0cm;
	margin-bottom:12.0pt;
	margin-left:0cm;
	text-align:center;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:15.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#345A8A;
	mso-themecolor:accent1;
	mso-themeshade:181;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;
	font-weight:bold;}
p.MsoDate, li.MsoDate, div.MsoDate
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	margin:0cm;
	margin-bottom:.0001pt;
	text-align:center;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoBlockText, li.MsoBlockText, div.MsoBlockText
	{mso-style-priority:9;
	mso-style-qformat:yes;
	mso-style-parent:"Corps de texte";
	margin-top:5.0pt;
	margin-right:24.0pt;
	margin-bottom:5.0pt;
	margin-left:24.0pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
a:link, span.MsoHyperlink
	{mso-style-priority:99;
	color:blue;
	mso-themecolor:hyperlink;
	text-decoration:underline;
	text-underline:single;}
a:visited, span.MsoHyperlinkFollowed
	{mso-style-noshow:yes;
	color:purple;
	mso-themecolor:followedhyperlink;
	text-decoration:underline;
	text-underline:single;}
p.MsoCommentSubject, li.MsoCommentSubject, div.MsoCommentSubject
	{mso-style-noshow:yes;
	mso-style-qformat:yes;
	mso-style-parent:Commentaire;
	mso-style-link:"Objet du commentaire Car";
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;
	font-weight:bold;}
p.MsoAcetate, li.MsoAcetate, div.MsoAcetate
	{mso-style-noshow:yes;
	mso-style-qformat:yes;
	mso-style-link:"Texte de bulles Car";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:8.0pt;
	font-family:"Tahoma",sans-serif;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
span.MsoPlaceholderText
	{mso-style-noshow:yes;
	mso-style-unhide:no;
	color:gray;}
p.MsoRMPane, li.MsoRMPane, div.MsoRMPane
	{mso-style-noshow:yes;
	mso-style-unhide:no;
	mso-style-parent:"";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoListParagraph, li.MsoListParagraph, div.MsoListParagraph
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:36.0pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoListParagraphCxSpFirst, li.MsoListParagraphCxSpFirst, div.MsoListParagraphCxSpFirst
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-type:export-only;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:36.0pt;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoListParagraphCxSpMiddle, li.MsoListParagraphCxSpMiddle, div.MsoListParagraphCxSpMiddle
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-type:export-only;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:36.0pt;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoListParagraphCxSpLast, li.MsoListParagraphCxSpLast, div.MsoListParagraphCxSpLast
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-type:export-only;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:36.0pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoBibliography, li.MsoBibliography, div.MsoBibliography
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.MsoTocHeading, li.MsoTocHeading, div.MsoTocHeading
	{mso-style-priority:39;
	mso-style-qformat:yes;
	mso-style-parent:"Titre 1";
	margin-top:12.0pt;
	margin-right:0cm;
	margin-bottom:0cm;
	margin-left:0cm;
	margin-bottom:.0001pt;
	line-height:107%;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	mso-outline-level:1;
	font-size:16.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#365F91;
	mso-themecolor:accent1;
	mso-themeshade:191;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
span.LgendeCar
	{mso-style-name:"L\00E9gende Car";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-locked:yes;
	mso-style-link:L\00E9gende;}
span.VerbatimChar
	{mso-style-name:"Verbatim Char";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-locked:yes;
	mso-style-parent:"L\00E9gende Car";
	mso-style-link:"Source Code";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;}
span.FootnoteCharacters
	{mso-style-name:"Footnote Characters";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"L\00E9gende Car";
	vertical-align:super;}
span.FootnoteAnchor
	{mso-style-name:"Footnote Anchor";
	mso-style-unhide:no;
	mso-style-parent:"";
	vertical-align:super;}
span.InternetLink
	{mso-style-name:"Internet Link";
	mso-style-unhide:no;
	mso-style-parent:"L\00E9gende Car";
	color:#4F81BD;
	mso-themecolor:accent1;}
span.KeywordTok
	{mso-style-name:KeywordTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#007020;
	font-weight:bold;
	mso-bidi-font-weight:normal;}
span.DataTypeTok
	{mso-style-name:DataTypeTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#902000;}
span.DecValTok
	{mso-style-name:DecValTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#40A070;}
span.BaseNTok
	{mso-style-name:BaseNTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#40A070;}
span.FloatTok
	{mso-style-name:FloatTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#40A070;}
span.ConstantTok
	{mso-style-name:ConstantTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#880000;}
span.CharTok
	{mso-style-name:CharTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#4070A0;}
span.SpecialCharTok
	{mso-style-name:SpecialCharTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#4070A0;}
span.StringTok
	{mso-style-name:StringTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#4070A0;}
span.VerbatimStringTok
	{mso-style-name:VerbatimStringTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#4070A0;}
span.SpecialStringTok
	{mso-style-name:SpecialStringTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#BB6688;}
span.ImportTok
	{mso-style-name:ImportTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;}
span.CommentTok
	{mso-style-name:CommentTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#60A0B0;
	font-style:italic;
	mso-bidi-font-style:normal;}
span.DocumentationTok
	{mso-style-name:DocumentationTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#BA2121;
	font-style:italic;
	mso-bidi-font-style:normal;}
span.AnnotationTok
	{mso-style-name:AnnotationTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#60A0B0;
	font-weight:bold;
	mso-bidi-font-weight:normal;
	font-style:italic;
	mso-bidi-font-style:normal;}
span.CommentVarTok
	{mso-style-name:CommentVarTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#60A0B0;
	font-weight:bold;
	mso-bidi-font-weight:normal;
	font-style:italic;
	mso-bidi-font-style:normal;}
span.OtherTok
	{mso-style-name:OtherTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#007020;}
span.FunctionTok
	{mso-style-name:FunctionTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#06287E;}
span.VariableTok
	{mso-style-name:VariableTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#19177C;}
span.ControlFlowTok
	{mso-style-name:ControlFlowTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#007020;
	font-weight:bold;
	mso-bidi-font-weight:normal;}
span.OperatorTok
	{mso-style-name:OperatorTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#666666;}
span.BuiltInTok
	{mso-style-name:BuiltInTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;}
span.ExtensionTok
	{mso-style-name:ExtensionTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;}
span.PreprocessorTok
	{mso-style-name:PreprocessorTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#BC7A00;}
span.AttributeTok
	{mso-style-name:AttributeTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#7D9029;}
span.RegionMarkerTok
	{mso-style-name:RegionMarkerTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;}
span.InformationTok
	{mso-style-name:InformationTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#60A0B0;
	font-weight:bold;
	mso-bidi-font-weight:normal;
	font-style:italic;
	mso-bidi-font-style:normal;}
span.WarningTok
	{mso-style-name:WarningTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:#60A0B0;
	font-weight:bold;
	mso-bidi-font-weight:normal;
	font-style:italic;
	mso-bidi-font-style:normal;}
span.AlertTok
	{mso-style-name:AlertTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:red;
	font-weight:bold;
	mso-bidi-font-weight:normal;}
span.ErrorTok
	{mso-style-name:ErrorTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;
	color:red;
	font-weight:bold;
	mso-bidi-font-weight:normal;}
span.NormalTok
	{mso-style-name:NormalTok;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Verbatim Char";
	mso-ansi-font-size:11.0pt;
	font-family:Consolas;
	mso-ascii-font-family:Consolas;
	mso-hansi-font-family:Consolas;}
span.TextedebullesCar
	{mso-style-name:"Texte de bulles Car";
	mso-style-noshow:yes;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-locked:yes;
	mso-style-link:"Texte de bulles";
	mso-ansi-font-size:8.0pt;
	mso-bidi-font-size:8.0pt;
	font-family:"Tahoma",sans-serif;
	mso-ascii-font-family:Tahoma;
	mso-hansi-font-family:Tahoma;
	mso-bidi-font-family:Tahoma;}
span.En-tteCar
	{mso-style-name:"En-t\00EAte Car";
	mso-style-unhide:no;
	mso-style-qformat:yes;}
span.PieddepageCar
	{mso-style-name:"Pied de page Car";
	mso-style-priority:99;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-locked:yes;
	mso-style-link:"Pied de page";}
span.CommentaireCar
	{mso-style-name:"Commentaire Car";
	mso-style-priority:99;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-locked:yes;
	mso-style-link:Commentaire;
	mso-ansi-font-size:10.0pt;
	mso-bidi-font-size:10.0pt;}
span.ObjetducommentaireCar
	{mso-style-name:"Objet du commentaire Car";
	mso-style-noshow:yes;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-locked:yes;
	mso-style-parent:"Commentaire Car";
	mso-style-link:"Objet du commentaire";
	mso-ansi-font-size:10.0pt;
	mso-bidi-font-size:10.0pt;
	font-weight:bold;}
span.Titre1Car
	{mso-style-name:"Titre 1 Car";
	mso-style-priority:9;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-locked:yes;
	mso-style-link:"Titre 1";
	mso-ansi-font-size:16.0pt;
	mso-bidi-font-size:16.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	mso-themecolor:accent1;
	font-weight:bold;}
span.LienInternet
	{mso-style-name:"Lien Internet";
	mso-style-priority:99;
	mso-style-qformat:yes;
	color:blue;
	text-decoration:underline;
	text-underline:single;}
span.Titre2Car
	{mso-style-name:"Titre 2 Car";
	mso-style-priority:9;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-locked:yes;
	mso-style-link:"Titre 2";
	mso-ansi-font-size:14.0pt;
	mso-bidi-font-size:14.0pt;
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	mso-themecolor:accent1;
	font-weight:bold;}
span.tlid-translation
	{mso-style-name:tlid-translation;
	mso-style-unhide:no;
	mso-style-qformat:yes;}
span.CorpsdetexteCar
	{mso-style-name:"Corps de texte Car";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-locked:yes;
	mso-style-link:"Corps de texte";}
span.Titre3Car
	{mso-style-name:"Titre 3 Car";
	mso-style-priority:9;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-locked:yes;
	mso-style-link:"Titre 3";
	font-family:"Calibri",sans-serif;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:Calibri;
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:Calibri;
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	mso-themecolor:accent1;
	font-weight:bold;}
span.help-block
	{mso-style-name:help-block;
	mso-style-unhide:no;
	mso-style-qformat:yes;}
span.ListLabel1
	{mso-style-name:"ListLabel 1";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Cambria",serif;
	mso-fareast-font-family:Cambria;}
span.ListLabel2
	{mso-style-name:"ListLabel 2";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel3
	{mso-style-name:"ListLabel 3";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel4
	{mso-style-name:"ListLabel 4";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel5
	{mso-style-name:"ListLabel 5";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel6
	{mso-style-name:"ListLabel 6";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel7
	{mso-style-name:"ListLabel 7";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel8
	{mso-style-name:"ListLabel 8";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Cambria",serif;
	mso-fareast-font-family:Cambria;}
span.ListLabel9
	{mso-style-name:"ListLabel 9";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel10
	{mso-style-name:"ListLabel 10";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel11
	{mso-style-name:"ListLabel 11";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel12
	{mso-style-name:"ListLabel 12";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel13
	{mso-style-name:"ListLabel 13";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel14
	{mso-style-name:"ListLabel 14";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel15
	{mso-style-name:"ListLabel 15";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel16
	{mso-style-name:"ListLabel 16";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel17
	{mso-style-name:"ListLabel 17";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel18
	{mso-style-name:"ListLabel 18";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel19
	{mso-style-name:"ListLabel 19";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel20
	{mso-style-name:"ListLabel 20";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";}
span.ListLabel21
	{mso-style-name:"ListLabel 21";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel22
	{mso-style-name:"ListLabel 22";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel23
	{mso-style-name:"ListLabel 23";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel24
	{mso-style-name:"ListLabel 24";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel25
	{mso-style-name:"ListLabel 25";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel26
	{mso-style-name:"ListLabel 26";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel27
	{mso-style-name:"ListLabel 27";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel28
	{mso-style-name:"ListLabel 28";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel29
	{mso-style-name:"ListLabel 29";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel30
	{mso-style-name:"ListLabel 30";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel31
	{mso-style-name:"ListLabel 31";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel32
	{mso-style-name:"ListLabel 32";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel33
	{mso-style-name:"ListLabel 33";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel34
	{mso-style-name:"ListLabel 34";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel35
	{mso-style-name:"ListLabel 35";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel36
	{mso-style-name:"ListLabel 36";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel37
	{mso-style-name:"ListLabel 37";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel38
	{mso-style-name:"ListLabel 38";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel39
	{mso-style-name:"ListLabel 39";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel40
	{mso-style-name:"ListLabel 40";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel41
	{mso-style-name:"ListLabel 41";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel42
	{mso-style-name:"ListLabel 42";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel43
	{mso-style-name:"ListLabel 43";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel44
	{mso-style-name:"ListLabel 44";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel45
	{mso-style-name:"ListLabel 45";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel46
	{mso-style-name:"ListLabel 46";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel47
	{mso-style-name:"ListLabel 47";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel48
	{mso-style-name:"ListLabel 48";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel49
	{mso-style-name:"ListLabel 49";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel50
	{mso-style-name:"ListLabel 50";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel51
	{mso-style-name:"ListLabel 51";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel52
	{mso-style-name:"ListLabel 52";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel53
	{mso-style-name:"ListLabel 53";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel54
	{mso-style-name:"ListLabel 54";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	mso-ansi-font-size:10.0pt;}
span.ListLabel55
	{mso-style-name:"ListLabel 55";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	mso-ansi-font-size:10.0pt;}
span.ListLabel56
	{mso-style-name:"ListLabel 56";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	mso-ansi-font-size:10.0pt;}
span.ListLabel57
	{mso-style-name:"ListLabel 57";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	mso-ansi-font-size:10.0pt;}
span.ListLabel58
	{mso-style-name:"ListLabel 58";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	mso-ansi-font-size:10.0pt;}
span.ListLabel59
	{mso-style-name:"ListLabel 59";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	mso-ansi-font-size:10.0pt;}
span.ListLabel60
	{mso-style-name:"ListLabel 60";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	mso-ansi-font-size:10.0pt;}
span.ListLabel61
	{mso-style-name:"ListLabel 61";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	mso-ansi-font-size:10.0pt;}
span.ListLabel62
	{mso-style-name:"ListLabel 62";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	mso-ansi-font-size:10.0pt;}
span.ListLabel63
	{mso-style-name:"ListLabel 63";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel64
	{mso-style-name:"ListLabel 64";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel65
	{mso-style-name:"ListLabel 65";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel66
	{mso-style-name:"ListLabel 66";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel67
	{mso-style-name:"ListLabel 67";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel68
	{mso-style-name:"ListLabel 68";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel69
	{mso-style-name:"ListLabel 69";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Cambria",serif;
	mso-fareast-font-family:Cambria;}
span.ListLabel70
	{mso-style-name:"ListLabel 70";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel71
	{mso-style-name:"ListLabel 71";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel72
	{mso-style-name:"ListLabel 72";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel73
	{mso-style-name:"ListLabel 73";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel74
	{mso-style-name:"ListLabel 74";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel75
	{mso-style-name:"ListLabel 75";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel76
	{mso-style-name:"ListLabel 76";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Cambria",serif;
	mso-fareast-font-family:Cambria;}
span.ListLabel77
	{mso-style-name:"ListLabel 77";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel78
	{mso-style-name:"ListLabel 78";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel79
	{mso-style-name:"ListLabel 79";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel80
	{mso-style-name:"ListLabel 80";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel81
	{mso-style-name:"ListLabel 81";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel82
	{mso-style-name:"ListLabel 82";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Courier New";
	mso-bidi-font-family:"Courier New";}
span.ListLabel83
	{mso-style-name:"ListLabel 83";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Times New Roman",serif;
	mso-bidi-font-family:"Times New Roman";
	color:windowtext;}
span.ListLabel84
	{mso-style-name:"ListLabel 84";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";}
span.ListLabel85
	{mso-style-name:"ListLabel 85";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Times New Roman",serif;
	mso-ascii-font-family:"Times New Roman";
	mso-hansi-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";
	color:black;
	mso-themecolor:text1;}
span.ListLabel86
	{mso-style-name:"ListLabel 86";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	font-family:"Times New Roman",serif;
	mso-ascii-font-family:"Times New Roman";
	mso-hansi-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";}
p.Heading, li.Heading, div.Heading
	{mso-style-name:Heading;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-next:"Corps de texte";
	margin-top:12.0pt;
	margin-right:0cm;
	margin-bottom:6.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	page-break-after:avoid;
	font-size:14.0pt;
	font-family:"Liberation Sans",sans-serif;
	mso-fareast-font-family:"AR PL SungtiL GB";
	mso-bidi-font-family:"Lohit Devanagari";
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.Index, li.Index, div.Index
	{mso-style-name:Index;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan no-line-numbers;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:"Lohit Devanagari";
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.FirstParagraph, li.FirstParagraph, div.FirstParagraph
	{mso-style-name:"First Paragraph";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Corps de texte";
	margin-top:9.0pt;
	margin-right:0cm;
	margin-bottom:9.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.Compact, li.Compact, div.Compact
	{mso-style-name:Compact;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"Corps de texte";
	margin-top:1.8pt;
	margin-right:0cm;
	margin-bottom:1.8pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.Author, li.Author, div.Author
	{mso-style-name:Author;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	margin:0cm;
	margin-bottom:.0001pt;
	text-align:center;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.Abstract, li.Abstract, div.Abstract
	{mso-style-name:Abstract;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	margin-top:15.0pt;
	margin-right:0cm;
	margin-bottom:15.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:10.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.DefinitionTerm, li.DefinitionTerm, div.DefinitionTerm
	{mso-style-name:"Definition Term";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan lines-together;
	page-break-after:avoid;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;
	font-weight:bold;
	mso-bidi-font-weight:normal;}
p.Definition, li.Definition, div.Definition
	{mso-style-name:Definition;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.TableCaption, li.TableCaption, div.TableCaption
	{mso-style-name:"Table Caption";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:L\00E9gende;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:6.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	page-break-after:avoid;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;
	font-style:italic;
	mso-bidi-font-style:normal;}
p.ImageCaption, li.ImageCaption, div.ImageCaption
	{mso-style-name:"Image Caption";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:L\00E9gende;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:6.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;
	font-style:italic;
	mso-bidi-font-style:normal;}
p.Figure, li.Figure, div.Figure
	{mso-style-name:Figure;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.CaptionedFigure, li.CaptionedFigure, div.CaptionedFigure
	{mso-style-name:"Captioned Figure";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:Figure;
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	page-break-after:avoid;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
p.SourceCode, li.SourceCode, div.SourceCode
	{mso-style-name:"Source Code";
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-link:"Verbatim Char";
	margin-top:0cm;
	margin-right:0cm;
	margin-bottom:10.0pt;
	margin-left:0cm;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
span.SpellE
	{mso-style-name:"";
	mso-spl-e:yes;}
span.GramE
	{mso-style-name:"";
	mso-gram-e:yes;}
.MsoChpDefault
	{mso-style-type:export-only;
	mso-default-props:yes;
	font-size:12.0pt;
	mso-ansi-font-size:12.0pt;
	mso-bidi-font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:Cambria;
	mso-fareast-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:Cambria;
	mso-bidi-theme-font:minor-bidi;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
 /* Page Definitions */
 @page
	{mso-footnote-separator:url("DiyabcRandomForestUserManual21-01-2021_fichiers/header.htm") fs;
	mso-footnote-continuation-separator:url("DiyabcRandomForestUserManual21-01-2021_fichiers/header.htm") fcs;
	mso-endnote-separator:url("DiyabcRandomForestUserManual21-01-2021_fichiers/header.htm") es;
	mso-endnote-continuation-separator:url("DiyabcRandomForestUserManual21-01-2021_fichiers/header.htm") ecs;}
@page WordSection1
	{size:612.0pt 792.0pt;
	margin:1.0cm 1.0cm 1.0cm 1.0cm;
	mso-header-margin:0cm;
	mso-footer-margin:36.0pt;
	mso-forms-protection:no;
	mso-footer:url("DiyabcRandomForestUserManual21-01-2021_fichiers/header.htm") f1;
	mso-paper-source:0;}
div.WordSection1
	{page:WordSection1;}
 /* List Definitions */
 @list l0
	{mso-list-id:30500696;
	mso-list-template-ids:-707858734;}
@list l0:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l0:level2
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l0:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l0:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l0:level5
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l0:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l0:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l0:level8
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l0:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l1
	{mso-list-id:133304356;
	mso-list-template-ids:-1185352960;}
@list l1:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l1:level2
	{mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Cambria",serif;
	mso-bidi-font-family:Cambria;}
@list l1:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l1:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l1:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l1:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l1:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l1:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l1:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l2
	{mso-list-id:143202209;
	mso-list-type:hybrid;
	mso-list-template-ids:-524246588 67895307 67895299 67895301 67895297 67895299 67895301 67895297 67895299 67895301;}
@list l2:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0D8;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l2:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l2:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l2:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l2:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l2:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l2:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l2:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l2:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l3
	{mso-list-id:220217536;
	mso-list-template-ids:1917207830;}
@list l3:level1
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;}
@list l3:level2
	{mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;}
@list l3:level3
	{mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;}
@list l3:level4
	{mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;}
@list l3:level5
	{mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;}
@list l3:level6
	{mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;}
@list l3:level7
	{mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;}
@list l3:level8
	{mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;}
@list l3:level9
	{mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;}
@list l4
	{mso-list-id:287123809;
	mso-list-template-ids:1074324104;}
@list l4:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l4:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l4:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l4:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l4:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l4:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l4:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l4:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l4:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l5
	{mso-list-id:315844875;
	mso-list-template-ids:520901986;}
@list l5:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l5:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l5:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l5:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l5:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l5:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l5:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l5:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l5:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l6
	{mso-list-id:412747448;
	mso-list-template-ids:-182968210;}
@list l6:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l6:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l6:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l6:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l6:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l6:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l6:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l6:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l6:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l7
	{mso-list-id:436147099;
	mso-list-template-ids:-1413996254;}
@list l7:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l7:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l7:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l7:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l7:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l7:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l7:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l7:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l7:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l8
	{mso-list-id:487984896;
	mso-list-template-ids:831189962;}
@list l8:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F020;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l8:level2
	{mso-level-number-format:bullet;
	mso-level-text:\F020;
	mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l8:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F020;
	mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l8:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F020;
	mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l8:level5
	{mso-level-number-format:bullet;
	mso-level-text:\F020;
	mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l8:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F020;
	mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l8:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F020;
	mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l8:level8
	{mso-level-number-format:bullet;
	mso-level-text:\F020;
	mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l8:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F020;
	mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l9
	{mso-list-id:515389508;
	mso-list-template-ids:1869356982;}
@list l9:level1
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;}
@list l9:level2
	{mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;}
@list l9:level3
	{mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;}
@list l9:level4
	{mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;}
@list l9:level5
	{mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;}
@list l9:level6
	{mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;}
@list l9:level7
	{mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;}
@list l9:level8
	{mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;}
@list l9:level9
	{mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;}
@list l10
	{mso-list-id:543837260;
	mso-list-type:hybrid;
	mso-list-template-ids:-1205543104 67895297 67895299 67895301 67895297 67895299 67895301 67895297 67895299 67895301;}
@list l10:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l10:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l10:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l10:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l10:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l10:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l10:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l10:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l10:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l11
	{mso-list-id:563415426;
	mso-list-template-ids:-28932204;}
@list l11:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l11:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l11:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l11:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l11:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l11:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l11:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l11:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l11:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l12
	{mso-list-id:639186883;
	mso-list-template-ids:-828588578;}
@list l12:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l12:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l12:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l12:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l12:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l12:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l12:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l12:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l12:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l13
	{mso-list-id:680621734;
	mso-list-template-ids:206701058;}
@list l13:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l13:level2
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l13:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l13:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l13:level5
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l13:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l13:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l13:level8
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l13:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l14
	{mso-list-id:731974873;
	mso-list-template-ids:177927004;}
@list l14:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l14:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l14:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l14:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l14:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l14:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l14:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l14:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l14:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l15
	{mso-list-id:763842518;
	mso-list-template-ids:-8359516;}
@list l15:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l15:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l15:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l15:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l15:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l15:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l15:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l15:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l15:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l16
	{mso-list-id:765730731;
	mso-list-template-ids:1526366750;}
@list l16:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l16:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l16:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l16:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l16:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l16:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l16:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l16:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l16:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l17
	{mso-list-id:816990092;
	mso-list-template-ids:829485956;}
@list l17:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l17:level2
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l17:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l17:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l17:level5
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l17:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l17:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l17:level8
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l17:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l18
	{mso-list-id:830102740;
	mso-list-template-ids:-737147270;}
@list l18:level1
	{mso-level-number-format:roman-lower;
	mso-level-text:"\(%1\)";
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:39.0pt;
	text-indent:-36.0pt;}
@list l18:level2
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:57.0pt;
	text-indent:-18.0pt;}
@list l18:level3
	{mso-level-number-format:roman-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:right;
	margin-left:93.0pt;
	text-indent:-9.0pt;}
@list l18:level4
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:129.0pt;
	text-indent:-18.0pt;}
@list l18:level5
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:165.0pt;
	text-indent:-18.0pt;}
@list l18:level6
	{mso-level-number-format:roman-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:right;
	margin-left:201.0pt;
	text-indent:-9.0pt;}
@list l18:level7
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:237.0pt;
	text-indent:-18.0pt;}
@list l18:level8
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:273.0pt;
	text-indent:-18.0pt;}
@list l18:level9
	{mso-level-number-format:roman-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:right;
	margin-left:309.0pt;
	text-indent:-9.0pt;}
@list l19
	{mso-list-id:922228786;
	mso-list-template-ids:-338923104;}
@list l19:level1
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:54.0pt;
	text-indent:-18.0pt;}
@list l19:level2
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:90.0pt;
	text-indent:-18.0pt;}
@list l19:level3
	{mso-level-number-format:roman-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:right;
	margin-left:126.0pt;
	text-indent:-9.0pt;}
@list l19:level4
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:162.0pt;
	text-indent:-18.0pt;}
@list l19:level5
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:198.0pt;
	text-indent:-18.0pt;}
@list l19:level6
	{mso-level-number-format:roman-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:right;
	margin-left:234.0pt;
	text-indent:-9.0pt;}
@list l19:level7
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:270.0pt;
	text-indent:-18.0pt;}
@list l19:level8
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:306.0pt;
	text-indent:-18.0pt;}
@list l19:level9
	{mso-level-number-format:roman-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:right;
	margin-left:342.0pt;
	text-indent:-9.0pt;}
@list l20
	{mso-list-id:1042560041;
	mso-list-template-ids:-1968791418;}
@list l20:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l20:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l20:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l20:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l20:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l20:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l20:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l20:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l20:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l21
	{mso-list-id:1067530820;
	mso-list-template-ids:662064856;}
@list l21:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l21:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l21:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l21:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l21:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l21:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l21:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l21:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l21:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l22
	{mso-list-id:1115830137;
	mso-list-template-ids:-603702332;}
@list l22:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l22:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l22:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l22:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l22:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l22:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l22:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l22:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l22:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l23
	{mso-list-id:1130635503;
	mso-list-template-ids:1932170904;}
@list l23:level1
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;}
@list l23:level2
	{mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;}
@list l23:level3
	{mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;}
@list l23:level4
	{mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;}
@list l23:level5
	{mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;}
@list l23:level6
	{mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;}
@list l23:level7
	{mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;}
@list l23:level8
	{mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;}
@list l23:level9
	{mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;}
@list l24
	{mso-list-id:1165246800;
	mso-list-template-ids:-699606380;}
@list l24:level1
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;}
@list l24:level2
	{mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;}
@list l24:level3
	{mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;}
@list l24:level4
	{mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;}
@list l24:level5
	{mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;}
@list l24:level6
	{mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;}
@list l24:level7
	{mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;}
@list l24:level8
	{mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;}
@list l24:level9
	{mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;}
@list l25
	{mso-list-id:1262566619;
	mso-list-type:hybrid;
	mso-list-template-ids:1023299482 67895297 67895299 67895301 67895297 67895299 67895301 67895297 67895299 67895301;}
@list l25:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l25:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l25:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l25:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l25:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l25:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l25:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l25:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l25:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l26
	{mso-list-id:1271354287;
	mso-list-template-ids:1602622032;}
@list l26:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l26:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l26:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l26:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l26:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l26:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l26:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l26:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l26:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l27
	{mso-list-id:1434672144;
	mso-list-template-ids:1392009784;}
@list l27:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l27:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l27:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l27:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l27:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l27:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l27:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l27:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l27:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l28
	{mso-list-id:1471289799;
	mso-list-template-ids:815551254;}
@list l28:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l28:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l28:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l28:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l28:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l28:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l28:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l28:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l28:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l29
	{mso-list-id:1488015373;
	mso-list-template-ids:-1995400458;}
@list l29:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l29:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l29:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l29:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l29:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l29:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l29:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l29:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l29:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l30
	{mso-list-id:1489514662;
	mso-list-template-ids:522373804;}
@list l30:level1
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;}
@list l30:level2
	{mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;}
@list l30:level3
	{mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;}
@list l30:level4
	{mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;}
@list l30:level5
	{mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;}
@list l30:level6
	{mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;}
@list l30:level7
	{mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;}
@list l30:level8
	{mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;}
@list l30:level9
	{mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;}
@list l31
	{mso-list-id:1505322736;
	mso-list-template-ids:-912761498;}
@list l31:level1
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;}
@list l31:level2
	{mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;}
@list l31:level3
	{mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;}
@list l31:level4
	{mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;}
@list l31:level5
	{mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;}
@list l31:level6
	{mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;}
@list l31:level7
	{mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;}
@list l31:level8
	{mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;}
@list l31:level9
	{mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;}
@list l32
	{mso-list-id:1629778937;
	mso-list-template-ids:-994246976;}
@list l32:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l32:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l32:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l32:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l32:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l32:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l32:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l32:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l32:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l33
	{mso-list-id:1742019776;
	mso-list-template-ids:1050729462;}
@list l33:level1
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;}
@list l33:level2
	{mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;}
@list l33:level3
	{mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;}
@list l33:level4
	{mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;}
@list l33:level5
	{mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;}
@list l33:level6
	{mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;}
@list l33:level7
	{mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;}
@list l33:level8
	{mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;}
@list l33:level9
	{mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;}
@list l34
	{mso-list-id:1759593338;
	mso-list-type:hybrid;
	mso-list-template-ids:-692966358 67895297 67895299 67895301 67895297 67895299 67895301 67895297 67895299 67895301;}
@list l34:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l34:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l34:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l34:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l34:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l34:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l34:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l34:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l34:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l35
	{mso-list-id:1860926239;
	mso-list-type:hybrid;
	mso-list-template-ids:-311535566 67895297 67895299 67895301 67895297 67895299 67895301 67895297 67895299 67895301;}
@list l35:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l35:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l35:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l35:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l35:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l35:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l35:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;}
@list l35:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l35:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;}
@list l36
	{mso-list-id:1883902534;
	mso-list-template-ids:1801979258;}
@list l36:level1
	{mso-level-number-format:none;
	mso-level-suffix:none;
	mso-level-text:"";
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:0cm;
	text-indent:0cm;}
@list l36:level2
	{mso-level-number-format:none;
	mso-level-suffix:none;
	mso-level-text:"";
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:0cm;
	text-indent:0cm;}
@list l36:level3
	{mso-level-number-format:none;
	mso-level-suffix:none;
	mso-level-text:"";
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:0cm;
	text-indent:0cm;}
@list l36:level4
	{mso-level-number-format:none;
	mso-level-suffix:none;
	mso-level-text:"";
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:0cm;
	text-indent:0cm;}
@list l36:level5
	{mso-level-number-format:none;
	mso-level-suffix:none;
	mso-level-text:"";
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:0cm;
	text-indent:0cm;}
@list l36:level6
	{mso-level-number-format:none;
	mso-level-suffix:none;
	mso-level-text:"";
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:0cm;
	text-indent:0cm;}
@list l36:level7
	{mso-level-number-format:none;
	mso-level-suffix:none;
	mso-level-text:"";
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:0cm;
	text-indent:0cm;}
@list l36:level8
	{mso-level-number-format:none;
	mso-level-suffix:none;
	mso-level-text:"";
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:0cm;
	text-indent:0cm;}
@list l36:level9
	{mso-level-number-format:none;
	mso-level-suffix:none;
	mso-level-text:"";
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:0cm;
	text-indent:0cm;}
@list l37
	{mso-list-id:1910994166;
	mso-list-template-ids:-1810854666;}
@list l37:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l37:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l37:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l37:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l37:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l37:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l37:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l37:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l37:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l38
	{mso-list-id:1940212791;
	mso-list-template-ids:1595299598;}
@list l38:level1
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;}
@list l38:level2
	{mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;}
@list l38:level3
	{mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;}
@list l38:level4
	{mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;}
@list l38:level5
	{mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;}
@list l38:level6
	{mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;}
@list l38:level7
	{mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;}
@list l38:level8
	{mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;}
@list l38:level9
	{mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;}
@list l39
	{mso-list-id:1995909367;
	mso-list-template-ids:-1671244944;}
@list l39:level1
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l39:level2
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l39:level3
	{mso-level-number-format:roman-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:right;
	text-indent:-9.0pt;}
@list l39:level4
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l39:level5
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l39:level6
	{mso-level-number-format:roman-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:right;
	text-indent:-9.0pt;}
@list l39:level7
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l39:level8
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;}
@list l39:level9
	{mso-level-number-format:roman-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:right;
	text-indent:-9.0pt;}
@list l40
	{mso-list-id:2039888009;
	mso-list-template-ids:1016364636;}
@list l40:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:24.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l40:level2
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:36.0pt;
	mso-level-number-position:left;
	margin-left:60.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l40:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:72.0pt;
	mso-level-number-position:left;
	margin-left:96.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l40:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:108.0pt;
	mso-level-number-position:left;
	margin-left:132.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l40:level5
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:144.0pt;
	mso-level-number-position:left;
	margin-left:168.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l40:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:180.0pt;
	mso-level-number-position:left;
	margin-left:204.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l40:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:216.0pt;
	mso-level-number-position:left;
	margin-left:240.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l40:level8
	{mso-level-number-format:bullet;
	mso-level-text:\F02D;
	mso-level-tab-stop:252.0pt;
	mso-level-number-position:left;
	margin-left:276.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l40:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:288.0pt;
	mso-level-number-position:left;
	margin-left:312.0pt;
	text-indent:-24.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l41
	{mso-list-id:2046171599;
	mso-list-template-ids:-1983747694;}
@list l41:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l41:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l41:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l41:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l41:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l41:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l41:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l41:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l41:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l42
	{mso-list-id:2054844034;
	mso-list-template-ids:-2079028434;}
@list l42:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l42:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l42:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l42:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l42:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l42:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l42:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l42:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l42:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l43
	{mso-list-id:2118405395;
	mso-list-template-ids:1241537370;}
@list l43:level1
	{mso-level-start-at:5;
	mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Calibri",sans-serif;}
@list l43:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l43:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l43:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l43:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l43:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l43:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l43:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l43:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l44
	{mso-list-id:2120642271;
	mso-list-template-ids:-2107333996;}
@list l44:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l44:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l44:level3
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l44:level4
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l44:level5
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l44:level6
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
@list l44:level7
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Symbol;
	mso-bidi-font-family:Symbol;}
@list l44:level8
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:"Courier New";}
@list l44:level9
	{mso-level-number-format:bullet;
	mso-level-text:\F0A7;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-18.0pt;
	font-family:Wingdings;
	mso-bidi-font-family:Wingdings;}
ol
	{margin-bottom:0cm;}
ul
	{margin-bottom:0cm;}
-->
</style>
<!--[if gte mso 10]>
<style>
 /* Style Definitions */
 table.MsoNormalTable
	{mso-style-name:"Tableau Normal";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-priority:99;
	mso-style-parent:"";
	mso-padding-alt:0cm 5.4pt 0cm 5.4pt;
	mso-para-margin:0cm;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
table.Table
	{mso-style-name:Table;
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-qformat:yes;
	mso-style-parent:"";
	mso-padding-alt:0cm 5.4pt 0cm 5.4pt;
	mso-para-margin:0cm;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Cambria",serif;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-ansi-language:EN-US;
	mso-fareast-language:EN-US;}
</style>
<![endif]--><!--[if gte mso 9]><xml>
 <o:shapedefaults v:ext="edit" spidmax="1027"/>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <o:shapelayout v:ext="edit">
  <o:idmap v:ext="edit" data="1"/>
 </o:shapelayout></xml><![endif]-->
</head>

<body lang=FR link=blue vlink=purple style='tab-interval:36.0pt'>

<div class=WordSection1>

<h1 align=center style='text-align:center'><a name="_Toc62127923"></a><a
name="_Toc62034874"><span style='mso-bookmark:_Toc62127923'><span lang=EN-US
style='font-size:20.0pt'>USER MANUAL for DIYABC Random Forest v1.0</span></span></a><span
lang=EN-US style='font-size:20.0pt'><o:p></o:p></span></h1>

<h1 align=center style='margin-left:36.0pt;text-align:center'><a
name="_Toc62127924"></a><a name="_Toc62034875"><span style='mso-bookmark:_Toc62127924'><span
lang=EN-US style='font-size:14.0pt'>- 21th of January 2021 -</span></span></a><span
lang=EN-US style='font-size:14.0pt'><o:p></o:p></span></h1>

<p class=MsoBodyText align=center style='text-align:center'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:200%'><span lang=EN-US style='font-family:"Times New Roman",serif'>Fran&ccedil;ois-David
Collin <sup>1,*</sup>, Ghislain Durif <sup>1,*</sup>, Louis <span class=SpellE>Raynal</span>
<sup>1</sup>, Eric <span class=SpellE>Lombaert</span> <sup>2</sup>, Mathieu
Gautier <sup>3</sup>, Renaud Vitalis <sup>3</sup>, Jean-Michel Marin <sup>1,&amp;</sup>,
Arnaud Estoup <sup>3,&amp;</sup><o:p></o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:200%'><span lang=EN-US style='font-family:"Times New Roman",serif'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:200%'><span lang=EN-US style='font-family:"Times New Roman",serif'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:200%'><sup><span lang=EN-US style='font-family:"Times New Roman",serif;
mso-fareast-font-family:Calibri'>1</span></sup><span lang=EN-US
style='font-family:"Times New Roman",serif;mso-fareast-font-family:Calibri'>
IMAG, Univ Montpellier, CNRS, UMR 5149, Montpellier, France<o:p></o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:200%'><sup><span lang=EN-US style='font-family:"Times New Roman",serif'>2
</span></sup><span lang=EN-US style='font-family:"Times New Roman",serif'>ISA,
INRAE, CNRS, Univ C&ocirc;te d&#8217;Azur,</span><span lang=EN-US> </span><span
lang=EN-US style='font-family:"Times New Roman",serif'>Sophia Antipolis, France<o:p></o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:200%'><sup><span lang=EN-US style='font-family:"Times New Roman",serif'>3</span></sup><span
lang=EN-US style='font-family:"Times New Roman",serif'> CBGP, Univ Montpellier,
CIRAD, INRAE, <span class=SpellE>Institut</span> <span class=SpellE>Agro</span>,
IRD, Montpellier, France<o:p></o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:200%'><span lang=EN-US style='font-family:"Times New Roman",serif'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:150%'><sup><span lang=EN-US style='mso-bidi-font-family:
"Times New Roman"'>*</span></sup><span lang=EN-US style='mso-bidi-font-family:
"Times New Roman"'> Equal contribution (F-D.C.: computation part of the
program; S.D.: interface part of the program)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:150%'><sup><span lang=EN-US style='mso-bidi-font-family:
"Times New Roman"'>&amp;</span></sup><span lang=EN-US style='mso-bidi-font-family:
"Times New Roman"'> These authors are joint senior authors on this work<o:p></o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:150%'><span lang=EN-US style='mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:150%'><span lang=EN-US style='mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:150%'><span lang=EN-US style='mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:150%'><span lang=EN-US style='mso-bidi-font-family:"Times New Roman"'>Corresponding
authors: <o:p></o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:150%'><span lang=EN-US style='mso-bidi-font-family:"Times New Roman"'>Arnaud
Estoup. E-mail: </span><span lang=EN-US><a href="mailto:arnaud.estoup@.inrae.fr"><span
class=InternetLink><span style='mso-bidi-font-family:"Times New Roman";
color:windowtext;text-decoration:none;text-underline:none'>arnaud.estoup@.inrae.fr</span></span></a></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:150%'><span style='mso-bidi-font-family:"Times New Roman";
mso-ansi-language:FR'>Francois-<span class=SpellE>David.Collin</span>:
Francois-David.Collin@umontpellier.fr<o:p></o:p></span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;text-align:
justify;line-height:150%'><span lang=EN-US style='mso-bidi-font-family:"Times New Roman"'>Ghislain
Durif: ghislain.durif@umontpellier.fr<o:p></o:p></span></p>

<p class=MsoBodyText><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<span lang=EN-US style='font-size:12.0pt;font-family:"Cambria",serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
minor-latin;mso-hansi-theme-font:minor-latin;mso-bidi-font-family:Cambria;
mso-bidi-theme-font:minor-bidi;mso-ansi-language:EN-US;mso-fareast-language:
EN-US;mso-bidi-language:AR-SA'><br clear=all style='mso-special-character:line-break;
page-break-before:always'>
</span>

<p class=MsoNormal><a name=preface></a><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<h1><a name="_Toc62127925"></a><a name="_Toc62034876"><span style='mso-bookmark:
_Toc62127925'><span lang=EN-US>CONTENTS</span></span></a></h1>

<p class=MsoBodyText><b style='mso-bidi-font-weight:normal'><span lang=EN-US>Warning:
You might want to activate the option &#8220;Navigation panel&#8221; (accessible
through the &#8220;Display&#8221; tab) after opening the present manual file<i
style='mso-bidi-font-style:normal'>. </i>This navigation panel, visible on the
left size of the document once activated, will allow you reaching directly
through simple click actions the different sections and sub-sections described
below.<o:p></o:p></span></b></p>

<w:Sdt SdtDocPart="t" DocPartType="Table of Contents" DocPartUnique="t"
 ID="-1281259351">
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><a name="_Toc45280460"></a><span
 style='mso-bookmark:_Toc45280460'></span><!--[if supportFields]><span
 lang=EN-US style='font-size:16.0pt;font-family:"Calibri",sans-serif;
 mso-ascii-theme-font:major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
 major-fareast;mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;
 color:#365F91;mso-themecolor:accent1;mso-themeshade:191'><span
 style='mso-element:field-begin'></span></span><span lang=EN-US><span
 style='mso-spacerun:yes'>&nbsp;</span>TOC \o &quot;1-3&quot; \h \z \u </span><span
 lang=EN-US style='font-size:16.0pt;font-family:"Calibri",sans-serif;
 mso-ascii-theme-font:major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
 major-fareast;mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;
 color:#365F91;mso-themecolor:accent1;mso-themeshade:191'><span
 style='mso-element:field-separator'></span></span><![endif]--><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span><span
 style='mso-ansi-language:FR'><w:sdtPr></w:sdtPr></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127926"><span style='mso-no-proof:yes'>1. (BRIEF) INTRODUCTION</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp; </span>4</span></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127927"><span style='mso-no-proof:yes'>1.1 General context</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127927 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>4<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900320037000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127928"><span style='mso-no-proof:yes'>1.2 How to cite the program
 DIYABC Random Forest v1.0</span><span style='color:windowtext;display:none;
 mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127928 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>4<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900320038000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127929"><span style='mso-no-proof:yes'>1.3 Web site</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127929 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>4<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900320039000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127930"><span style='mso-no-proof:yes'>1.4. System requirements,
 installing and launching the program</span><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127930 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>4<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900330030000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127931"><span style='mso-no-proof:yes'>1.5 Acknowledgements</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127931 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>5<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900330031000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127932"><span style='mso-no-proof:yes'>2. FORMALIZATION OF
 SCENARIOS AND GENERATION OF THE TRAINING SET</span><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127932 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>5<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900330032000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127933"><span style='mso-no-proof:yes'>2.1 Basic notions</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127933 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>5<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900330033000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127934"><span style='mso-no-proof:yes'>2.2 Main features of
 algorithms for data simulation</span><span style='color:windowtext;display:
 none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127934 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>5<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900330034000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127935"><span style='mso-no-proof:yes'>2.3 Historical model
 parameterization</span><span style='color:windowtext;display:none;mso-hide:
 screen;mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127935 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>7<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900330035000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc3 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127936"><span style='mso-no-proof:yes'>2.3.1 Key notes</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127936 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>8<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900330036000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc3 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127937"><span style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:
 major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
 mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;mso-bidi-font-weight:
 bold;mso-no-proof:yes'>2.3.2 Examples</span><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127937 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>9<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900330037000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127938"><span style='mso-no-proof:yes'>2.4 Mutation model
 parameterization (microsatellite and DNA sequence loci)</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127938 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>12<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900330038000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc3 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127939"><span style='mso-no-proof:yes'>2.4.1 Microsatellite loci</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127939 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>13<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900330039000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc3 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127940"><span style='mso-no-proof:yes'>2.4.2 DNA sequence loci</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127940 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>13<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900340030000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc3 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127941"><span style='mso-no-proof:yes'>2.4.3 SNPs do not require
 mutation model parameterization &#8211; notion of MAF and MRC</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127941 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>13<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900340031000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127942"><span style='mso-no-proof:yes'>2.5 Prior distributions</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127942 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>14<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900340032000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127943"><span style='mso-no-proof:yes'>2.6 Summary statistics as
 components of the feature vector</span><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127943 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>15<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900340033000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc3 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127944"><span style='mso-no-proof:yes'>2.6.1 For microsatellite
 loci</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127944 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>15<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900340034000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc3 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127945"><span style='mso-no-proof:yes'>2.6.2 For DNA sequence
 loci</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127945 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>15<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900340035000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc3 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127946"><span style='mso-no-proof:yes'>2.6.3 For SNP loci</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127946 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>16<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900340036000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127947"><span style='mso-no-proof:yes'>2.7 Generating the
 training set</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127947 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>16<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900340037000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127948"><span style='mso-no-proof:yes'>3. RANDOM FOREST ANALYSIS</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127948 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>17<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900340038000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127950"><span style='mso-no-proof:yes'>3.1 Addition of linear
 combinations of summary statistics to the vector feature</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127950 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>17<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900350030000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127951"><span style='mso-no-proof:yes'>3.2 Prediction using
 Random Forest: scenario choice</span><span style='color:windowtext;display:
 none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127951 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>17<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900350031000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127952"><span style='mso-no-proof:yes'>3.3 Prediction using
 Random Forest: parameter estimation</span><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127952 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>17<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900350032000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127953"><span style='mso-no-proof:yes'>3.4 Assessing the quality
 of predictions</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127953 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>18<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900350033000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc3 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127954"><span style='mso-no-proof:yes'>3.4.1 Indices for scenario
 choice</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127954 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>18<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900350034000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc3 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127955"><span style='mso-no-proof:yes'>3.4.2 Indices for
 parameter estimation</span><span style='color:windowtext;display:none;
 mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127955 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>18<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900350035000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127956"><span style='mso-no-proof:yes'>4. PRACTICAL
 CONSIDERATIONS FOR ABC-RF TREATMENTS</span><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127956 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>18<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900350036000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127957"><span style='mso-no-proof:yes'>4.1 Are my scenarios
 and/or associated priors compatible with the observed dataset?</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127957 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>19<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900350037000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127958"><span style='mso-no-proof:yes'>4.2 Did I simulate enough
 datasets for my training set?</span><span style='color:windowtext;display:
 none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127958 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>19<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900350038000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127959"><span style='mso-no-proof:yes'>4.3 Did my forest grow
 enough trees?</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127959 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>19<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900350039000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127960"><span style='mso-no-proof:yes'>5. RUNNING (EXAMPLE)
 DATASET TREATMENTS USING THE GRAPHIC USER INTERFACE (GUI)</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127960 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>20<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900360030000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127961"><span style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:
 major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
 mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;mso-bidi-font-weight:
 bold;mso-no-proof:yes'>5.1 Launching the GUI</span><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'> </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127961 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>20<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900360031000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127962"><span style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:
 major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
 mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;mso-bidi-font-weight:
 bold;mso-no-proof:yes'>5.2 What is a DIYABC Random Forest project?</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127962 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>20<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900360032000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127963"><span style='mso-no-proof:yes'>5.3 Main options of the
 home screen</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127963 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>21<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900360033000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127964"><span style='mso-no-proof:yes'>5.4 How to generate a new
 IndSeq SNP training set</span><span style='color:windowtext;display:none;
 mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127964 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>22<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900360034000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127965"><span style='mso-no-proof:yes'>5.4.1 Step 1: defining a
 new IndSeq SNP project</span><span style='color:windowtext;display:none;
 mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127965 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>22<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900360035000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127966"><span style='mso-no-proof:yes'>5.4.2 Step 2: choosing the
 data file</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127966 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>23<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900360036000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127967"><span style='mso-no-proof:yes'>5.4.3 Step 3: Inform the
 Historical model</span><span style='color:windowtext;display:none;mso-hide:
 screen;mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'> </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127967 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>25<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900360037000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127968"><span style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:
 major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
 mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;mso-bidi-font-weight:
 bold;mso-no-proof:yes'>5.4.4 Step 4: Inform number of loci to simulate (and
 chromosome type)</span><span style='color:windowtext;display:none;mso-hide:
 screen;mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127968 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>27<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900360038000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127969"><span style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:
 major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
 mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;mso-bidi-font-weight:
 bold;mso-no-proof:yes'>5.4.5 Step 5: Summary statistics</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127969 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>28<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900360039000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127970"><span style='mso-no-proof:yes'>5.4.6 Step 6: Simulate the
 training set</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127970 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>29<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900370030000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127971"><span style='mso-no-proof:yes'>5.4.7 Step 7:
 Prior-scenario checking PRE-ANALYSIS (optional but recommended)</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127971 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>30<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900370031000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127972"><span style='mso-no-proof:yes'>5.5 How to generate a new
 PoolSeq SNP training set</span><span style='color:windowtext;display:none;
 mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127972 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>31<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900370032000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127973"><span style='mso-no-proof:yes'>5.6 How to generate a new
 microsatellite (and/or DNA sequence) training set</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127973 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>31<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900370033000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc3 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127974"><span style='mso-no-proof:yes'>5.6.1 Step 4: Inform the
 genetic model</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'> </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127974 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>32<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900370034000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc3 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127975"><span style='mso-no-proof:yes'>5.6.2 Step 5: Summary
 statistics</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127975 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>36<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900370035000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127976"><span style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:
 major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
 mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;mso-bidi-font-weight:
 bold;mso-no-proof:yes'>5.7 How to work from an &#8220;Existing project&#8221;
 (for any type of markers)</span><span style='color:windowtext;display:none;
 mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127976 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>37<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900370036000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127977"><span style='mso-no-proof:yes'>6. PERFORMING RANDOM
 FOREST ANALYSES</span><span style='color:windowtext;display:none;mso-hide:
 screen;mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127977 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>40<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900370037000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127978"><span style='mso-no-proof:yes'>6.1 Files needed</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127978 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>40<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900370038000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127979"><span style='mso-no-proof:yes'>6.2. Scenario choice
 analysis</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127979 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>41<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900370039000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127980"><span style='mso-no-proof:yes'>6.3. Parameter estimation
 analysis</span><span style='color:windowtext;display:none;mso-hide:screen;
 mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127980 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>44<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900380030000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127981"><span style='mso-no-proof:yes'>7. KEY FILES</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127981 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>47<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900380031000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127982"><span style='mso-no-proof:yes'>7.1 Data files</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127982 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>47<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900380032000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127983"><span style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:
 major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
 mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;mso-bidi-font-weight:
 bold;mso-no-proof:yes'>7.1.1 IndSeq SNP data</span><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127983 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>47<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900380033000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127984"><span style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:
 major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
 mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;mso-bidi-font-weight:
 bold;mso-no-proof:yes'>7.1.2 PoolSeq SNP data</span><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127984 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>49<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900380034000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127985"><span style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:
 major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
 mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;mso-bidi-font-weight:
 bold;mso-no-proof:yes'>7.1.3 Microsatellite and DNA sequence data</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127985 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>51<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900380035000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127986"><span style='mso-no-proof:yes'>7.2 Training set file(s)</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127986 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>53<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900380036000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127987"><span style='mso-no-proof:yes'>7.3 Output files produced
 by a Random Forest analysis</span><span style='color:windowtext;display:none;
 mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:none'><span
 style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127987 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>53<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900380037000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc2 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127988"><span style='mso-no-proof:yes'>7.4 Other files</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127988 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>56<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900380038000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127989"><span style='mso-fareast-font-family:Calibri;mso-fareast-theme-font:
 major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
 mso-bidi-font-weight:bold;mso-no-proof:yes'>8. GENERATE SYNTHETIC DATAFILES</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127989 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>56<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900380039000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127990"><span style='mso-fareast-font-family:Calibri;mso-fareast-theme-font:
 major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
 mso-bidi-font-weight:bold;mso-no-proof:yes'>9. USING DIYABC RANDOM FOREST ON A
 COMPUTER SERVER AND EXAMPLES OF COMMAND LINES</span><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127990 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>57<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900390030000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127991"><span style='mso-fareast-font-family:Calibri;mso-fareast-theme-font:
 major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
 mso-bidi-font-weight:bold;mso-no-proof:yes'>10. TOY EXAMPLES</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127991 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>59<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900390031000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoToc1 style='tab-stops:right 554.8pt'><span lang=EN-US><a
 href="#_Toc62127992"><span style='mso-fareast-font-family:Calibri;mso-fareast-theme-font:
 major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
 mso-bidi-font-weight:bold;mso-no-proof:yes'>11. REFERENCES CITED</span><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-tab-count:1'>&nbsp;&nbsp; </span></span><!--[if supportFields]><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'><span style='mso-element:field-begin'></span>
 PAGEREF _Toc62127992 \h <span style='mso-element:field-separator'></span></span><![endif]--><span
 style='color:windowtext;display:none;mso-hide:screen;mso-no-proof:yes;
 text-decoration:none;text-underline:none'>60<!--[if gte mso 9]><xml>
  <w:data>08D0C9EA79F9BACE118C8200AA004BA90B02000000080000000D0000005F0054006F006300360032003100320037003900390032000000</w:data>
 </xml><![endif]--></span><!--[if supportFields]><span style='color:windowtext;
 display:none;mso-hide:screen;mso-no-proof:yes;text-decoration:none;text-underline:
 none'><span style='mso-element:field-end'></span></span><![endif]--></a></span><span
 style='font-size:11.0pt;mso-fareast-font-family:Cambria;mso-fareast-theme-font:
 minor-fareast;mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><o:p></o:p></span></p>
 <p class=MsoNormal><!--[if supportFields]><b><span lang=EN-US><span
 style='mso-element:field-end'></span></span></b><![endif]--><span lang=EN-US><o:p>&nbsp;</o:p></span></p>
</w:Sdt>

<h1><a name=general-context-and-history-of-diyabc></a><a name="_Toc62127927"><span
style='mso-bookmark:general-context-and-history-of-diyabc'><span lang=EN-US>1. (BRIEF)
INTRODUCTION</span></span></a></h1>

<h2><span style='mso-bookmark:_Toc62127927'><span style='mso-bookmark:general-context-and-history-of-diyabc'><span
lang=EN-US>1.1 General context</span></span></span><span style='mso-bookmark:
general-context-and-history-of-diyabc'><span lang=EN-US> </span></span></h2>

<p class=MsoNormal style='margin-left:24.0pt'><span lang=EN-US>Simulation-based
methods such as Approximate Bayesian Computation (ABC) are well adapted to the
analysis of complex models of populations and species genetic history (Beaumont
2010). In this context, supervised machine learning (SML) methods provide
attractive statistical solutions to conduct efficient inferences about both
scenario choice and parameter estimation (</span><span class=SpellE><span
lang=EN-US style='font-family:"Times New Roman",serif'>Schrider</span></span><span
lang=EN-US style='font-family:"Times New Roman",serif'> &amp; Kern 2018)</span><span
lang=EN-US>. The Random Forest methodology (RF) is a powerful ensemble of SML
algorithms used for both classification and/or regression problems (<span
class=SpellE>Breiman</span> 2001). RF allows conducting inferences at a lower
computational cost than ABC, without preliminary selection of the relevant
components of the ABC summary statistics, and bypassing the derivation of ABC
tolerance levels (<span class=SpellE>Pudlo</span> et al. 2018; <span
class=SpellE>Raynal</span> et al. 2019). We have implemented a set of RF
algorithms to process inferences using simulated datasets generated from an
extended version of the population genetic simulator implemented in DIYABC
v2.1.0 (<span class=SpellE>Cornuet</span> et al. 2014). The resulting computer
package, named DIYABC Random Forest v1.0, integrates two functionalities into a
user-friendly interface: the simulation under custom evolutionary scenarios of
different types of molecular data (microsatellites, DNA sequences or SNPs
&#8211; including traditional <span class=SpellE><b style='mso-bidi-font-weight:
normal'>IndSeq</b></span> and more recent <span class=SpellE><b
style='mso-bidi-font-weight:normal'>PoolSeq</b></span> SNP data) and RF
treatments including statistical tools to evaluate the power and accuracy of
inferences (Collin et al. 2020). Because of the properties inherent of the
implemented RF methods and the large feature vector (including various summary
statistics and their linear combinations) available for SNP data, DIYABC Random
Forest v1.0 can efficiently contribute to the analysis of large-size SNP (and
other molecular markers) datasets to make inferences about complex population
genetic histories.<b style='mso-bidi-font-weight:normal'><o:p></o:p></b></span></p>

<h2><a name="_Toc62127928"><span lang=EN-US>1.2 How to cite the program DIYABC
Random Forest v1.0</span></a></h2>

<p class=MsoListParagraph style='margin-left:42.0pt;mso-add-space:auto'><span
lang=EN-US>Collin F-D, Durif G, <span class=SpellE>Raynal</span> L, <span
class=SpellE>Lombaert</span> E, Gautier M, Vitalis R, Marin J-M, Estoup A (2021)
DIYABC Random Forest v1.0: extending approximate Bayesian computation with
supervised machine learning to infer demographic history from genetic
polymorphisms. <i style='mso-bidi-font-style:normal'>Molecular Ecology
Resources</i>. DOI: 10.22541/au.159480722.26357192</span></p>

<h2><a name="_Toc62127929"><span lang=EN-US>1.3 Web site</span></a></h2>

<p class=FirstParagraph><span lang=EN-US><a href="https://diyabc.github.io/"><span
class=InternetLink><span style='text-decoration:none;text-underline:none'>https://diyabc.github.io</span></span></a>.
</span></p>

<p class=FirstParagraph><span lang=EN-US>You can get from this <span
class=SpellE>github</span> site the executable files for different operating
systems, the latest version of this manual document, as well as examples of
DIYABC Random Forest analyses for different type of markers.</span></p>

<h2><a name="_Toc62127930"></a><a name=system-requirements><span
style='mso-bookmark:_Toc62127930'><span lang=EN-US>1.4. System requirement</span></span></a><span
style='mso-bookmark:_Toc62127930'><span lang=EN-US>s, installing and launching
the program</span></span></h2>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l17 level1 lfo39'><a
name=acknowledgements><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US style='font-family:"Times New Roman",serif'>The
software DIYABC Random Forest (hereafter DIYABC-RF) v1.0 is composed of three
parts: the dataset simulator, the Random Forest inference engine and the
graphical user interface. The whole is packaged as a standalone and
user-friendly graphical application named DIYABC-RF GUI and available at </span></a><a
href="https://diyabc.github.io/"><span style='mso-bookmark:acknowledgements'><span
lang=EN-US style='font-family:"Times New Roman",serif;color:windowtext;
text-decoration:none;text-underline:none'>https://diyabc.github.io</span></span></a><span
style='mso-bookmark:acknowledgements'><span lang=EN-US style='font-family:"Times New Roman",serif'>.
The different developer and user manuals for each component of the software are
available on the same website. DIYABC-RF is a multithreaded software on three
operating systems: GNU/Linux, Microsoft Windows and MacOS. One can use the
program can be used through a modern and user-friendly graphical interface
designed as an R shiny application (Chang et al. 2019). For a fluid and
simplified user experience, this interface is available through a standalone
application, which does not </span></span><span style='mso-bookmark:acknowledgements'><span
lang=EN-US style='font-family:"Times New Roman",serif;mso-fareast-font-family:
Calibri'>require installing R or any dependencies </span></span><span
style='mso-bookmark:acknowledgements'><span lang=EN-US style='font-family:"Times New Roman",serif'>and
hence can be used independently. The application is also implemented in an R
package providing a standard shiny web application (with the same graphical
interface) that can be run locally as any shiny application, or hosted as a web
service to provide a DIYABC-RF server for multiple users.</span></span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l17 level1 lfo39'><span
style='mso-bookmark:acknowledgements'><![if !supportLists]><span lang=EN-US
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US style='font-family:"Times New Roman",serif'>To
install and run the program, please check the instructions available on the
website https://diyabc.github.io/</span></span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l17 level1 lfo39'><span
style='mso-bookmark:acknowledgements'><![if !supportLists]><span lang=EN-US
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Minimum 4GB of RAM; 6GB of RAM
recommended</span></span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l17 level1 lfo39'><span
style='mso-bookmark:acknowledgements'><![if !supportLists]><span lang=EN-US
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>From 1 to 10 GB free disk space
for each </span></span><span style='mso-bookmark:acknowledgements'><span
lang=EN-US style='font-family:"Times New Roman",serif'>DIYABC-RF</span><span
lang=EN-US> project depending on the project configuration and the number of
simulated datasets recorded in the training set file (<span class=SpellE>reftableRF.bin</span>
file).</span></span></p>

<h2><span style='mso-bookmark:acknowledgements'><a name="_Toc62127931"><span
lang=EN-US>1.5 Acknowledgements</span></a></span></h2>

<p class=FirstParagraph><span lang=EN-US>We thank Pierre <span class=SpellE>Pudlo</span>
for useful discussions and Jean-Marie <span class=SpellE>Cornuet</span> for
computer code expertise at the onset of the ABC Random Forest project. We also
thank several &#8220;beta-users&#8221;, especially XXXNAMEXXX, who tested the
software DIYABC Random Forest v1.0 with their data. This work was supported by
funds from the French <span class=SpellE>Agence</span> National pour la
Recherche (ANR projects SWING and GANDHI), the INRAE scientific division SPE
(AAP-SPE 2016), and the <span class=SpellE>LabEx</span> NUMEV (NUMEV,
ANR10-LABX-20).</span></p>

<h1><a name="_Toc62127932"><span lang=EN-US>2. FORMALIZATION OF SCENARIOS AND
GENERATION OF THE TRAINING SET</span></a></h1>

<h2><a name="_Toc62127933"></a><a name=basic-notions-on-abc></a><a
name=methodology></a><span style='mso-bookmark:_Toc62127933'><span
style='mso-bookmark:basic-notions-on-abc'><span lang=EN-US>2.1 Basic notions</span></span></span></h2>

<p class=FirstParagraph><span lang=EN-US>Before processing Random Forest
analyses, one need to generate a <i style='mso-bidi-font-style:normal'>training
set</i> which corresponds to the so called <i style='mso-bidi-font-style:normal'>reference
table</i> in a standard ABC framework (<span class=SpellE>Cornuet</span> et al.
2014 and see the associated program DIYABC v2.1.0). The datasets composing the
training set can be simulated under different models (hereafter referred to as
scenarios) and samples (i.e. <span class=SpellE>nb</span> of markers and
sampled individuals) configurations, using parameter values drawn from prior
distributions. Each resulting dataset is summarized using a set of descriptive
statistics. Scenarios and prior distributions are formalized in the main
pipeline of the software DIYABC Random Forest v1.0 (DIYABC-RF) and summary
statistics are computed using the &#8220;<u>Training set simulation</u>&#8221; module
of the main pipeline of DIYABC-RF, which essentially corresponds to an extended
version of the population genetics simulator implemented in DIYABC v2.1.0 (<span
class=SpellE>Cornuet</span> et al. 2014). As in the latter program, DIYABC-RF allows
considering complex population histories including any combination of
population divergence events, symmetrical or asymmetrical admixture events (but
not any continuous gene flow between populations) and changes in past
population size, with population samples potentially collected at different
times. Statistical analysis based on Random Forest algorithms of an observed
dataset using a given training set are computed using the &#8220;<u>Random
Forest Analysis&#8221;</u> module of the main pipeline of DIYABC-RF.</span></p>

<h2><a name="_Toc62127934"></a><a
name=X45f629f13f061e96ee74162059f3a93975759df><span style='mso-bookmark:_Toc62127934'><span
lang=EN-US>2.2 Main features of algorithms for data simulation</span></span></a></h2>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US>The ABC part of DIYABC-RF is a simulation-based method. Data
simulation is based on the Wright-Fisher model. It consists in generating the
genealogy of all sampled genes until their most recent common ancestor using
(backward in time) coalescence theory. This begins by randomly drawing a
complete set of parameters from their own prior distributions and that satisfy
all imposed conditions. Then, once events have been ordered by increasing
times, a sequence of <i style='mso-bidi-font-style:normal'>actions</i> is
constructed. If there is more than one locus, the same sequence of actions is
used for all successive loci. </span></p>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US>Possible <i style='mso-bidi-font-style:normal'>actions</i> fall into
four categories:</span></p>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US>- <b style='mso-bidi-font-weight:normal'>adding a sample to a
population</b> = Add as many gene lineages to the population as there are genes
in the sample.</span></p>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US>- <b style='mso-bidi-font-weight:normal'>merge two populations</b> =
Move the lineages of the second population into the first population.</span></p>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US>- <b style='mso-bidi-font-weight:normal'>split between two
populations</b> = Distribute the lineages of the admixed population among the
two parental populations according to the admixture rate.</span></p>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US>- <b style='mso-bidi-font-weight:normal'>coalesce and mutate
lineages within a population</b> = There are two possibilities here, depending
on whether the population is <i style='mso-bidi-font-style:normal'>terminal</i>
or not. We call <i style='mso-bidi-font-style:normal'>terminal</i> the
population including the most recent common ancestor of the whole genealogy. In
a terminal population, coalescences and mutations stop when the MRCA is reached
whereas in a non-terminal population, coalescence and mutations stop when the
upper (most ancient) limit is reached. In the latter case, coalescences can
stop before the upper limit is reached because there remains a single lineage,
but this single remaining lineage can still mutate. </span></p>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt;text-indent:
36.0pt'><span lang=EN-US>Two different coalescence algorithms are implemented:
a generation by generation simulation or a continuous time simulation. The
choice, automatically performed by the program, is based on an empirical
criterion which ensures that the continuous time algorithm is chosen whenever
it is faster than generation by generation while keeping the relative error on
the coalescence rate below 5% (see <span class=SpellE>Cornuet</span> et al.
2008 for a description of this criterion). In any case, a coalescent tree is
generated over all sampled genes. </span></p>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt;text-indent:
36.0pt'><span lang=EN-US>Then the mutational simulation process diverges
depending on the type of markers: for microsatellite or DNA sequence loci,
mutations are distributed over the branches according to a Poisson process
whereas for SNP loci, one mutation is applied to a single branch of the
coalescent tree, this branch being drawn at random with probability
proportional to its length. Eventually, starting from an ancestral allelic
state (established as explained below), all allelic states of the genealogy are
deduced forward in time according to the mutation process. For microsatellite
loci, the ancestral allelic state is taken at random in the stationary
distribution of the mutation model (not considering potential single nucleotide
indel mutations). For DNA sequence loci, the procedure is slightly more
complicated. First, the total number of mutations over the entire tree is
evaluated. Then according to the proportion of constant sites and the gamma
distribution of individual site mutation rates, the number and position of
mutated sites are generated. Finally, these mutated sites are given
&#8216;A&#8217;, &#8216;T&#8217;, &#8216;G&#8217; or &#8216;C&#8217; states
according to the selected mutation model. For SNP loci, the ancestral allelic state
is arbitrarily set to 0 and it becomes equal to 1 after mutation.</span></p>

<p class=DefinitionTerm style='text-indent:36.0pt'><span lang=EN-US
style='font-weight:normal'>Each category of loci has its own coalescence rate
deduced from male and female effective population sizes. In order to combine
different categories (e.g.&nbsp;autosomal and mitochondrial), we have to take
into account the relationships among the corresponding effective population
sizes. This can be achieved by linking the different effective population sizes
to the effective number of males ( </span><!--[if gte msEquation 12]><m:oMath><m:sSub><m:sSubPr><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
      m:val="roman"/><m:sty m:val="bi"/></m:rPr>N</m:r></span></i></m:e><m:sub><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
      m:val="roman"/><m:sty m:val="bi"/></m:rPr>M</m:r></span></i></m:sub></m:sSub></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shapetype
 id="_x0000_t75" coordsize="21600,21600" o:spt="75" o:preferrelative="t"
 path="m@4@5l@4@11@9@11@9@5xe" filled="f" stroked="f">
 <v:stroke joinstyle="miter"/>
 <v:formulas>
  <v:f eqn="if lineDrawn pixelLineWidth 0"/>
  <v:f eqn="sum @0 1 0"/>
  <v:f eqn="sum 0 0 @1"/>
  <v:f eqn="prod @2 1 2"/>
  <v:f eqn="prod @3 21600 pixelWidth"/>
  <v:f eqn="prod @3 21600 pixelHeight"/>
  <v:f eqn="sum @0 0 1"/>
  <v:f eqn="prod @6 1 2"/>
  <v:f eqn="prod @7 21600 pixelWidth"/>
  <v:f eqn="sum @8 21600 0"/>
  <v:f eqn="prod @7 21600 pixelHeight"/>
  <v:f eqn="sum @10 21600 0"/>
 </v:formulas>
 <v:path o:extrusionok="f" gradientshapeok="t" o:connecttype="rect"/>
 <o:lock v:ext="edit" aspectratio="t"/>
</v:shapetype><v:shape id="_x0000_i1025" type="#_x0000_t75" style='width:17.4pt;
 height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image001.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=23 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image002.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US style='font-weight:normal'><span
style='mso-spacerun:yes'>&nbsp;</span>) and females (</span><!--[if gte msEquation 12]><m:oMath><m:sSub><m:sSubPr><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
      m:val="roman"/><m:sty m:val="bi"/></m:rPr>N</m:r></span></i></m:e><m:sub><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
      m:val="roman"/><m:sty m:val="bi"/></m:rPr>F</m:r></span></i></m:sub></m:sSub></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:15pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image003.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=20 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image004.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US style='font-weight:normal'>) through the sum </span><!--[if gte msEquation 12]><m:oMath><m:sSub><m:sSubPr><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
      m:val="roman"/><m:sty m:val="bi"/></m:rPr>N</m:r></span></i></m:e><m:sub><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
      m:val="roman"/><m:sty m:val="bi"/></m:rPr>T</m:r></span></i></m:sub></m:sSub><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
    m:val="roman"/><m:sty m:val="bi"/></m:rPr>=</m:r></span></i><m:sSub><m:sSubPr><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
      m:val="roman"/><m:sty m:val="bi"/></m:rPr>N</m:r></span></i></m:e><m:sub><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
      m:val="roman"/><m:sty m:val="bi"/></m:rPr>F</m:r></span></i></m:sub></m:sSub><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
    m:val="roman"/><m:sty m:val="bi"/></m:rPr>+</m:r></span></i><m:sSub><m:sSubPr><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
      m:val="roman"/><m:sty m:val="bi"/></m:rPr>N</m:r></span></i></m:e><m:sub><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
      m:val="roman"/><m:sty m:val="bi"/></m:rPr>M</m:r></span></i></m:sub></m:sSub></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:77.4pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image005.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=103 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image006.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US style='font-weight:normal'><span
style='mso-spacerun:yes'>&nbsp;</span>and the ratio </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r><m:rPr><m:scr
    m:val="roman"/><m:sty m:val="bi"/></m:rPr>r</m:r><m:r><m:rPr><m:scr m:val="roman"/><m:sty
    m:val="bi"/></m:rPr>=</m:r></span></i><m:f><m:fPr><m:type m:val="lin"/><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:fPr><m:num><m:sSub><m:sSubPr><span
     style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
     mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
     style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
     "Cambria Math",serif'><m:r><m:rPr><m:scr m:val="roman"/><m:sty m:val="bi"/></m:rPr>N</m:r></span></i></m:e><m:sub><i
     style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
     "Cambria Math",serif'><m:r><m:rPr><m:scr m:val="roman"/><m:sty m:val="bi"/></m:rPr>M</m:r></span></i></m:sub></m:sSub></m:num><m:den><m:d><m:dPr><span
     style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
     mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:dPr><m:e><m:sSub><m:sSubPr><span
       style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
       mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
       style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
       "Cambria Math",serif'><m:r><m:rPr><m:scr m:val="roman"/><m:sty m:val="bi"/></m:rPr>N</m:r></span></i></m:e><m:sub><i
       style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
       "Cambria Math",serif'><m:r><m:rPr><m:scr m:val="roman"/><m:sty m:val="bi"/></m:rPr>F</m:r></span></i></m:sub></m:sSub><i
     style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
     "Cambria Math",serif'><m:r><m:rPr><m:scr m:val="roman"/><m:sty m:val="bi"/></m:rPr>+</m:r></span></i><m:sSub><m:sSubPr><span
       style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
       mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
       style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
       "Cambria Math",serif'><m:r><m:rPr><m:scr m:val="roman"/><m:sty m:val="bi"/></m:rPr>N</m:r></span></i></m:e><m:sub><i
       style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
       "Cambria Math",serif'><m:r><m:rPr><m:scr m:val="roman"/><m:sty m:val="bi"/></m:rPr>M</m:r></span></i></m:sub></m:sSub></m:e></m:d></m:den></m:f></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:102.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image007.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=137 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image008.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US style='font-weight:normal'>. We use the following formulae for the
probability of coalescence (<i style='mso-bidi-font-style:normal'>p</i>) of two
lineages within this population:</span></p>

<p class=Definition><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US style='font-size:10.0pt'>Autosomal
diploid loci:<o:p></o:p></span></p>

<p class=Definition><!--[if gte msEquation 12]><m:oMathPara><m:oMath><i
  style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>p</m:r><m:r>=</m:r></span></i><m:f><m:fPr><span
    style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
    mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:fPr><m:num><i
    style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
    "Cambria Math",serif'><m:r>1</m:r></span></i></m:num><m:den><i
    style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
    "Cambria Math",serif'><m:r>8</m:r><m:r>r</m:r></span></i><m:d><m:dPr><span
      style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
      mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:dPr><m:e><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>1-</m:r><m:r>r</m:r></span></i></m:e></m:d><m:sSub><m:sSubPr><span
      style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
      mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>N</m:r></span></i></m:e><m:sub><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>T</m:r></span></i></m:sub></m:sSub></m:den></m:f></m:oMath></m:oMathPara><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
mso-ansi-language:FR;mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:86.4pt;height:29.4pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image009.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=115 height=39
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image010.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US style='font-size:10.0pt'><o:p></o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US style='font-size:10.0pt'>Autosomal
haploid loci:<o:p></o:p></span></p>

<p class=Definition><!--[if gte msEquation 12]><m:oMathPara><m:oMath><i
  style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>p</m:r><m:r>=</m:r></span></i><m:f><m:fPr><span
    style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
    mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:fPr><m:num><i
    style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
    "Cambria Math",serif'><m:r>1</m:r></span></i></m:num><m:den><i
    style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
    "Cambria Math",serif'><m:r>4</m:r><m:r>r</m:r></span></i><m:d><m:dPr><span
      style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
      mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:dPr><m:e><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>1-</m:r><m:r>r</m:r></span></i></m:e></m:d><m:sSub><m:sSubPr><span
      style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
      mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>N</m:r></span></i></m:e><m:sub><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>T</m:r></span></i></m:sub></m:sSub></m:den></m:f></m:oMath></m:oMathPara><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
mso-ansi-language:FR;mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:86.4pt;height:29.4pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image011.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=115 height=39
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image012.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US style='font-size:10.0pt'><o:p></o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US style='font-size:10.0pt'>X-linked loci
/ haplo-diploid loci:<o:p></o:p></span></p>

<p class=Definition><!--[if gte msEquation 12]><m:oMathPara><m:oMath><i
  style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>p</m:r><m:r>=</m:r></span></i><m:f><m:fPr><span
    style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
    mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:fPr><m:num><i
    style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
    "Cambria Math",serif'><m:r>1+</m:r><m:r>r</m:r></span></i></m:num><m:den><i
    style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
    "Cambria Math",serif'><m:r>9</m:r><m:r>r</m:r></span></i><m:d><m:dPr><span
      style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
      mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:dPr><m:e><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>1-</m:r><m:r>r</m:r></span></i></m:e></m:d><m:sSub><m:sSubPr><span
      style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
      mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>N</m:r></span></i></m:e><m:sub><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>T</m:r></span></i></m:sub></m:sSub></m:den></m:f></m:oMath></m:oMathPara><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
mso-ansi-language:FR;mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:86.4pt;height:29.4pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image013.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=115 height=39
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image014.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US style='font-size:10.0pt'><o:p></o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US style='font-size:10.0pt'>Y-linked
loci:<o:p></o:p></span></p>

<p class=Definition><!--[if gte msEquation 12]><m:oMathPara><m:oMath><i
  style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>p</m:r><m:r>=</m:r></span></i><m:f><m:fPr><span
    style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
    mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:fPr><m:num><i
    style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
    "Cambria Math",serif'><m:r>1</m:r></span></i></m:num><m:den><i
    style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
    "Cambria Math",serif'><m:r>r</m:r></span></i><m:sSub><m:sSubPr><span
      style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
      mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>N</m:r></span></i></m:e><m:sub><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>T</m:r></span></i></m:sub></m:sSub></m:den></m:f></m:oMath></m:oMathPara><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
mso-ansi-language:FR;mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:42.6pt;height:29.4pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image015.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=57 height=39
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image016.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US style='font-size:10.0pt'><o:p></o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US style='font-size:10.0pt'>Mitochondrial
loci:<o:p></o:p></span></p>

<p class=Definition><!--[if gte msEquation 12]><m:oMathPara><m:oMath><i
  style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>p</m:r><m:r>=</m:r></span></i><m:f><m:fPr><span
    style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
    mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:fPr><m:num><i
    style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
    "Cambria Math",serif'><m:r>1</m:r></span></i></m:num><m:den><m:d><m:dPr><span
      style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
      mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:dPr><m:e><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>1-</m:r><m:r>r</m:r></span></i></m:e></m:d><m:sSub><m:sSubPr><span
      style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
      mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>N</m:r></span></i></m:e><m:sub><i
      style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
      "Cambria Math",serif'><m:r>T</m:r></span></i></m:sub></m:sSub></m:den></m:f></m:oMath></m:oMathPara><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
mso-ansi-language:FR;mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:73.8pt;height:29.4pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image017.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=98 height=39
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image018.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US style='font-size:10.0pt'><o:p></o:p></span></p>

<p class=Definition><span lang=EN-US>Users have to provide a (total) effective
size </span><!--[if gte msEquation 12]><m:oMath><m:sSub><m:sSubPr><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>N</m:r></span></i></m:e><m:sub><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>T</m:r></span></i></m:sub></m:sSub></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:14.4pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image019.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=19 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image020.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>(on which inferences
will be made) and a sex-ratio </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>r</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image021.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=8 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image022.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>. If no sex ratio is provided, the default value of </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>r</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image021.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=8 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image022.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>is taken as 0.5.</span></p>

<h2><a name="_Toc62127935"></a><a name=historical-model-parameterization><span
style='mso-bookmark:_Toc62127935'><span lang=EN-US>2.3 Historical model
parameterization</span></span></a></h2>

<p class=FirstParagraph><span lang=EN-US>The evolutionary scenario, which is
characterized by the historical model, can be described in a dedicated panel of
the interface of the program as a succession in time of &quot;events&quot; and
&quot;inter event periods&quot;. In the current version of the program, we
consider 4 categories of events: population divergence, discrete change of
effective population size, admixture and sampling (the last one <span
class=GramE>allow</span> considering samples taken at different times). Between
two successive events affecting a population, we assume that populations evolve
independently (e.g.&nbsp;without migration) and with a fixed effective size.
The usual parameters of the historical model are the times of occurrence of the
various events (counted in number of generations), the effective sizes of
populations and the admixture rates. <b style='mso-bidi-font-weight:normal'>When
writing the scenario, events have to be coded sequentially backward in time</b>
(see section 2.3.2 for examples). Although this choice may not be natural at
first sight, it is coherent with coalescence theory on which are based all data
simulations in the program. For that reason, the keywords for a divergence or
an admixture event are </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>merge</span></span><span
lang=EN-US> and </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>split</span></span><span
lang=EN-US>, respectively. Two keywords </span><span class=SpellE><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>varNe</span></span></span><span lang=EN-US> and </span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>sample</span></span><span lang=EN-US> correspond to a discrete change
in effective population size and a gene sampling within a given population,
respectively. A scenario takes the form of a succession of lines (one line per
event), each line starting with the time of the event, then the nature of the
event, and ending with several other data depending on the nature of the event.
Following is the syntax used for each category of event:</span></p>

<p class=DefinitionTerm><span style='mso-ansi-language:FR'>Population <span
class=SpellE>sample</span><o:p></o:p></span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>time</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:33.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image023.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=45 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image024.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span></span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>sample</span></span><span lang=EN-US> </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:28.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image025.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=38 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image026.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span></span><span lang=EN-US
style='mso-fareast-font-family:Cambria;mso-fareast-theme-font:minor-fareast'><o:p></o:p></span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US style='mso-fareast-font-family:Cambria;mso-fareast-theme-font:minor-fareast'>Where
</span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:normal'><span
 lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>time</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:33.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image023.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=45 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image024.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>is the time (always
counted in number of generations) at which the sample was collected and </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:28.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image025.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=38 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image026.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>is the population number
from which is taken the sample. It is worth stressing here that <b
style='mso-bidi-font-weight:normal'>samples are considered in the same order as
they appear in the data file</b>. The number of lines will thus be exactly
equal to the number of samples in the datafile.<br style='mso-special-character:
line-break'>
<![if !supportLineBreakNewLine]><br style='mso-special-character:line-break'>
<![endif]><b style='mso-bidi-font-weight:normal'><o:p></o:p></b></span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><b
style='mso-bidi-font-weight:normal'><span lang=EN-US>Population size variation<o:p></o:p></span></b></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>time</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:33.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image023.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=45 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image024.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span></span><span
class=SpellE><span class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;
mso-bidi-font-size:12.0pt'>varNe</span></span></span><span lang=EN-US> </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:28.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image025.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=38 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image026.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span></span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>Ne</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:23.4pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image027.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=31 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image028.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><br>
From time </span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>time</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:33.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image023.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=45 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image024.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>, looking backward in time, population </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:28.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image025.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=38 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image026.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>will have an effective
size </span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>Ne</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:23.4pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image027.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=31 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image028.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>.</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US>Population divergence</span></p>

<p class=Definition><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>time</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:33.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image023.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=45 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image024.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span></span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>merge</span></span><span lang=EN-US> </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>1</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image029.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image030.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span></span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>0</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image031.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image032.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><br>
At time </span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>time</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:33.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image023.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=45 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image024.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>, looking backward in time, population </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>0</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image031.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image032.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>&quot;merges&quot; with
population </span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>1</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image029.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image030.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>. Hereafter, only </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>1</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image029.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image030.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>&quot;remains&quot;.</span></p>

<p class=DefinitionTerm><span lang=EN-US>Population admixture</span></p>

<p class=Definition><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>time</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:33.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image023.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=45 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image024.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span></span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>split</span></span><span lang=EN-US> </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>0</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image031.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image032.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span></span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>1</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image029.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image030.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span></span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>2</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image033.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image034.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span></span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>rate</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:31.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image035.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=42 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image036.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><br>
At time </span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>time</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:33.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image023.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=45 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image024.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>, looking backward in time, population </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>0</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image031.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image032.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>&quot;splits&quot;
between populations </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>1</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image029.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image030.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>and </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>2</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image033.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image034.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>. A gene lineage from population </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>0</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image031.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image032.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>joins population </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>1</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image029.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image030.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>(respectively </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>2</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image033.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image034.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>) with probability </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>rate</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:31.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image035.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=42 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image036.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>(respectively 1-</span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>rate</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:31.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image035.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=42 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image036.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>). Hereafter, only </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>1</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image029.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image030.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>and </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#10216;</m:r><m:r>pop</m:r><m:r>2</m:r><m:r>&#10217;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:34.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image033.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=46 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image034.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>&quot;remain&quot;.</span></p>

<p class=FirstParagraph><span lang=EN-US>Note that one needs to write a first
line giving the <b style='mso-bidi-font-weight:normal'>effective sizes of the
sampled populations before the first event described</b>, looking backward in
time. Expressions between arrows, other than population numbers, can be either
a numeric value (e.g.&nbsp;25) or a character string (e.g.&nbsp;</span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>t0</span></span><span lang=EN-US>). In the latter case, it is
considered as a parameter of the model. The program offers the possibility to
add or remove scenarios, by just clicking on the corresponding buttons. The
usual shortcuts (e.g. CTRL+C, CTRL+V and CTRL+X) can be used to edit the
different scenarios. Some or all parameters can be in common among scenarios.</span></p>

<h3 style='margin-top:0cm;margin-right:0cm;margin-bottom:10.0pt;margin-left:
0cm'><a name="_Toc62127936"><span lang=EN-US>2.3.1 Key notes</span></a></h3>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l23 level1 lfo3'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>1.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>There are two ways of giving a
fixed value to effective population sizes, times and admixture rates. Either
the fixed value appears as a numeric value in the scenario windows or it is
given as a string value like any parameter. In the latter case, one gives this
parameter a fixed value by choosing a Uniform distribution and setting the
minimum and maximum to that value in the prior setting of the corresponding
interface panel.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l23 level1 lfo3'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>2.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>All expressions must be
separated by at least one space.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l23 level1 lfo3'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>3.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>All expressions relative to
parameters can include sums or differences of parameters of the same type (e.g.
divergence times or effective population size but not a mixture of both). For
instance, it is possible to write:<br>
</span><span class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;
mso-bidi-font-size:12.0pt'>t0 merge 2 3</span></span><span lang=EN-US><br>
</span><span class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;
mso-bidi-font-size:12.0pt'>t0+t1 merge 1 2</span></span><span lang=EN-US><br>
This means that </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>t1</span></span><span
lang=EN-US> is the time elapsed between the two merge events. Note that one
cannot mix a parameter and a numeric value (e.g.&nbsp;</span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>t1+150</span></span><span lang=EN-US> will result in an error). This
can be done by writing </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>t1+t2</span></span><span
lang=EN-US> and fixing </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>t2</span></span><span
lang=EN-US> by choosing a uniform distribution with lower and upper bounds both
equal to 150 in the interface.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l23 level1 lfo3'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>4.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>Time is always given in generations</span></b><span lang=EN-US>.
Since we look backward, time increases towards past.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l23 level1 lfo3'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>5.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Negative times are allowed
(e.g.&nbsp;the example given in section 2.3.2), but not recommended.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l23 level1 lfo3'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>6.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Population numbers must be
consecutive natural integers starting at 1. The number of <span class=GramE>population</span>
can exceed the number of samples and vice versa: in other words, <i
style='mso-bidi-font-style:normal'>unsampled populations</i> can be considered
in the scenario on one hand, and the same population can be sampled more than
once on the other hand.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l23 level1 lfo3'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>7.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>Multi-furcating population trees</span></b><span lang=EN-US> can be
considered, by writing <b style='mso-bidi-font-weight:normal'>several
divergence events occurring at the same time.</b> However, one has to be
careful to the order of the </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>merge</span></span><span
lang=EN-US> events. For instance, the following piece of scenario will fail:<br>
</span><span class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;
mso-bidi-font-size:12.0pt'>100 merge 1 2</span></span><span lang=EN-US><br>
</span><span class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;
mso-bidi-font-size:12.0pt'>100 merge 2 3</span></span><span lang=EN-US><br>
This is because, after the first line, population 2, which has merged with
population 1, does not &quot;exist&quot; anymore (the remaining population is
population 1). So, it cannot receive lineages of population 3 as it should as a
result of the second line. The correct ways are either to put line 2 before
line 1, or to change line 2 to:<br>
</span><span class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;
mso-bidi-font-size:12.0pt'>100 merge 1 3</span></span><span lang=EN-US>.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l23 level1 lfo3'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>8.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Since times of events can be
parameters, the order of events can change according to the values taken by the
time parameters. In any case, before simulating a dataset, the program sorts
out events by increasing times. Note that sorting out events by increasing times
can only be done when all time values are known, i.e.&nbsp;when simulating
datasets. When checking scenarios in the interface, all time values are not yet
defined, so that when visualizing a scenario, events are represented in the
same order as they appear in the window used to define the scenario. If two or
more events occur at the same time, the order is that of the scenario as it is
written by the user.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l23 level1 lfo3'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>9.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Most scenarios begin with
sampling events. We then need to know the effective size of the populations to
perform the simulation of coalescences until the next event concerning each
population. We decided to provide the effective size (first line) and the
sampling description (following lines) on distinct lines.</span></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;mso-pagination:
widow-orphan lines-together;page-break-after:avoid;mso-outline-level:3'><a
name="_Toc62127937"><b><span lang=EN-US style='font-family:"Calibri",sans-serif;
mso-ascii-theme-font:major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'>2.3.2 Examples</span></b></a><b><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'><o:p></o:p></span></b></p>

<p class=FirstParagraph><span lang=EN-US>Below are some usual scenarios with
increasing complexity. Each scenario is coded (as in the corresponding
interface panel of the program) on the left side and a graphic representation
(given by DIYABC Random Forest) is printed on the right side</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l31 level1 lfo4'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>1.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>One population from which
several samples have been taken at various generations: 0, 3 and 10. Generation
0 could correspond for instance to the most recent sampling date. The only
unknown historical-demographical parameter of the scenario is the (constant)
effective population size which is defined in the first line at sampling time.<br
style='mso-special-character:line-break'>
<![if !supportLineBreakNewLine]><br style='mso-special-character:line-break'>
<![endif]></span></p>

<table class=Table border=0 cellspacing=0 cellpadding=0 width=476
 style='width:356.85pt;border-collapse:collapse;mso-yfti-tbllook:1984;
 mso-padding-alt:0cm 5.4pt 0cm 5.4pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;mso-yfti-lastrow:yes'>
  <td width=150 valign=top style='width:112.8pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='margin-left:24.0pt;text-align:center;
  text-indent:-24.0pt;mso-list:l8 level1 lfo1'><![if !supportLists]><span
  lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;
  mso-bidi-font-family:Symbol'><span style='mso-list:Ignore'><span
  style='mso-spacerun:yes'>&nbsp;</span><span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='mso-ansi-language:FR;mso-fareast-language:
  FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Picture" o:spid="_x0000_i1075"
   type="#_x0000_t75" alt="image" style='width:78pt;height:58.8pt;visibility:visible;
   mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image037.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=104 height=78
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image038.gif" alt=image
  v:shapes="Picture"><![endif]></span></p>
  </td>
  <td width=325 valign=top style='width:244.0pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='margin-left:24.0pt;text-align:center;
  text-indent:-24.0pt;mso-list:l8 level1 lfo1'><![if !supportLists]><span
  lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;
  mso-bidi-font-family:Symbol'><span style='mso-list:Ignore'><span
  style='mso-spacerun:yes'>&nbsp;</span><span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='mso-ansi-language:FR;mso-fareast-language:
  FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image1" o:spid="_x0000_i1074"
   type="#_x0000_t75" alt="image" style='width:189.6pt;height:155.4pt;
   visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image039.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=253 height=207
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image040.gif" alt=image
  v:shapes="Image1"><![endif]></span></p>
  </td>
 </tr>
</table>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l31 level1 lfo4'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>2.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Two populations of size </span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>N1</span></span><span lang=EN-US> and </span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>N2</span></span><span
lang=EN-US> (defined in the first line at sampling time) have diverged </span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>t</span></span><span lang=EN-US> generations in the past from an
ancestral population of size </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>N1+N2</span></span><span
lang=EN-US>.<br style='mso-special-character:line-break'>
<![if !supportLineBreakNewLine]><br style='mso-special-character:line-break'>
<![endif]></span></p>

<table class=Table border=0 cellspacing=0 cellpadding=0 width=491
 style='width:368.4pt;border-collapse:collapse;mso-yfti-tbllook:1984;
 mso-padding-alt:0cm 5.4pt 0cm 5.4pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;mso-yfti-lastrow:yes'>
  <td width=186 valign=top style='width:139.8pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='margin-left:24.0pt;text-align:center;
  text-indent:-24.0pt;mso-list:l8 level1 lfo1'><![if !supportLists]><span
  lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;
  mso-bidi-font-family:Symbol'><span style='mso-list:Ignore'><span
  style='mso-spacerun:yes'>&nbsp;</span><span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='mso-ansi-language:FR;mso-fareast-language:
  FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image2" o:spid="_x0000_i1073"
   type="#_x0000_t75" alt="image" style='width:105pt;height:61.8pt;
   visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image041.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=140 height=82
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image042.gif" alt=image
  v:shapes="Image2"><![endif]></span></p>
  </td>
  <td width=305 valign=top style='width:228.55pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='margin-left:24.0pt;text-align:center;
  text-indent:-24.0pt;mso-list:l8 level1 lfo1'><![if !supportLists]><span
  lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;
  mso-bidi-font-family:Symbol'><span style='mso-list:Ignore'><span
  style='mso-spacerun:yes'>&nbsp;</span><span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='mso-ansi-language:FR;mso-fareast-language:
  FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image3" o:spid="_x0000_i1072"
   type="#_x0000_t75" alt="image" style='width:193.8pt;height:153.6pt;
   visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image043.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=258 height=205
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image044.gif" alt=image
  v:shapes="Image3"><![endif]></span></p>
  </td>
 </tr>
</table>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l31 level1 lfo4'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>3.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Two parental populations (1 and
1) with constant effective population sizes </span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>N1</span></span><span
lang=EN-US> and </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>N2</span></span><span
lang=EN-US> have diverged at time </span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>td</span></span><span
lang=EN-US> from an ancestral population of size </span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>NA</span></span><span lang=EN-US>. At time </span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>ta</span></span><span lang=EN-US>, there has been an <u>admixture event</u>
between the two populations giving birth to an admixed population (3) with
effective size </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>N3</span></span><span
lang=EN-US> and with an admixture rate </span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>ra</span></span><span
lang=EN-US> relative to population 1.<br style='mso-special-character:line-break'>
<![if !supportLineBreakNewLine]><br style='mso-special-character:line-break'>
<![endif]></span></p>

<table class=Table border=0 cellspacing=0 cellpadding=0 width=561
 style='width:421.1pt;border-collapse:collapse;mso-yfti-tbllook:1984;
 mso-padding-alt:0cm 5.4pt 0cm 5.4pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;mso-yfti-lastrow:yes'>
  <td width=192 valign=top style='width:144.3pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='margin-left:24.0pt;text-align:center;
  text-indent:-24.0pt;mso-list:l8 level1 lfo1'><![if !supportLists]><span
  lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;
  mso-bidi-font-family:Symbol'><span style='mso-list:Ignore'><span
  style='mso-spacerun:yes'>&nbsp;</span><span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='mso-ansi-language:FR;mso-fareast-language:
  FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image4" o:spid="_x0000_i1071"
   type="#_x0000_t75" alt="image" style='width:109.2pt;height:88.2pt;
   visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image045.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=146 height=118
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image046.gif" alt=image
  v:shapes="Image4"><![endif]></span></p>
  </td>
  <td width=369 valign=top style='width:276.75pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='margin-left:24.0pt;text-align:center;
  text-indent:-24.0pt;mso-list:l8 level1 lfo1'><![if !supportLists]><span
  lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;
  mso-bidi-font-family:Symbol'><span style='mso-list:Ignore'><span
  style='mso-spacerun:yes'>&nbsp;</span><span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='mso-ansi-language:FR;mso-fareast-language:
  FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image5" o:spid="_x0000_i1070"
   type="#_x0000_t75" alt="image" style='width:242.4pt;height:187.2pt;
   visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image047.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=323 height=250
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image048.gif" alt=image
  v:shapes="Image5"><![endif]></span></p>
  </td>
 </tr>
</table>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l31 level1 lfo4'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>4.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The next scenario includes <u>four
population samples and two admixture events</u>. All populations have identical
effective sizes (</span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>Ne</span></span><span
lang=EN-US>).<br style='mso-special-character:line-break'>
<![if !supportLineBreakNewLine]><br style='mso-special-character:line-break'>
<![endif]></span></p>

<table class=Table border=0 cellspacing=0 cellpadding=0 width=561
 style='width:421.1pt;border-collapse:collapse;mso-yfti-tbllook:1984;
 mso-padding-alt:0cm 5.4pt 0cm 5.4pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;mso-yfti-lastrow:yes'>
  <td width=192 valign=top style='width:144.3pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='margin-left:24.0pt;text-align:center;
  text-indent:-24.0pt;mso-list:l8 level1 lfo1'><![if !supportLists]><span
  lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;
  mso-bidi-font-family:Symbol'><span style='mso-list:Ignore'><span
  style='mso-spacerun:yes'>&nbsp;</span><span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='mso-ansi-language:FR;mso-fareast-language:
  FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image6" o:spid="_x0000_i1069"
   type="#_x0000_t75" alt="image" style='width:109.2pt;height:88.2pt;
   visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image049.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=146 height=118
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image050.gif" alt=image
  v:shapes="Image6"><![endif]></span></p>
  </td>
  <td width=369 valign=top style='width:276.75pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='margin-left:24.0pt;text-align:center;
  text-indent:-24.0pt;mso-list:l8 level1 lfo1'><![if !supportLists]><span
  lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;
  mso-bidi-font-family:Symbol'><span style='mso-list:Ignore'><span
  style='mso-spacerun:yes'>&nbsp;</span><span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='mso-ansi-language:FR;mso-fareast-language:
  FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image7" o:spid="_x0000_i1068"
   type="#_x0000_t75" alt="image" style='width:242.4pt;height:196.8pt;
   visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image051.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=323 height=262
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image052.gif" alt=image
  v:shapes="Image7"><![endif]></span></p>
  </td>
 </tr>
</table>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l8 level1 lfo1'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'><span
style='mso-spacerun:yes'>&nbsp;</span><span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Note that although there are
only four samples, the scenario includes a fifth <u>unsampled population</u>.
This unsampled population which diverged from population 1 at time </span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>t3</span></span><span lang=EN-US> was a parent in the admixture event
occurring at time </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>t2</span></span><span
lang=EN-US>. Note also that the first line must include the effective sizes of
the <i style='mso-bidi-font-style:normal'>five</i> populations.</span></p>

<p class=MsoListParagraphCxSpFirst style='margin-left:24.0pt;mso-add-space:
auto;text-indent:-24.0pt;mso-list:l31 level1 lfo4'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>5.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Example of DIYABC-RF code for <u>unidirectional
admixture/introgression</u> where a fraction r2 of genes from population 'Pop
2' (with an effective population size N2) <span class=SpellE>introgresses</span>
into population 'Pop 1' (with a population size N1) at time t1. The code also
describes the merge of the two populations 'Pop 1' and 'Pop 2' into a common
ancestor (with a population size Na) at time t2. It is worth noting that: (<span
class=SpellE>i</span>) in the observed datafile, there are only two sampled
populations (Pop 1 and Pop 2) and the user will need to define two (additional)
<u>unsampled</u> populations in the first line of code (Pop 3 and Pop 4); (ii) we
advise in this case to draw the fraction of genes transferred during the unidirectional
admixture (r2) into a uniform distribution with min and max bounds equal to <span
class=SpellE>eg.</span> 0.01 (very weak introgression from 'Pop 2' into 'Pop
1') and 0.5 (strong introgression from 'Pop 2' into 'Pop 1'), respectively.</span></p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:24.0pt;mso-add-space:
auto'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:24.0pt;mso-add-space:
auto'><!--[if gte vml 1]><v:shape id="Image_x0020_24" o:spid="_x0000_s1026"
 type="#_x0000_t75" style='position:absolute;left:0;text-align:left;
 margin-left:263.85pt;margin-top:-21.15pt;width:225.6pt;height:204.25pt;
 z-index:-251658240;visibility:visible;mso-wrap-style:square;
 mso-width-percent:0;mso-height-percent:0;mso-wrap-distance-left:9pt;
 mso-wrap-distance-top:0;mso-wrap-distance-right:9pt;
 mso-wrap-distance-bottom:0;mso-position-horizontal:absolute;
 mso-position-horizontal-relative:text;mso-position-vertical:absolute;
 mso-position-vertical-relative:text;mso-width-percent:0;mso-height-percent:0;
 mso-width-relative:margin;mso-height-relative:margin'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image053.jpg"
  o:title=""/>
 <w:wrap type="through"/>
</v:shape><![endif]--><![if !vml]><img width=301 height=272
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image054.jpg" align=left
hspace=12 v:shapes="Image_x0020_24"><![endif]><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:
FR'>N1 N2 <span class=SpellE>N2</span> <span class=SpellE>N2</span><span
style='mso-tab-count:1'>&nbsp;&nbsp; </span><span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span></b><i
style='mso-bidi-font-style:normal'><span style='font-size:10.0pt;font-family:
"Courier New";mso-ansi-language:FR'># comment 1</span></i><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:
FR'><o:p></o:p></span></b></p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:24.0pt;mso-add-space:
auto'><b style='mso-bidi-font-weight:normal'><span style='font-size:10.0pt;
font-family:"Courier New";mso-ansi-language:FR'>0 <span class=SpellE>sample</span>
1<o:p></o:p></span></b></p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:24.0pt;mso-add-space:
auto'><b style='mso-bidi-font-weight:normal'><span style='font-size:10.0pt;
font-family:"Courier New";mso-ansi-language:FR'>0 <span class=SpellE>sample</span>
2<o:p></o:p></span></b></p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:24.0pt;mso-add-space:
auto'><span class=GramE><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:FR'>t</span></b></span><b
style='mso-bidi-font-weight:normal'><span style='font-size:10.0pt;font-family:
"Courier New";mso-ansi-language:FR'>1 split 2 3 4 r2<span
style='mso-spacerun:yes'>&nbsp;&nbsp; </span></span></b><i style='mso-bidi-font-style:
normal'><span style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:
FR'># comment 2<b style='mso-bidi-font-weight:normal'><span style='mso-tab-count:
1'> </span></b></span></i><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:FR'><span
style='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><o:p></o:p></span></b></p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:24.0pt;mso-add-space:
auto'><span class=GramE><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:FR'>t</span></b></span><b
style='mso-bidi-font-weight:normal'><span style='font-size:10.0pt;font-family:
"Courier New";mso-ansi-language:FR'>1 merge 1 3<span style='mso-tab-count:1'>&nbsp; </span><span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp; </span><span
style='mso-spacerun:yes'>&nbsp;</span></span></b><i style='mso-bidi-font-style:
normal'><span style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:
FR'># comment 3</span></i><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:FR'><o:p></o:p></span></b></p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:24.0pt;mso-add-space:
auto'><span class=GramE><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:FR'>t</span></b></span><b
style='mso-bidi-font-weight:normal'><span style='font-size:10.0pt;font-family:
"Courier New";mso-ansi-language:FR'>2 merge 1 4<span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><span
style='mso-spacerun:yes'>&nbsp;</span></span></b><i style='mso-bidi-font-style:
normal'><span style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:
FR'># comment 4</span></i><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:FR'><o:p></o:p></span></b></p>

<p class=MsoListParagraphCxSpLast style='margin-left:24.0pt;mso-add-space:auto'><span
class=GramE><b style='mso-bidi-font-weight:normal'><span style='font-size:10.0pt;
font-family:"Courier New";mso-ansi-language:FR'>t</span></b></span><b
style='mso-bidi-font-weight:normal'><span style='font-size:10.0pt;font-family:
"Courier New";mso-ansi-language:FR'>2 VarNe 1 Na<span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp; </span><span
style='mso-spacerun:yes'>&nbsp;&nbsp;</span><span
style='mso-spacerun:yes'>&nbsp;</span></span></b><i style='mso-bidi-font-style:
normal'><span style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:
FR'># comment 5</span></i><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:"Courier New";mso-ansi-language:FR'><o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt'><span style='font-size:9.0pt;
mso-ansi-language:FR'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt'><span style='font-size:9.0pt;
mso-ansi-language:FR'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt'><span style='font-size:9.0pt;
mso-ansi-language:FR'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt'><span style='font-size:9.0pt;
mso-ansi-language:FR'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt'><span style='font-size:9.0pt;
mso-ansi-language:FR'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt'><span style='font-size:9.0pt;
mso-ansi-language:FR'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt'><span style='font-size:9.0pt;
mso-ansi-language:FR'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt'><span style='font-size:9.0pt;
mso-ansi-language:FR'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt'><span style='font-size:9.0pt;
mso-ansi-language:FR'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt'><i style='mso-bidi-font-style:normal'><span
lang=EN-US style='font-size:9.0pt'># comment 1: both unsampled populations 'Pop
3' and 'Pop 4' correspond to 'Pop 2', hence their population sizes N2.</span><span
lang=EN-US><br>
</span></i><i style='mso-bidi-font-style:normal'><span lang=EN-US
style='font-size:9.0pt'># comment 2: at time t1 a fraction r2 of genes from
'Pop 2' goes into the unsampled population 'Pop 3' and a fraction 1-r2 goes
into the unsampled population 'Pop 4' ('Pop 2' disappears)<o:p></o:p></span></i></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt'><i style='mso-bidi-font-style:normal'><span
lang=EN-US style='font-size:9.0pt'># comment 3: we immediately merge (cf. still
time t1) the unsampled population 'Pop 3' (with the fraction r2 of genes from
'Pop 2') into the population 'Pop 1' - hence the unidirectional admixture of r2
genes from 'Pop 2' into 'Pop 1' ('Pop 3' disappears)<o:p></o:p></span></i></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt'><i style='mso-bidi-font-style:normal'><span
lang=EN-US style='font-size:9.0pt'># comment 4: 'Pop 4' - which now corresponds
to the ex-'Pop 2' with the fraction 1-r2 of genes - can be merge into 'Pop 1'<o:p></o:p></span></i></p>

<p class=MsoListParagraph style='margin-top:0cm;margin-right:0cm;margin-bottom:
0cm;margin-left:24.1pt;margin-bottom:.0001pt;mso-add-space:auto'><i
style='mso-bidi-font-style:normal'><span lang=EN-US style='font-size:9.0pt'>#
comment 5: the population ancestral to 'Pop 1' and 'Pop 2' now has its own
population size Na<o:p></o:p></span></i></p>

<p class=MsoNormal style='margin-left:24.0pt'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l31 level1 lfo4'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>6.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The following three scenarios
correspond to a classic invasion history from an ancestral population
(population 1<u>). In scenario 1, population 3 is derived from population 2,
itself derived from population 1. In scenario 2, population 2 derived from
population 3, itself derived from population 1. In scenario 3, both populations
2 and 3 derived independently from population 1.</u> Note that when a new
population is created from its ancestral population, there is an initial size
reduction (noted here </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>N2b</span></span><span
lang=EN-US> for population 2 and </span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>N3b</span></span><span
lang=EN-US> for population 3) for a given number of generation (here </span><span
class=SpellE><span lang=EN-US style='font-family:Consolas;mso-bidi-font-family:
Consolas'>db</span></span><span lang=EN-US> for both populations) mimicking a
demographic bottleneck since the invasive population generally starts with a
few immigrants. For instance, a low number of <span class=SpellE>Nxb</span>
individuals for <span class=SpellE>db</span> generations could corresponds to <span
class=SpellE>Nxb</span> values ranging between 5 to 100 individuals and <span
class=SpellE>db</span> values ranging from 1 to 10 generations (e.g. <span
class=SpellE>Fraimout</span> et al. 2017). If <span class=SpellE>db</span> = 0
then no bottleneck occurs which can be also a coding choice.</span></p>

<p class=FirstParagraph><span lang=EN-US><span style='mso-spacerun:yes'>&nbsp;
</span>Scenario 1</span></p>

<table class=Table border=0 cellspacing=0 cellpadding=0 width=601
 style='width:450.6pt;border-collapse:collapse;mso-yfti-tbllook:1984;
 mso-padding-alt:0cm 5.4pt 0cm 5.4pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;mso-yfti-lastrow:yes'>
  <td width=166 valign=top style='width:124.8pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='text-align:center'><span
  style='mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape
   id="Image8" o:spid="_x0000_i1067" type="#_x0000_t75" alt="image" style='width:113.4pt;
   height:121.2pt;visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image055.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=151 height=162
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image056.gif" alt=image
  v:shapes="Image8"><![endif]></span></p>
  </td>
  <td width=434 valign=top style='width:325.75pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='text-align:center'><span
  style='mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape
   id="Image9" o:spid="_x0000_i1066" type="#_x0000_t75" alt="image" style='width:238.2pt;
   height:223.2pt;visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image057.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=318 height=298
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image058.jpg" alt=image
  v:shapes="Image9"><![endif]></span></p>
  </td>
 </tr>
</table>

<p class=MsoBodyText><span lang=EN-US><span style='mso-spacerun:yes'>&nbsp;
</span>Scenario 2</span></p>

<table class=Table border=0 cellspacing=0 cellpadding=0 width=600
 style='width:449.85pt;border-collapse:collapse;mso-yfti-tbllook:1984;
 mso-padding-alt:0cm 5.4pt 0cm 5.4pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;mso-yfti-lastrow:yes'>
  <td width=171 valign=top style='width:128.35pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='text-align:center'><span
  style='mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape
   id="Image10" o:spid="_x0000_i1065" type="#_x0000_t75" alt="image" style='width:117.6pt;
   height:111pt;visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image059.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=157 height=148
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image060.gif" alt=image
  v:shapes="Image10"><![endif]></span></p>
  </td>
  <td width=429 valign=top style='width:321.45pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='text-align:center'><span
  style='mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape
   id="Image11" o:spid="_x0000_i1064" type="#_x0000_t75" alt="image" style='width:234.6pt;
   height:220.2pt;visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image061.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=313 height=294
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image062.jpg" alt=image
  v:shapes="Image11"><![endif]></span></p>
  </td>
 </tr>
</table>

<p class=MsoBodyText><span lang=EN-US><span
style='mso-spacerun:yes'>&nbsp;</span></span></p>

<p class=MsoBodyText><span lang=EN-US><span
style='mso-spacerun:yes'>&nbsp;</span>Scenario 3</span></p>

<table class=Table border=0 cellspacing=0 cellpadding=0 width=583
 style='width:437.4pt;border-collapse:collapse;mso-yfti-tbllook:1984;
 mso-padding-alt:0cm 5.4pt 0cm 5.4pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;mso-yfti-lastrow:yes'>
  <td width=161 valign=top style='width:120.55pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='text-align:center'><span
  style='mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape
   id="Image12" o:spid="_x0000_i1063" type="#_x0000_t75" alt="image" style='width:109.8pt;
   height:94.8pt;visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image063.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=146 height=126
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image064.gif" alt=image
  v:shapes="Image12"><![endif]></span></p>
  </td>
  <td width=422 valign=top style='width:316.8pt;padding:0cm 5.4pt 0cm 5.4pt'>
  <p class=Compact align=center style='text-align:center'><span
  style='mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape
   id="Image13" o:spid="_x0000_i1062" type="#_x0000_t75" alt="image" style='width:255pt;
   height:204.6pt;visibility:visible;mso-wrap-style:square'>
   <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image065.png"
    o:title="image"/>
  </v:shape><![endif]--><![if !vml]><img border=0 width=340 height=273
  src="DiyabcRandomForestUserManual21-01-2021_fichiers/image066.jpg" alt=image
  v:shapes="Image13"><![endif]></span></p>
  </td>
 </tr>
</table>

<h2><a name="_Toc62127938"></a><a
name=Xc43f7bb58de29b50adf97c1b0251e0ff311883c><span style='mso-bookmark:_Toc62127938'><span
lang=EN-US>2.4 Mutation model parameterization (microsatellite and DNA sequence
loci)</span></span></a></h2>

<p class=FirstParagraph><span lang=EN-US>The program can analyze microsatellite
data and DNA sequence data altogether as well as separately. SNP loci can be
also analyzed separately from microsatellite data and DNA sequence data. It is
worth stressing that all loci in an analysis must be <u>genetically independent</u>.
Second, for DNA sequence loci, <span class=SpellE>intralocus</span>
recombination is not considered. Loci are grouped by the user according to its
needs. For microsatellite and DNA sequence, a different mutation model can be
defined for each group. For instance, one group can include all microsatellites
with motifs that are 2 bp long and another group those with a 4 bp long motif.
Also, with DNA sequence loci, nuclear loci can be grouped together and a
mitochondrial locus form a separate group. <u>SNPs do not require mutation
model parameterization (see below for details)</u>.</span></p>

<p class=FirstParagraph><span lang=EN-US>We now describe the parameterization
of microsatellite and DNA sequence markers.</span></p>

<h3><a name="_Toc62127939"></a><a name=microsatellite-loci><span
style='mso-bookmark:_Toc62127939'><span lang=EN-US>2.4.1 Microsatellite loci</span></span></a></h3>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US>Although a variety of mutation models have been proposed for
microsatellite loci, it is usually sufficient to consider only the simplest
models (e.g. Estoup et al. 2002). This has the non-negligible advantage of
reducing the number of parameters. This is why we chose the Generalized
Stepwise Mutation model (GSM). Under this model, a mutation increases or
decreases the length of the microsatellite by a number of repeated motifs
following a geometric distribution. This model necessitates only two
parameters: the mutation rate (</span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>&micro;</span></span><span
lang=EN-US>) and the parameter of the geometric distribution (</span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>P</span></span><span lang=EN-US>). The same mutation model is imposed
to all loci of a given group. However, each locus has its own parameters
(&micro;<span class=SpellE><sub>i</sub></span> and P<sub>i</sub></span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>) </span></span><span lang=EN-US>and, following a hierarchical scheme,
each locus parameter is drawn from a gamma distribution with mean equal to the
mean parameter value. </span></p>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US>Note also that:</span></p>

<p class=MsoListParagraphCxSpFirst style='margin-top:0cm;margin-right:0cm;
margin-bottom:0cm;margin-left:54.0pt;margin-bottom:.0001pt;mso-add-space:auto;
text-indent:-18.0pt;mso-list:l19 level1 lfo13'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>1.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Individual loci parameters
(&micro;<span class=SpellE>i</span> and Pi) are considered as nuisance
parameters and hence are never recorded. Only mean parameters are recorded.</span></p>

<p class=MsoListParagraphCxSpMiddle style='margin-top:0cm;margin-right:0cm;
margin-bottom:0cm;margin-left:54.0pt;margin-bottom:.0001pt;mso-add-space:auto;
text-indent:-18.0pt;mso-list:l19 level1 lfo13'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>2.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The variance or shape parameter
of the gamma distributions are set by the user and are NOT considered as
parameters.</span></p>

<p class=MsoListParagraphCxSpMiddle style='margin-top:0cm;margin-right:0cm;
margin-bottom:0cm;margin-left:54.0pt;margin-bottom:.0001pt;mso-add-space:auto;
text-indent:-18.0pt;mso-list:l19 level1 lfo13'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>3.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The SMM or Stepwise Mutation
Model is a special case of the GSM in which the number of repeats involved in a
mutation is always one. Such a model can be easily achieved by setting the
maximum value of mean P (</span><!--[if gte msEquation 12]><m:oMath><m:bar><m:barPr><m:pos
    m:val="top"/><span style='font-family:"Cambria Math",serif;mso-ascii-font-family:
   "Cambria Math";mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:barPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>P</m:r></span></i></m:e></m:bar></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:7.8pt;height:16.2pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image067.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=10 height=22
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image068.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>) to 0. In this case, all loci have their </span><!--[if gte msEquation 12]><m:oMath><m:sSub><m:sSubPr><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>P</m:r></span></i></m:e><m:sub><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>i</m:r></span></i></m:sub></m:sSub></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:10.2pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image069.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=14 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image070.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>set equal to 0 whatever
the shape of the gamma distribution.</span></p>

<p class=MsoListParagraphCxSpLast style='margin-top:0cm;margin-right:0cm;
margin-bottom:0cm;margin-left:54.0pt;margin-bottom:.0001pt;mso-add-space:auto;
text-indent:-18.0pt;mso-list:l19 level1 lfo13'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>4.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>All loci can be given the same
value of a parameter by setting the shape of the corresponding gamma
distribution to 0 (this is NOT a limiting case of the gamma, but only a way of
telling the program).</span></p>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US>To give more flexibility to the mutation model, the program offers
the possibility to consider <u>mutations that insert or delete a single
nucleotide</u> to the microsatellite sequence by using a mean parameter (named </span><!--[if gte msEquation 12]><m:oMath><m:sSub><m:sSubPr><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>&#956;</m:r></span></i></m:e><m:sub><m:d><m:dPr><span
     style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
     mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:dPr><m:e><i
     style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:
     "Cambria Math",serif'><m:r>SNI</m:r></span></i></m:e></m:d></m:sub></m:sSub></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:4.5pt;mso-text-raise:-4.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:28.8pt;height:24.6pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image071.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=38 height=33
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image072.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>) with a prior to be defined and individual loci having either
values identical to the mean parameter or drawn from a Gamma distribution.</span></p>

<h3><a name="_Toc62127940"></a><a name=dna-sequence-loci><span
style='mso-bookmark:_Toc62127940'><span lang=EN-US>2.4.2 DNA sequence loci</span></span></a></h3>

<p class=FirstParagraph><span lang=EN-US>The program does not consider
insertion-deletion mutations, mainly because there does not seem to be much
consensus on this topic. Concerning substitutions, only the simplest models are
considered. We chose the Jukes-Cantor (1969) one parameter model, the Kimura
(1980) two parameter model, the Hasegawa-<span class=SpellE>Kishino</span>-Yano
(1985) and the Tamura-<span class=SpellE>Nei</span> (1993) models. The last two
models include the ratios of each nucleotide as parameters. In order to reduce
the number of parameters, these ratios have been fixed to the values calculated
from the observed dataset for each DNA sequence locus. Consequently, this
leaves two and three parameters for the Hasegawa-<span class=SpellE>Kishino</span>-Yano
(HKY) and Tamura-<span class=SpellE>Nei</span> (TN), respectively. Also, two
adjustments are possible: one can fix the fraction of constant sites (those
that cannot mutate) on the one hand and the shape of the Gamma distribution of
mutations among sites on the other hand. As for microsatellites, all sequence
loci of the same group are given the same mutation model with mean parameter(s)
drawn from priors and each locus has its own parameter(s) drawn from a Gamma
distribution (same hierarchical scheme). Notes 1, 2 and 4 of previous
subsection (2.4.1) apply also for sequence loci.</span></p>

<h3><a name="_Toc62127941"></a><a
name=X967a0daa6c052f45bf185add8c047b9aac1fd89><span style='mso-bookmark:_Toc62127941'><span
lang=EN-US>2.4.3 SNPs do not require mutation model parameterization</span></span></a><span
style='mso-bookmark:_Toc62127941'><span lang=EN-US> &#8211; notion of MAF and
MRC</span></span></h3>

<p class=FirstParagraph><span lang=EN-US>SNPs have two characteristics that
allow to get rid of mutation models: they are (necessarily) polymorphic and
they present only two allelic states (ancestral and derived). In order to be
sure that all analyzed SNP loci have the two characteristics, <u>non-polymorphic
loci are discarded right from the beginning of analyses</u>. Note that a
warning message will appear if the observed dataset include monomorphic loci,
the latter being automatically removed from further analyses by the program.
Consequently, one can assume that there occurred one and only one mutation in
the coalescence tree of sampled genes. We will see below that this largely
simplifies (and speeds up) SNP data simulation as one can use in this case the
efficient &#8220;-s&#8221; algorithm of Hudson (2002) (<span class=SpellE>Cornuet</span>
et al.&nbsp;2014). Also, this advantageously reduces the dimension of the
parameter space as mutation parameters are not needed in this case. There is
however a potential drawback which is the absence of any calibration generally
brought by priors on mutation parameters. <u>Consequently, time/effective size
ratios rather than original time or effective size parameters will be
informative</u>.</span></p>

<p class=FirstParagraph style='text-indent:36.0pt'><span lang=EN-US>It is worth
noting that, using the Hudson&#8217;s simulation algorithm for SNP markers
leads to applying a default <u>MAF (minimum allele frequency)</u> criterion on
the simulated dataset. As a matter of fact, each locus in both the observed and
simulated datasets will be characterized by the presence of at least one copy
of the SNP alleles over all genes sampled from all studied populations (i.e.
pooling all genes genotyped at the locus). In DIYABC Random Forest v1.0, it is
possible to impose a given MAF criterion on the observed and simulated
datasets. This MAF is computed pooling all genes genotyped over all studied
population samples. For instance, the specification of a MAF equal to 5% will
automatically select a subset of <i style='mso-bidi-font-style:normal'>m</i>
loci characterized by a minimum allele frequency &gt; 5% out of the <i
style='mso-bidi-font-style:normal'>l</i> loci of the observed dataset. In
agreement with this, only <i style='mso-bidi-font-style:normal'>m</i> loci with
a MAF&gt;5% will be retained in a simulated dataset. </span><span lang=EN-US
style='mso-bidi-font-family:CMR10'>In practice, the instruction for a given MAF
has to be indicated directly in the headline of the file of the observed
dataset (see section 7.1.1 for data file examples). For instance, if one wants
to consider only loci with a MAF equal to 5% one will write </span><span
lang=EN-US style='mso-bidi-font-family:CMMI10'>&lt;</span><span lang=EN-US
style='mso-bidi-font-family:CMSS10'>MAF=0.05</span><span lang=EN-US
style='mso-bidi-font-family:CMMI10'>&gt; </span><span lang=EN-US
style='mso-bidi-font-family:CMR10'>in the headline. Writing </span><span
lang=EN-US style='mso-bidi-font-family:CMMI10'>&lt;</span><span lang=EN-US
style='mso-bidi-font-family:CMSS10'>MAF=<span class=SpellE>hudson</span></span><span
lang=EN-US style='mso-bidi-font-family:CMMI10'>&gt; </span><span lang=EN-US
style='mso-bidi-font-family:CMR10'>(or omitting to write any instruction with
respect to the MAF) will bring the program to use the standard Hudson's
algorithm without further selection. The selection of a subset of loci fitting
a given MAF allows: (<span class=SpellE>i</span>) to remove the loci with very
low level of polymorphism from the dataset and hence increase the mean level of
genetic variation of both the observed and simulated datasets, without producing
any bias in the analyses; and (ii) to reduce the proportion of loci for which
the observed variation may corresponds to sequencing errors.</span><span
lang=EN-US> In practice MAF values &#8804;10% are considered. To check for the
consistency/robustness of the ABC results obtained, it may be useful to treat a
SNP dataset considering different MAFs (for instance MAF=<span class=SpellE>hudson</span>,
MAF=1% and MAF=5%).</span></p>

<p class=MsoBodyText style='text-indent:36.0pt'><span lang=EN-US>It is worth
stressing that the MAF criterion applies to standard Individual sequencing
(hereafter <span class=SpellE><b style='mso-bidi-font-weight:normal'>IndSeq</b></span>)
SNP data.<span style='mso-spacerun:yes'>&nbsp; </span>In addition to <span
class=SpellE>IndSeq</span> data, DIYABC Random Forest v1.0 allows the
simulation and analyses of pool-sequencing SNP data (hereafter <span
class=SpellE><b style='mso-bidi-font-weight:normal'>PoolSeq</b></span> data),
which basically consist of whole-genome sequences of pools of tens to hundreds
of individual DNAs (Gautier et al., 2013; <span class=SpellE>Schl&ouml;tterer</span>
et al., 2014). In practice, the simulation of <span class=SpellE>PoolSeq</span>
data consists first in simulating individual SNP genotypes for all individuals
in each population pool, and then generating pool read counts from a binomial
distribution parameterized with the simulated allele counts (obtained from
individual SNP genotypes) and the total pool read coverage (e.g., <span
class=SpellE>Hivert</span>, et al. 2018). A criterion somewhat similar to the
MAF was implemented for <span class=SpellE>PoolSeq</span> data: <u>the minimum
read count (MRC)</u>. The MRC is the number of sequence reads of the minor
allele frequency allele when pooling the reads <span class=SpellE>over all</span>
population samples. The specification of a MRC equal for instance to 5 will
automatically select a subset of <i style='mso-bidi-font-style:normal'>m</i> <span
class=SpellE>PoolSeq</span> loci characterized by more than five reads over all
studied pools among the <i style='mso-bidi-font-style:normal'>l</i> loci of the
observed dataset. In agreement with this, only <i style='mso-bidi-font-style:
normal'>m</i> loci with more than five reads will be retained in a simulated
dataset.</span></p>

<h2><a name="_Toc62127942"></a><a name=prior-distributions><span
style='mso-bookmark:_Toc62127942'><span lang=EN-US>2.5 Prior distributions</span></span></a></h2>

<p class=FirstParagraph><span lang=EN-US>The Bayesian aspect of the ABC-RF
approach implies that parameter estimations use prior knowledge about these
parameters that is given by prior distributions of parameters. The program
offers a choice among usual probability distributions, i.e.&nbsp;Uniform,
Log-Uniform, Normal or Log-Normal for historical parameters and Uniform,
Log-Uniform or Gamma for mutation parameters. Extremum values (min and max) and
other parameters (e. g. mean and standard deviation) must be filled in by the
user. It is worth noting that one can impose some simple conditions on
historical parameters. For instance, there can be two times parameters with
overlapping prior distributions. However, we want that the first one, say </span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>t1</span></span><span lang=EN-US>, to always be larger than the second
one, say </span><span class=VerbatimChar><span lang=EN-US style='font-size:
11.0pt;mso-bidi-font-size:12.0pt'>t2</span></span><span lang=EN-US>. For that,
we just need to set </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>t1</span></span><span
lang=EN-US>&gt;</span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>t2</span></span><span
lang=EN-US> in the corresponding edit-windows. Such a condition needs to be
between two parameters and more precisely between two parameters of the same
category (i.e.&nbsp;two effective sizes, two times or two admixture rates). The
limit to the number of conditions is imposed by the logics, not by the program.
The only binary relationships accepted here are &lt;, &gt;, &#8804;, and
&#8805;.</span></p>

<h2><a name="_Toc62127943"></a><a name=summary-statistics><span
style='mso-bookmark:_Toc62127943'><span lang=EN-US>2.6 Summary statistics</span></span></a><span
style='mso-bookmark:_Toc62127943'><span lang=EN-US> as components of the
feature vector</span></span><span lang=EN-US> </span></h2>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US>The training set includes values of a feature vector which is a
multidimensional representation of any data point (i.e. simulated or observed
datasets) made up of measurements (or features) taken from it. More
specifically, the feature vector includes a large number of statistics that
summarize genetic variation in the way that they allow capturing different
aspects of gene genealogies and hence various features of molecular patterns
generated by selectively neutral population histories (e.g. Beaumont 2010; <span
class=SpellE>Cornuet</span> et al. 2014). For each category (microsatellite,
DNA sequences or SNP) of loci, the program proposes a series of summary
statistics among those used by population geneticists. </span></p>

<p class=MsoBodyText><span lang=EN-US>Note: The code name of each statistic
mentioned in the various program outputs is given between brackets [XXX] in
sections 2.6.1, 2.6.2 and 2.6.3.</span></p>

<p class=MsoBodyText style='margin:0cm;margin-bottom:.0001pt'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<h3 style='margin-top:0cm;margin-right:0cm;margin-bottom:10.0pt;margin-left:
0cm'><a name="_Toc62127944"></a><a name=for-microsatellite-loci><span
style='mso-bookmark:_Toc62127944'><span lang=EN-US>2.6.1 For microsatellite
loci</span></span></a></h3>

<p class=DefinitionTerm><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US>Single sample statistics:</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l24 level1 lfo5'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>1.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[NAL] - mean number of alleles
across loci</span></p>

<p class=Definition style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.0pt;margin-bottom:.0001pt;text-indent:-24.0pt;mso-list:l24 level1 lfo5'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>2.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US><span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;</span>[HET] - mean gene
diversity across loci (<span class=SpellE>Nei</span> 1987)</span></p>

<p class=Definition style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.1pt;margin-bottom:.0001pt;text-indent:-24.1pt;mso-list:l24 level1 lfo5'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>3.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US><span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;</span>[VAR] - mean allele
size variance across loci</span></p>

<p class=MsoListParagraph style='margin-left:24.0pt;mso-add-space:auto;
text-indent:-24.0pt;mso-list:l24 level1 lfo5'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>4.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US><span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;</span>[MGW] - mean M index
across loci (Garza and Williamson 2001; <span class=SpellE>Excoffier</span> et
al. 2005)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US>Two sample statistics:</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l3 level1 lfo6'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>1.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[N2P] - mean number of alleles
across loci (two samples)</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l3 level1 lfo6'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>2.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[H2P] - mean gene diversity
across loci (two samples)</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l3 level1 lfo6'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>3.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[V2P] - mean allele size
variance across loci (two samples)</span></p>

<p class=Definition style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.0pt;margin-bottom:.0001pt;text-indent:-24.0pt;mso-list:l3 level1 lfo6'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>4.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US style='mso-fareast-font-family:
Cambria;mso-fareast-theme-font:minor-fareast'><span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;</span></span><span
lang=EN-US>[FST] -</span><!--[if gte msEquation 12]><m:oMath><m:sSub><m:sSubPr><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>F</m:r></span></i></m:e><m:sub><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>ST</m:r></span></i></m:sub></m:sSub></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:16.2pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image073.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=22 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image074.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>between two samples
(Weir and Cockerham 1984)</span></p>

<p class=Definition style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.0pt;margin-bottom:.0001pt;text-indent:-24.0pt;mso-list:l3 level1 lfo6'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>5.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US><span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;</span>[LIK] - mean index of
classification (two samples) </span></p>

<p class=Definition style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.0pt;margin-bottom:.0001pt'><span lang=EN-US><span
style='mso-spacerun:yes'>&nbsp;&nbsp; </span></span><span style='mso-ansi-language:
FR'>(<span class=SpellE>Rannala</span> and <span class=SpellE>Moutain</span> <span
class=GramE>1997;</span> Pascual et al. 2007)<o:p></o:p></span></p>

<p class=Definition style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.0pt;margin-bottom:.0001pt;text-indent:-24.0pt;mso-list:l3 level1 lfo6'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>6.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='mso-ansi-language:FR'><span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;</span></span><span
lang=EN-US>[DAS] - shared allele distance between two samples (Chakraborty and <span
class=SpellE>Jin</span> 1993)</span></p>

<p class=Definition style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.0pt;margin-bottom:.0001pt;text-indent:-24.0pt;mso-list:l3 level1 lfo6'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>7.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US style='mso-fareast-font-family:
Cambria;mso-fareast-theme-font:minor-fareast'><span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;</span></span><span
lang=EN-US>[DM2] -<span style='mso-spacerun:yes'>&nbsp; </span>distance between
two samples (<span class=SpellE>Golstein</span> et al. 1995)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US>Three sample statistics:</span></p>

<p class=Definition style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.0pt;margin-bottom:.0001pt;text-indent:-24.0pt;mso-list:l30 level1 lfo7'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>1.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[AML] - Maximum likelihood
coefficient of admixture (<span class=SpellE>Choisy</span> et al. 2004)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<h3 style='margin-top:0cm;margin-right:0cm;margin-bottom:10.0pt;margin-left:
0cm'><a name="_Toc62127945"></a><a name=for-dna-sequence-loci><span
style='mso-bookmark:_Toc62127945'><span lang=EN-US>2.6.2 For DNA sequence loci</span></span></a></h3>

<p class=MsoBodyText style='margin:0cm;margin-bottom:.0001pt'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US>Single sample statistics:</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l33 level1 lfo8'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>1.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[NHA] - number of distinct
haplotypes</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l33 level1 lfo8'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>2.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[NSS] - number of segregating
sites</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l33 level1 lfo8'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>3.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[MPD] - mean pairwise
difference</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l33 level1 lfo8'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>4.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[VPD] - variance of the number
of pairwise differences</span></p>

<p class=Definition style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.0pt;margin-bottom:.0001pt;text-indent:-24.0pt;mso-list:l33 level1 lfo8'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>5.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US><span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;</span>[DTA] - Tajima&#8217;s
D statistics (Tajima 1989)</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l33 level1 lfo8'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>6.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[PSS] - Number of private
segregating sites </span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt;text-indent:
36.0pt'><span lang=EN-US>(=number of segregating sites if there is only one
sample)</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l33 level1 lfo8'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>7.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[MNS] - Mean of the numbers of
the rarest nucleotide at segregating sites</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l33 level1 lfo8'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>8.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[VNS] - Variance of the numbers
of the rarest nucleotide at segregating sites</span></p>

<p class=DefinitionTerm><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US>Two sample statistics:</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l38 level1 lfo9'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>1.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[NH2] - number of distinct
haplotypes in the pooled sample</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l38 level1 lfo9'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>2.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[NS2] - number of segregating
sites in the pooled sample</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l38 level1 lfo9'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>3.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[MP2] - mean of within sample
pairwise differences</span></p>

<p class=Definition style='margin:0cm;margin-bottom:.0001pt;text-indent:0cm;
mso-list:l38 level1 lfo9'><![if !supportLists]><span lang=EN-US
style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>4.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[MPB] - mean of between sample
pairwise differences</span></p>

<p class=Definition style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.0pt;margin-bottom:.0001pt;text-indent:-24.0pt;mso-list:l38 level1 lfo9'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>5.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US style='mso-fareast-font-family:
Cambria;mso-fareast-theme-font:minor-fareast'><span
style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></span><span
lang=EN-US>[HST] -</span><!--[if gte msEquation 12]><m:oMath><m:sSub><m:sSubPr><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>F</m:r></span></i></m:e><m:sub><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>ST</m:r></span></i></m:sub></m:sSub></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:16.2pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image073.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=22 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image074.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>between two samples
(Hudson et al. 1992)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm style='margin-left:2.85pt'><span lang=EN-US>Three
sample statistics:</span></p>

<p class=Definition style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:24.0pt;margin-bottom:.0001pt;text-indent:-24.0pt;mso-list:l9 level1 lfo10'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>1.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>[SML] Maximum likelihood
coefficient of admixture (adapted from <span class=SpellE>Choisy</span> et al.
2004)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<h3 style='margin-top:0cm;margin-right:0cm;margin-bottom:10.0pt;margin-left:
2.85pt'><a name="_Toc62127946"></a><a name=for-snp-loci><span style='mso-bookmark:
_Toc62127946'><span lang=EN-US>2.6.3 For SNP loci</span></span></a></h3>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US>In addition to
&#8220;standard&#8221; individual-sequencing SNP data (i.e. <span class=SpellE><b
style='mso-bidi-font-weight:normal'>IndSeq</b></span> data), the program allows
the simulation and analyses of pool-sequencing SNP data (i.e. <span
class=SpellE><b style='mso-bidi-font-weight:normal'>PoolSeq</b></span> data),
which basically consist of whole-genome sequences of pools of tens to hundreds
of individual DNAs (Gautier et al., 2013; <span class=SpellE>Schl&ouml;tterer</span>
et al., 2014). For both <span class=SpellE>IndSeq</span> and <span
class=SpellE>PoolSeq</span> SNPs, we have implemented the following (same) set
of summary statistics. </span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.85pt;margin-bottom:.0001pt;text-indent:-33.0pt'><span
lang=EN-US>1. <span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>[ML1p]
[ML2p] [ML3p] [ML4p] - <i style='mso-bidi-font-style:normal'>Proportion of
monomorphic loci</i> for each population, as well as for each pair, triplet and
quadruplet of populations. </span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>Mean (m suffix added to the code name) and variance (v suffix) over
loci values are computed for all subsequent summary statistics<o:p></o:p></span></b></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.85pt;margin-bottom:.0001pt;text-indent:-33.0pt'><span
lang=EN-US>2. <span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>[<span
class=SpellE>HWm</span>] [<span class=SpellE>HWv</span>] [<span class=SpellE>HBm</span>]
[<span class=SpellE>HBv</span>] - <i style='mso-bidi-font-style:normal'>Heterozygosity</i>
for each population and for each pair of populations (<span class=SpellE>Hivert</span>
et al. 2018). </span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.85pt;margin-bottom:.0001pt;text-indent:-33.0pt'><span
lang=EN-US>3.<span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>[FST1m]
[FST1v] [FST2m] [FST2v] [FST3m] [FST3v] [FST4m] [FST4v] [<span class=SpellE>FSTGm</span>]
[<span class=SpellE>FSTGv</span>] - <i style='mso-bidi-font-style:normal'>FST</i>-<i
style='mso-bidi-font-style:normal'>related statistics</i> for each population
(i.e., population-specific FST; Weir &amp; <span class=SpellE>Goudet</span>
2017), for each pair, triplet, quadruplet and overall populations (when the
dataset includes more than four populations) (<span class=SpellE>Hivert</span>
et al. 2018). </span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.85pt;margin-bottom:.0001pt;text-indent:-33.0pt'><span
lang=EN-US>4. <span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>[F3m]
[F3v] [F4m] [F4v] - <i style='mso-bidi-font-style:normal'>Patterson&#8217;s
f-statistics</i> for each triplet (f3-statistics) and quadruplet
(f4-statistics) of populations (Patterson et al. 2012 and <span class=SpellE>Leblois</span>
et al. 2018 for the <span class=SpellE>PoolSeq</span> unbiased f3-statistics) </span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US>5. <span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>[<span
class=SpellE>NEIm</span>] [<span class=SpellE>NEIv</span>] - <span
class=SpellE><i style='mso-bidi-font-style:normal'>Nei&#8217;s</i></span><i
style='mso-bidi-font-style:normal'> (1972) distance</i> for each pair of
populations</span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.85pt;margin-bottom:.0001pt;text-indent:-33.0pt'><span
lang=EN-US>6. <span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>[<span
class=SpellE>AMLm</span>] [<span class=SpellE>AMLv</span>] - <i
style='mso-bidi-font-style:normal'>Maximum likelihood coefficient of admixture</i>
computed for each triplet of populations (adapted from <span class=SpellE>Choisy</span>
et al. 2004).</span></p>

<h2><a name="_Toc62127947"><span lang=EN-US>2.7 Generating the training set</span></a></h2>

<p class=FirstParagraph style='margin:0cm;margin-bottom:.0001pt'><span
lang=EN-US>The training set can include as many scenarios as desired. The prior
probability of each scenario is uniform so that each scenario will have
approximately the same number of simulated datasets. Only parameters that are
defined for the drawn scenario are generated from their respective prior
distribution. Scenarios may or may not share parameters. When conditions apply
to some parameters (see section 2.3), parameter sets are drawn in their
respective prior distributions until all conditions are fulfilled. The simulated
datasets (summarized with the statistics detailed in section 2.6) of the
training set are recorded for further statistical analyses (using Random Forest
algorithms) in a key binary file named <span class=SpellE><u>reftableRF.bin</u></span>.
</span><a name=the-graphic-user-interface><b><span lang=EN-US style='font-size:
16.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'><o:p></o:p></span></b></a></p>

<h1><span style='mso-bookmark:the-graphic-user-interface'><span lang=EN-US>3. <a
name="_Toc62127948">RANDOM FOREST ANALYSIS</a></span></span></h1>

<h1><span style='mso-bookmark:the-graphic-user-interface'><span lang=EN-US
style='font-size:12.0pt;font-family:"Cambria",serif;mso-ascii-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;color:black;mso-themecolor:text1;font-weight:
normal;mso-bidi-font-weight:bold'>Once the training set has been generated, one
can start various statistical treatments using the Random Forest (RF)
algorithms implemented in the program and by running the &#8220;Random Forest
analyses&#8221; module</span></span></h1>

<h1><span style='mso-bookmark:the-graphic-user-interface'><a name="_Toc62127950"><span
lang=EN-US>3.1 Addition of linear combinations of summary statistics to the vector
feature</span></a><span lang=EN-US> </span></span></h1>

<p class=MsoBodyText><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US>For scenario choice, the feature vector can be enriched before
processing RF predictions (default option that can be disabled) by values of
the <i style='mso-bidi-font-style:normal'>d</i> axes of a linear discriminant
analysis (LDA) processed on the above summary statistics (with <i
style='mso-bidi-font-style:normal'>d</i> equal to the number of scenarios minus
1; <span class=SpellE>Pudlo</span> et al. 2016). In the same spirit, for
parameter estimation, the feature vector can be completed (default option that
can be disabled) by values of a subset of the <i style='mso-bidi-font-style:
normal'>s</i> axes of a Partial Least Squares Regression analysis (PLS) also
processed on the above summary statistics (with <i style='mso-bidi-font-style:
normal'>s</i> equal to the number of summary statistics). The number of PLS
axes added to the feature vector is determined as the number of PLS axes
providing a given fraction of the maximum amount of variance explained by all
PLS axes (i.e., 95% by default, but this parameter can be adjusted). Note that,
according to our own experience on this issue, the addition into the feature
vector of LDA or PLS axes better extract genetic information from the training
set and hence globally improved statistical inferences. While the inferential
gain turned out to be systematic and substantial for LDA axes (scenario
choice), we found that including PLS axes improved parameter estimation in a
heterogeneous way, with a negligible gain in some cases (e.g. Collin et al.
2020).</span></span></p>

<h1><span style='mso-bookmark:the-graphic-user-interface'><a name="_Toc62127951"><span
lang=EN-US>3.2 Prediction using Random Forest: scenario choice</span></a></span></h1>

<p class=MsoBodyText><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US>For scenario choice, the outcome of the first step of RF prediction
applied to a given target dataset is a <u>classification vote for each scenario</u>
which represents the number of times a scenario is selected in a forest of <i
style='mso-bidi-font-style:normal'>n</i> trees. The scenario with the highest
classification vote corresponds to the scenario best suited to the target
dataset among the set of compared scenarios. This first RF predictor is good
enough to select the most likely scenario but not to derive directly the
associated posterior probabilities. A second analytical step based on a second
Random Forest in regression is necessary to provide an estimation of the <u>posterior
probability of the best supported scenario</u> (<span class=SpellE>Pudlo</span>
et al. 2016). </span></span></p>

<h1><span style='mso-bookmark:the-graphic-user-interface'><a name="_Toc62127952"><span
lang=EN-US>3.3 Prediction using Random Forest: parameter estimation</span></a></span></h1>

<p class=MsoBodyText><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US>For parameter estimation, <span class=SpellE>Raynal</span> et al.
(2019) extended the RF approach developed in the context of (non-parametric)
regression to estimate the posterior distributions of a given parameter under a
given scenario. The approach requires the derivation of a new Random Forest for
each component of interest of the parameter vector (i.e. &#8220;one parameter,
one forest strategy&#8221;; <span class=SpellE>Raynal</span> et al. 2019).
Quite often, practitioners of Bayesian inference report the posterior mean,
posterior variance or posterior quantiles, rather than the full posterior
distribution, since the former are easier to interpret than the latter. We
implemented in DIYABC Random Forest the methodologies detailed in <span
class=SpellE>Raynal</span> et al. (2019) to provide estimations of the <u>posterior
mean, variance, median (i.e. 50% quantile) as well as 5% and 95% quantiles (and
hence 90% credibility interval) of each parameter of interest</u>. The
posterior distribution of each parameter of interest is also inferred using
importance weights following <span class=SpellE>Meinshausen</span>
(2006)&#8217;s work on quantile regression forests.</span></span></p>

<h1><span style='mso-bookmark:the-graphic-user-interface'><a name="_Toc62127953"><span
lang=EN-US>3.4 Assessing the quality of predictions</span></a><span lang=EN-US>
</span></span></h1>

<p class=MsoBodyText><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US>To evaluate the robustness of inferences, DIYABC Random Forest
provides:</span></span></p>

<p class=MsoBodyText style='margin-left:39.0pt;text-indent:-36.0pt;mso-list:
l18 level1 lfo14'><span style='mso-bookmark:the-graphic-user-interface'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>(i)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Global (i.e. prior)
error/accuracy indices corresponding to prediction quality measures computed <i
style='mso-bidi-font-style:normal'>over the entire data space</i>;</span></span></p>

<p class=MsoBodyText style='margin-left:39.0pt;text-indent:-36.0pt;mso-list:
l18 level1 lfo14'><span style='mso-bookmark:the-graphic-user-interface'><![if !supportLists]><span
lang=EN-US style='mso-bidi-font-family:Cambria;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>(ii)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Local (i.e. posterior)
error/accuracy indices computed conditionally on the observed dataset and hence
corresponding to prediction quality <i style='mso-bidi-font-style:normal'>exactly
at the position of the observed dataset</i>. </span></span></p>

<p class=MsoBodyText style='margin-left:3.0pt'><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US>Note that the program used the out-of-bag prediction method for
estimating global and local error/accuracy measures (<span class=SpellE>Pudlo</span>
et al. 2016; <span class=SpellE>Raynal</span> et al. 2019; <span class=SpellE>Chapuis</span>
et al. 2020). This method is computationally efficient as it makes use of the
datasets already present in the training set and hence avoids the
(computationally costly) simulations of additional test datasets.</span></span></p>

<h3 style='margin-top:0cm;margin-right:0cm;margin-bottom:10.0pt;margin-left:
2.85pt'><span style='mso-bookmark:the-graphic-user-interface'><a
name="_Toc62127954"><span lang=EN-US>3.4.1 Indices for scenario choice</span></a></span></h3>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l42 level1 lfo15'><span style='mso-bookmark:the-graphic-user-interface'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><u><span lang=EN-US>Global prior errors</span></u><span
lang=EN-US> including the <i style='mso-bidi-font-style:normal'>confusion
matrix</i> (i.e. the contingency table of the true and predicted classes
&#8211; here scenarios - for each example in the training set) and the <i
style='mso-bidi-font-style:normal'>mean misclassification error rate</i> (over
all scenarios).</span></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l42 level1 lfo15'><span style='mso-bookmark:the-graphic-user-interface'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><u><span lang=EN-US>Local posterior error</span></u><span
lang=EN-US> which corresponds to 1 minus the posterior probability of the
selected scenario (<span class=SpellE>Chapuis</span> et al. 2020). </span></span></p>

<h3 style='margin-top:0cm;margin-right:0cm;margin-bottom:10.0pt;margin-left:
2.85pt'><span style='mso-bookmark:the-graphic-user-interface'><a
name="_Toc62127955"><span lang=EN-US>3.4.2 Indices for parameter estimation</span></a></span></h3>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l1 level1 lfo16'><span style='mso-bookmark:the-graphic-user-interface'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><u><span lang=EN-US>Global (prior) and local
(posterior) NMAE</span></u><span lang=EN-US> (i.e. normalized mean absolute
error): it is the average absolute difference between the point estimate and
the true simulated value divided by the true simulated value, with the mean or
the median taken as point estimate.</span></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l1 level1 lfo16'><span style='mso-bookmark:the-graphic-user-interface'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><u><span lang=EN-US>Global (prior) and local
(posterior) MSE and NMSE</span></u><span lang=EN-US> (i.e. the mean square
error): it is the average squared difference between the point estimate and the
true simulated value for MSE, divided by the true simulated value for NMSE,
again with the mean or the median taken as point estimate.</span></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l1 level1 lfo16'><span style='mso-bookmark:the-graphic-user-interface'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><u><span lang=EN-US>Several confidence interval
measures</span></u><span lang=EN-US> computed only at the global (prior) scale:</span></span></p>

<p class=MsoBodyText style='margin-left:36.0pt'><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US>- <i style='mso-bidi-font-style:normal'>90% coverage</i>: it is the
proportion of true simulated values located between the estimated 5% and 95%
quantiles.</span></span></p>

<p class=MsoBodyText style='margin-left:36.0pt'><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US>- <i style='mso-bidi-font-style:normal'>Mean or median of the 90%
amplitude</i>: it is the mean or median of the difference between the estimated
5% and 95% quantiles.</span></span></p>

<p class=MsoBodyText style='margin-left:36.0pt'><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US>- <i style='mso-bidi-font-style:normal'>Mean or median of the
relative 90% amplitude</i>: it is the mean or median of the difference between
the estimated 5% and 95% divided by the true simulated value.</span></span></p>

<h1><span style='mso-bookmark:the-graphic-user-interface'><a name="_Toc62127956"><span
lang=EN-US>4. PRACTICAL CONSIDERATIONS FOR ABC-RF TREATMENTS</span></a></span></h1>

<p class=MsoBodyText><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US>Random Forest is often (positively) considered as a
&#8220;tuning-free&#8221; method in the sense that it does not require
meticulous calibrations. This represents an important advantage of this method,
especially for non-expect users. In practice, we nevertheless advise users to
consider several checking points, thereafter formalized as questions, before
finalizing inferential treatments using DIYABC Random Forest v1.0.</span></span></p>

<h1><span style='mso-bookmark:the-graphic-user-interface'><a name="_Toc62127957"><span
lang=EN-US style='font-size:14.0pt'>4.1 Are my scenarios and/or associated
priors compatible with the observed dataset?</span></a></span><span
style='mso-bookmark:the-graphic-user-interface'><span lang=EN-US
style='font-size:14.0pt'><o:p></o:p></span></span></h1>

<p class=MsoBodyText><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US>This question is of prime interest and applies to ABC-RF as well as
to any alternative ABC treatments. This issue is particularly crucial, given
that complex scenarios and high dimensional datasets (i.e., large and hence
very informative datasets) are becoming the norm in population genomics.
Basically, if none of the proposed scenario / prior combinations produces some
simulated datasets in a reasonable vicinity of the observed dataset, this is a
signal of incompatibility and it is not recommended to attempt any inferences.
In such situations, we strongly advise reformulating the compared scenarios
and/or the associated prior distributions in order to achieve some
compatibility in the above sense. DIYABC Random Forest v1.0 proposes a visual
way to address this issue through the simultaneous projection of datasets of
the training database and of the observed dataset on the first Linear
Discriminant Analysis (LDA) axes. In the LDA projection, the (target) observed
dataset has to be reasonably located within the clouds of simulated datasets.
The program also proposes some complementary dedicated tools: (<span
class=SpellE>i</span>) a Principal Component Analysis (PCA) representing on
2-axes plans the simulated dataset from the training set and the observed
dataset; and (ii), <u>for each summary statistics</u>, the proportion of
simulated data (considering the total training set) that have a value below the
value of the observed dataset. A star indicates proportions lower than 5% or
greater than 95% (two stars, &lt;1% or &gt;1%; three stars, &lt;0.1% or &gt;0.1%).
The latter numerical results can help users to reformulating the compared
scenarios and/or the associated prior distributions in order to achieve some
compatibility (see e.g. <span class=SpellE>Cornuet</span> et al. 2010).</span></span></p>

<h1><span style='mso-bookmark:the-graphic-user-interface'><a name="_Toc62127958"><span
lang=EN-US style='font-size:14.0pt'>4.2 Did I simulate enough datasets for my
training set?</span></a></span><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US style='font-size:14.0pt'><o:p></o:p></span></span></h1>

<p class=MsoBodyText><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US>A rule of thumb is, for scenario choice to simulate between 2,000
and 20,000 datasets per scenario among those compared (<span class=SpellE>Pudlo</span>
et al. 2016; Estoup et al. 2018), and for parameter estimation to simulate
between 10,000 and 100,000 datasets under a given scenario (<span class=SpellE>Raynal</span>
et al. 2019; <span class=SpellE>Chapuis</span> et al. 2020). To evaluate
whether or not this number is sufficient for RF analysis, we recommend to
compute error/accuracy metrics such as those proposed by DIYABC Random Forest
v1.0 from both the entire training set and a subset of the latter (for instance
from a subset of 80,000 simulated datasets if the training set includes a total
of 100,000 simulated datasets). If error (accuracy) metrics from the subset are
similar, or only slightly higher (lower) than the value obtained from the
entire database, one can consider that the training set contains enough
simulated datasets. If a substantial difference is observed between both
values, then we recommend increasing the number of simulated datasets in the
training set.</span></span></p>

<h1><span style='mso-bookmark:the-graphic-user-interface'><a name="_Toc62127959"><span
lang=EN-US style='font-size:14.0pt'>4.3 Did my forest grow enough trees?</span></a></span><span
style='mso-bookmark:the-graphic-user-interface'><span lang=EN-US
style='font-size:14.0pt'><o:p></o:p></span></span></h1>

<p class=MsoBodyText><span style='mso-bookmark:the-graphic-user-interface'><span
lang=EN-US>According to our experience, a forest made of 500 to 2,000 trees
often constitutes an interesting trade-off between computation efficiency and
statistical precision (<span class=SpellE>Breiman</span>, 2001; <span
class=SpellE>Chapuis</span> et al. 2020; <span class=SpellE>Pudlo</span> et al.
2016; <span class=SpellE>Raynal</span> et al. 2019). To evaluate whether or not
this number is sufficient, we recommend plotting error/accuracy metrics as a
function of the number of trees in the forest. The shapes of the curves provide
a visual diagnostic of whether such key metrics stabilize when the number of
trees tends to a given value. DIYABC Random Forest v1.0 provides such a
plot-figure as output.</span></span><span style='mso-bookmark:the-graphic-user-interface'><b><span
lang=EN-US style='font-size:16.0pt;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'><o:p></o:p></span></b></span></p>

<h1><span style='mso-bookmark:the-graphic-user-interface'><a name="_Toc62127960"><span
lang=EN-US style='font-family:"Cambria",serif;mso-ascii-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin'>5. RUNNING (EXAMPLE) DATASET TREATMENTS USING
THE GRAPHIC USER INTERFACE</span></a></span><span style='mso-bookmark:_Toc62127960'><span
lang=EN-US style='font-family:"Cambria",serif;mso-ascii-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin'> (GUI)</span></span><span lang=EN-US
style='font-family:"Cambria",serif;mso-ascii-theme-font:minor-latin;mso-hansi-theme-font:
minor-latin'><o:p></o:p></span></h1>

<p class=MsoNormal style='margin-top:10.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:2'><a name="doc_newSNPProjectButton"></a><a
name="_Toc62127961"><span style='mso-bookmark:doc_newSNPProjectButton'><b><span
lang=EN-US style='font-size:14.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'>5.1 Launching the GUI</span></b></span></a><span
style='mso-bookmark:doc_newSNPProjectButton'><b><span lang=EN-US
style='font-size:14.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'><o:p></o:p></span></b></span></p>

<ul style='margin-top:0cm' type=disc>
 <li class=MsoNormal style='color:black;mso-themecolor:text1;margin-top:9.0pt;
     margin-bottom:9.0pt;mso-list:l14 level1 lfo42'><span style='mso-bookmark:
     doc_newSNPProjectButton'><span lang=EN-US>The GUI is the standard and
     user-friendly interface to use the software DIYABC-RF. You can configure
     the required parameters and follow the progress directly in the interface
     and through log files. A given GUI session (i.e. a window) allows working
     on a single project. You can launch several GUI sessions (i.e. several
     windows) on a single computer hence allowing working on several projects
     in parallel. <o:p></o:p></span></span></li>
 <li class=MsoNormal style='color:black;mso-themecolor:text1;margin-top:9.0pt;
     margin-bottom:9.0pt;mso-list:l14 level1 lfo42'><span style='mso-bookmark:
     doc_newSNPProjectButton'><span lang=EN-US>When using the operating system
     Microsoft Windows, launch the GUI by double-clicking on <s>the icon of </s>the
     file DIYABC-RF_GUI.bat included in the DIYABC-RF_GUI_xxx_windows.zip file
     available from </span></span><span style='mso-bookmark:doc_newSNPProjectButton'></span><a
     href="https://diyabc.github.io/"><span style='mso-bookmark:doc_newSNPProjectButton'><span
     lang=EN-US style='font-family:"Times New Roman",serif;color:black;
     mso-themecolor:text1;text-decoration:none;text-underline:none'>h</span></span></a><span
     style='mso-bookmark:doc_newSNPProjectButton'></span><a
     href="https://diyabc.github.io/"><span style='mso-bookmark:doc_newSNPProjectButton'><span
     lang=EN-US style='font-family:"Times New Roman",serif;color:black;
     mso-themecolor:text1;text-decoration:none;text-underline:none'>ttps://diyabc.github.io</span></span></a><span
     style='mso-bookmark:doc_newSNPProjectButton'><span lang=EN-US
     style='font-family:"Times New Roman",serif'> </span><span lang=EN-US>.
     When using MacOS, launch the GUI by double-clicking on the icon of the
     file DIYABC-<span class=SpellE>RF_GUI.command</span> included in the
     DIYABC-RF_GUI_xxx_macos.zip file available from </span></span><span
     style='mso-bookmark:doc_newSNPProjectButton'></span><a
     href="https://diyabc.github.io/"><span style='mso-bookmark:doc_newSNPProjectButton'><span
     lang=EN-US style='font-family:"Times New Roman",serif;color:black;
     mso-themecolor:text1;text-decoration:none;text-underline:none'>https://diyabc.github.io</span></span></a><span
     style='mso-bookmark:doc_newSNPProjectButton'><span lang=EN-US> <o:p></o:p></span></span></li>
</ul>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span style='mso-bookmark:doc_newSNPProjectButton'><span
lang=EN-US style='color:black;mso-themecolor:text1'>It is worth stressing that
the program can also be run locally as any shiny application using R (only
possibility on Linux). You can visit </span></span><a
href="https://diyabc.github.io/gui/#r-package-installation"><span
style='mso-bookmark:doc_newSNPProjectButton'><span lang=EN-US style='color:
black;mso-themecolor:text1;text-decoration:none;text-underline:none'>https://diyabc.github.io/gui/#r-package-installation</span></span></a><span
style='mso-bookmark:doc_newSNPProjectButton'><span lang=EN-US style='color:
black;mso-themecolor:text1'> for more details. <o:p></o:p></span></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span style='mso-bookmark:doc_newSNPProjectButton'><span
lang=EN-US style='color:black;mso-themecolor:text1'>The main pipeline of
DIYABC-RF includes two <s>main</s> modules corresponding to the two main
phases:<o:p></o:p></span></span></p>

<ul style='margin-top:0cm' type=disc>
 <li class=MsoNormal style='color:black;mso-themecolor:text1;margin-top:9.0pt;
     margin-bottom:9.0pt;mso-list:l29 level1 lfo40'><span style='mso-bookmark:
     doc_newSNPProjectButton'><span lang=EN-US>Module 1 = &#8220;<u>Training
     set simulation</u>&#8221; where users specify how simulated data will be
     generated under the ABC framework to produce a training set.<o:p></o:p></span></span></li>
</ul>

<ul style='margin-top:0cm' type=disc>
 <li class=MsoNormal style='color:black;mso-themecolor:text1;margin-top:9.0pt;
     margin-bottom:9.0pt;mso-list:l7 level1 lfo41'><span style='mso-bookmark:
     doc_newSNPProjectButton'><span lang=EN-US>Module 2 = &#8220;<u>Random
     Forest analysis</u>&#8221; (necessitating the presence of a training set)
     which guides users through scenario choice and parameter inference by
     providing a simple interface for the supervised learning framework based
     on Random Forest methodologies.<span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><o:p></o:p></span></span></li>
</ul>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span style='mso-bookmark:doc_newSNPProjectButton'><span
lang=EN-US style='color:black;mso-themecolor:text1'>An additional (auxiliary)
module named &#8220;<u>Synthetic data file generation</u>&#8221; is also
available from the GUI (cf. section 8). It can be used to easily generate data
file(s) corresponding to synthetic &#8220;ground truth&#8221; raw data (not
summarized through statistics) under a given historical scenario and a set of
fixed parameter values, for instance for benchmarking purpose. The formats of
such simulated data files are the same as those described in section 7.1 for
observed (i.e. real) datasets for which one usually wants to make inferences.<o:p></o:p></span></span></p>

<p class=MsoNormal style='margin-top:10.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:2'><span style='mso-bookmark:doc_newSNPProjectButton'><a
name="_Toc62127962"></a><a name="doc_openProjectButton"><span style='mso-bookmark:
_Toc62127962'><b><span lang=EN-US style='font-size:14.0pt;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'>5.2 What is a DIYABC Random Forest
project?</span></b></span></a></span><span style='mso-bookmark:doc_newSNPProjectButton'><b><span
lang=EN-US style='font-size:14.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'><o:p></o:p></span></b></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span style='mso-bookmark:doc_newSNPProjectButton'><span
lang=EN-US style='color:black;mso-themecolor:text1'>A DIYABC-RF project
consists of different files, related to the training set simulation and random
forest analyses. A project includes at least one <u>observed dataset</u> and
one <u>header file</u> associated to a (already generated or not) <u>training
set</u>. The header file, always named </span></span><span style='mso-bookmark:
doc_newSNPProjectButton'><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt;font-family:Consolas;color:black;mso-themecolor:text1'>headerRF.txt</span><span
lang=EN-US style='color:black;mso-themecolor:text1'>, contains all information
necessary to compute a training set associated with the data: i.e. the
scenarios, the scenario parameter priors, the characteristics of loci, the loci
parameter priors and the list of summary statistics to compute using the <i
style='mso-bidi-font-style:normal'>Training set simulation</i> module. As soon
as the first records of the training set have been simulated, they are saved in
the training set file, always named </span></span><span style='mso-bookmark:
doc_newSNPProjectButton'><span class=SpellE><span lang=EN-US style='font-size:
11.0pt;mso-bidi-font-size:12.0pt;font-family:Consolas;color:black;mso-themecolor:
text1'>reftableRF.bin</span></span><span lang=EN-US style='color:black;
mso-themecolor:text1'>. In addition, the file statobs</span></span><span
style='mso-bookmark:doc_newSNPProjectButton'><span lang=EN-US style='font-size:
11.0pt;mso-bidi-font-size:12.0pt;font-family:Consolas;color:black;mso-themecolor:
text1'>RF.txt</span><span lang=EN-US style='color:black;mso-themecolor:text1'>,
stores the value of the summary statistics computed from the observed dataset.<o:p></o:p></span></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span style='mso-bookmark:doc_newSNPProjectButton'><span
lang=EN-US style='color:black;mso-themecolor:text1'>Outputs of Random forest
analyses are available in sub-directories (each one corresponding to a specific
setting that you may have configured for the random forest analyses) in the
project directory (please see the corresponding section 7.3 for more details
regarding the content of outputs generated by a Random Forest analysis).<o:p></o:p></span></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span style='mso-bookmark:doc_newSNPProjectButton'><span
lang=EN-US style='color:black;mso-themecolor:text1'>It is worth stressing that
you need to SAVE a project to avoid losing your work before quitting the GUI. A
project zip file will be generated and can be saved. If you want to later
continue with a project, or clone/modify an existing project, you can provide
an existing project zip file in the GUI when starting a project.<o:p></o:p></span></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span style='mso-bookmark:doc_newSNPProjectButton'><span
lang=EN-US style='color:black;mso-themecolor:text1'>Finally, we strongly
advised NOT to manually edit project-related files, unless you know what you
are doing!<o:p></o:p></span></span></p>

<h2><span style='mso-bookmark:doc_newSNPProjectButton'><a name="_Toc62127963"><span
lang=EN-US>5.3 Main options of the home screen</span></a></span></h2>

<p class=FirstParagraph><span lang=EN-US>When launching the GUI, the home
screen appears like this:</span></p>

<p class=FirstParagraph><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_25" o:spid="_x0000_i1061"
 type="#_x0000_t75" style='width:574.8pt;height:323.4pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image075.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=766 height=431
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image076.jpg" v:shapes="Image_x0020_25"><![endif]></span></p>

<p class=FirstParagraph style='margin-left:36.0pt;text-indent:-18.0pt;
mso-list:l15 level1 lfo18'><![if !supportLists]><span lang=EN-US
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Access to main functionalities:
you can access to the two main functionalities of the program (i.e. training
set simulation and random forest analyses) in two ways: (<span class=SpellE>i</span>)
by clicking on the &#8220;Start&#8221; button in the middle of the Home panel
or (ii) by clicking on the &#8220;DIYABC-RF main pipeline&#8221; button in the
upper-left part of the panel.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l15 level1 lfo18'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Access to other
functionalities: </span></p>

<p class=MsoListParagraph style='text-indent:-18.0pt;mso-list:l43 level1 lfo20'><![if !supportLists]><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-fareast-font-family:
Calibri'><span style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>To access to the panels
allowing generating files corresponding to full in silico datasets (also called
pseudo-observed datasets) click on the &#8220;Synthetic data file
generation&#8221; button in the middle of the panel or on the &#8220;Synthetic
data file generation&#8221; button in the upper-left part of the panel.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l43 level1 lfo20'><![if !supportLists]><span lang=EN-US style='font-family:
"Calibri",sans-serif;mso-fareast-font-family:Calibri'><span style='mso-list:
Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Clicking on the
&#8220;Preference&#8221; button (upper left) gives access to the following
screen</span></p>

<p class=MsoBodyText style='margin-left:36.0pt'><span style='mso-ansi-language:
FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_29"
 o:spid="_x0000_i1060" type="#_x0000_t75" style='width:554.4pt;height:312pt;
 visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image077.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=739 height=416
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image078.jpg" v:shapes="Image_x0020_29"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt'><span lang=EN-US>The different
options proposed are self-meaning. Default values can be changed by the user.
Note that the loop-size option corresponds to the number of simulated datasets
distributed over all computer threads and stored in RAM before writing them
into the training set file (<span class=SpellE>reftableRF.bin</span>).</span></p>

<h2><a name="_Toc62127964"><span lang=EN-US>5.4 How to generate a new </span></a><span
class=SpellE><span style='mso-bookmark:_Toc62127964'><span lang=EN-US>IndSeq</span></span></span><span
style='mso-bookmark:_Toc62127964'><span lang=EN-US> SNP training set</span></span></h2>

<h2><a name="_Toc62127965"></a><a name=defining-a-new-project><span
style='mso-bookmark:_Toc62127965'><span lang=EN-US style='font-size:12.0pt'>5.4.1
Step 1: defining a new </span></span></a><span class=SpellE><span
style='mso-bookmark:defining-a-new-project'><span style='mso-bookmark:_Toc62127965'><span
lang=EN-US style='font-size:12.0pt'>IndSeq</span></span></span></span><span
style='mso-bookmark:defining-a-new-project'><span style='mso-bookmark:_Toc62127965'><span
lang=EN-US style='font-size:12.0pt'> SNP project</span></span></span><span
lang=EN-US style='font-size:12.0pt'><o:p></o:p></span></h2>

<p class=FirstParagraph><span lang=EN-US>Defining a new project requires
different steps which are not strictly the same whether the data are
microsatellites/DNA sequences (MSS) or SNP (<span class=SpellE>IndSeq</span> or
<span class=SpellE>PoolSeq</span>). Let&#8217;s start with an <span
class=SpellE>IndSeq</span> SNP project. Click on the &#8220;Start&#8221; button
in the middle of the Home panel (or on the &#8220;DIYABC-RF main
pipeline&#8221; button). The following panel appears.</span></p>

<p class=FirstParagraph><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_32" o:spid="_x0000_i1059"
 type="#_x0000_t75" style='width:555.6pt;height:312.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image079.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image080.jpg" v:shapes="Image_x0020_32"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l6 level1 lfo32'><![if !supportLists]><span lang=EN-US style='font-family:Symbol;
mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Enter and validate a project
name in the &#8220;Project name&#8221; window (here <span class=SpellE>IndSeq_demonstration_project</span>)</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l6 level1 lfo32'><![if !supportLists]><span lang=EN-US style='font-family:Symbol;
mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Select the Data type including
the locus type (<span class=SpellE>Microsat</span>/sequence or SNP) and the
sequencing mode (for SNP only; <span class=SpellE>IndSeq</span> or <span
class=SpellE>PoolSeq</span>). Here select SNP and Individual Seq.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l6 level1 lfo32'><![if !supportLists]><span lang=EN-US style='font-family:Symbol;
mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Select the &#8220;New
project&#8221; button as we want to implement a new project from scratch.</span></p>

<h2><a name="_Toc62127966"><span lang=EN-US style='font-size:12.0pt'>5.4.2 Step
2: choosing the data file</span></a><span lang=EN-US style='font-size:12.0pt'><o:p></o:p></span></h2>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l27 level1 lfo26'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Choose a data file (an <span
class=SpellE>IndSeq</span> format data file in this case) using the
&#8220;Select file&#8221; browser (here INDSNP_sim_dataset_4POP_001.snp). A
large size dataset might take some times to be uploaded and checked by the
program. A short summary of the specificities of the loaded datafile appears:
here the total number of loci (30000), the minimum allele frequency chosen for
simulation (MAF=0.05), the sex ratio indicated by the user (NM=1NF), the total
number of individuals and populations, the locus type and their corresponding
numbers (A= autosomal, M=mitochondrial&#8230;; see section 7.1 for details
about datafile format), and the number of <span class=help-block>available loci
based on the MAF criterion (here 15617 &lt;A&gt;, as14383 loci, including 0
monomorphic loci, have been filtered out by the program after scanning the
proposed observed dataset</span>).</span></p>

<p class=MsoBodyText style='margin-left:18.0pt'><span style='mso-ansi-language:
FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_33"
 o:spid="_x0000_i1058" type="#_x0000_t75" style='width:555.6pt;height:312.6pt;
 visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image081.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image082.jpg" v:shapes="Image_x0020_33"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l27 level1 lfo26'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>A &#8220;Project set up is
ok&#8221; message appears at the bottom of the panel if all items have gone
correctly. You can then go the next steps by clicking on the large blue color
&#8220;Training set simulation&#8221; &#8220;+&#8221; button&#8221;. The
following panel appears.</span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_37" o:spid="_x0000_i1057"
 type="#_x0000_t75" style='width:555.6pt;height:312.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image083.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image084.jpg" v:shapes="Image_x0020_37"><![endif]></span></p>

<h2><a name="_Toc62127967"><span lang=EN-US style='font-size:12.0pt'>5.4.3 Step
3: Inform the Historical model</span></a><a name=inform-the-historical-model></a><span
lang=EN-US style='font-size:12.0pt'><o:p></o:p></span></h2>

<p class=FirstParagraph><span lang=EN-US>Click on the &#8220;Add&#8221; button
of the &#8220;Define historical models&#8221; section. The following panel
appears:</span></p>

<p class=FirstParagraph><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_38" o:spid="_x0000_i1056"
 type="#_x0000_t75" style='width:475.2pt;height:267.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image085.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=634 height=357
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image086.jpg" v:shapes="Image_x0020_38"><![endif]></span></p>

<p class=MsoBodyText><span lang=EN-US>Write the code of a first scenario
(scenario 1) in the edit <span class=GramE>window &#8221;Describe</span> your
scenario&#8221; and click on the &#8220;Validate&#8221; button. </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l35 level1 lfo37'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>If we click on the &#8220;Draw
scenario&#8221; button, the logic of the scenario is checked and if it is found
OK, and if the scenario is draw-able, a graphical representation of the
scenario appears. You can save it by clicking on the &#8220;Save image&#8221;
button. Figures are saved in 'fig' sub-folder in the project directory.
Possible extensions are: 'eps', '<span class=SpellE>ps</span>', '<span
class=SpellE>tex</span>', 'pdf', 'jpeg', 'tiff', '<span class=SpellE>png</span>',
'bmp', '<span class=SpellE>svg</span>'.</span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_42" o:spid="_x0000_i1055"
 type="#_x0000_t75" style='width:483.6pt;height:272.4pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image087.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=645 height=363
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image088.jpg" v:shapes="Image_x0020_42"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l35 level1 lfo37'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The &#8220;Priors and
conditions&#8221; frame which appears below allows choosing the prior distribution
of each parameter of the scenario. A parameter is anything in the scenario that
is not a keyword (here <span class=VerbatimChar><span style='mso-ansi-font-size:
12.0pt;font-family:"Cambria",serif;mso-ascii-theme-font:minor-latin;mso-hansi-theme-font:
minor-latin'>sample</span></span>, <span class=VerbatimChar><span
style='mso-ansi-font-size:12.0pt;font-family:"Cambria",serif;mso-ascii-theme-font:
minor-latin;mso-hansi-theme-font:minor-latin'>merge and split</span></span>),
nor a numeric value. In our example scenario, parameters are hence: <span
class=VerbatimChar><span style='mso-ansi-font-size:12.0pt;font-family:"Cambria",serif;
mso-ascii-theme-font:minor-latin;mso-hansi-theme-font:minor-latin'>N1, N2, N3,
N4, ra, t21, t32</span></span> and <span class=VerbatimChar><span
style='mso-ansi-font-size:12.0pt;font-family:"Cambria",serif;mso-ascii-theme-font:
minor-latin;mso-hansi-theme-font:minor-latin'>t423</span></span>. We choose a
uniform distribution and change the min and max values (100 and 10000 for N
values - 10 and 1000 for t values; see panel below). </span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_43" o:spid="_x0000_i1054"
 type="#_x0000_t75" style='width:524.4pt;height:294.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image089.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=699 height=393
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image090.jpg" v:shapes="Image_x0020_43"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l35 level1 lfo37'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US><span
style='mso-spacerun:yes'>&nbsp;</span>In our example, we also need to specify
conditions for the priors on <span class=VerbatimChar><span style='mso-ansi-font-size:
12.0pt;font-family:"Cambria",serif;mso-ascii-theme-font:minor-latin;mso-hansi-theme-font:
minor-latin'>t21, t32 and t432 such that </span></span>t21&gt;t32 and t32&gt;t423.</span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_55" o:spid="_x0000_i1053"
 type="#_x0000_t75" style='width:490.8pt;height:276pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image091.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=654 height=368
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image092.jpg" v:shapes="Image_x0020_55"><![endif]></span></p>

<p class=MsoBodyText><span lang=EN-US>To do this we write t21&gt;t32 and t32&gt;t432
on two successive lines in the &#8220;Condition setting&#8221; frame. As noted
in the panel, conditions should have the following format: XX&lt;YY. where
&#8216;XX' and 'YY' are parameters of the same type. You can use the standard
comparison signs: '&gt;', '&gt;=', '&lt;', '=&lt;'. It is worth stressing that
the omission of such conditional constraints on merge times (cf.&nbsp;a
population needs to exist in the past to allow coalescence events in it) is one
of the most frequent implementation error made by DIYABC-RF users. If forgotten
a message &quot;Error in simulation process: check your scenarios, priors and
conditions&quot;&quot; appears. Note that the occurrence of a too large number
of time conditional constraints within a scenario may substantially slow down
simulations as a valid t parameter vector will be retain and run only once all
conditions are fulfilled.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l16 level1 lfo27'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>We then can <u>add some other
scenarios in additional windows</u> by clicking on the button &#8220;Add&#8221;
(one or several times if one wants to add a single or several other scenarios).
In the present example, we simulate datasets from six different scenarios that
we want to compare (see Collin et al. 2020 example based on pseudo-observed
datasets for details), hence the six scenario windows that we have completed
with instructions for each one. For the sixth scenario the panel looks like
this:</span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_62" o:spid="_x0000_i1052"
 type="#_x0000_t75" style='width:555.6pt;height:312.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image093.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image094.jpg" v:shapes="Image_x0020_62"><![endif]></span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR'>Complete time
conditions for the six scenarios <span class=GramE>are:</span> t21&gt;t32,
t42&lt;t21, t43&lt;t32, t32&gt;t423, t431&lt;t32 and t421&lt;t21<o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:10.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:2'><a name="_Toc62127968"><b><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'>5.4.4 Step 4: Inform number of loci to simulate (and
chromosome type)</span></b></a><b><span lang=EN-US style='font-family:"Calibri",sans-serif;
mso-ascii-theme-font:major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'><o:p></o:p></span></b></p>

<p class=FirstParagraph><span lang=EN-US>Choose the number of SNP locus that you
want to simulate for each chromosome type define in the data file (<i
style='mso-bidi-font-style:normal'>A</i> for autosomal diploid loci, <i
style='mso-bidi-font-style:normal'>H</i> for autosomal haploid loci, <i
style='mso-bidi-font-style:normal'>X</i> for X-linked or haplo-diploid loci,<i
style='mso-bidi-font-style:normal'> Y</i> for Y-linked loci and <i
style='mso-bidi-font-style:normal'>M</i> for mitochondrial loci; see section
7.1.1). Here we have only type &lt;A&gt; SNP loci. According to our own
experience, analyzing (evolutionary neutral) scenarios using 5000 to 20000 SNP
loci is sufficient to obtain robust results. In this example, we choose to
consider simulations based on a subset of 5000 SNP loci (with MAF=0.05 as
indicated previously) taken in order from the <u>first</u> SNP locus of the
data file, by replacing 30000 by 5000 in the corresponding frame. Click on the
button &#8220;Validate&#8221;. </span></p>

<p class=FirstParagraph><span lang=EN-US>It is worth stressing that choosing a
subset of 5000 SNP loci taken from the SNP locus 5001 of the data file would
lead to a training set generated from a set of 5000 loci completely independent
from the previous one. Generating by this way different training sets is a way
to process independent analyses of a given dataset (a sufficiently large number
of SNP loci available in the observed dataset is obviously needed to do that).</span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_68" o:spid="_x0000_i1051"
 type="#_x0000_t75" style='width:531.6pt;height:298.8pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image095.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=709 height=398
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image096.jpg" v:shapes="Image_x0020_68"><![endif]></span></p>

<p class=MsoNormal style='margin-top:10.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:2'><a name="_Toc62127969"><b><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'>5.4.5 Step 5: Summary statistics</span></b></a><b><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'><o:p></o:p></span></b></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US>In the present
version of the program, all available summary statistics are computed as
mentioned in the interface below the &#8220;Summary statistics&#8221; item (see
previous panel-copy) and as detailed when clicking on the button &#8220;?&#8221;
on the right:<span style='color:red;background:yellow;mso-highlight:yellow'><o:p></o:p></span></span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>For both <span class=SpellE>IndSeq</span> and <span class=SpellE>PoolSeq</span>
SNP loci, the following set of summary statistics has been implemented.<o:p></o:p></span></b></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.85pt;margin-bottom:.0001pt;text-indent:-33.0pt'><span
lang=EN-US>1. <span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><i
style='mso-bidi-font-style:normal'>Proportion of monomorphic loci</i> for each
population, as well as for each pair and triplet of populations (ML1p, ML2p,
ML3p)</span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>Mean and variance (over loci) values are computed for all subsequent
summary statistics. <o:p></o:p></span></b></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.85pt;margin-bottom:.0001pt;text-indent:-33.0pt'><span
lang=EN-US>2. <span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><i
style='mso-bidi-font-style:normal'>Heterozygosity</i> for each population (HW)
and for each pair of populations (HB)</span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.85pt;margin-bottom:.0001pt;text-indent:-33.0pt'><span
lang=EN-US>3.<span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><i
style='mso-bidi-font-style:normal'>FST</i>-<i style='mso-bidi-font-style:normal'>related
statistics</i> for each population (FST1), for each pair (FST2), triplet
(FST3), quadruplet (FST4) and overall (FSTG) populations (when the dataset
includes more than four populations)</span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.85pt;margin-bottom:.0001pt;text-indent:-33.0pt'><span
lang=EN-US>4. <span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><i
style='mso-bidi-font-style:normal'>Patterson&#8217;s f-statistics</i> for each
triplet (f3-statistics; F3) and quadruplet (f4-statistics; F4) of populations </span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US>5. <span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><span
class=SpellE><i style='mso-bidi-font-style:normal'>Nei&#8217;s</i></span><i
style='mso-bidi-font-style:normal'> distance</i> (NEI) for each pair of
populations</span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.85pt;margin-bottom:.0001pt;text-indent:-33.0pt'><span
lang=EN-US>6. <span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><i
style='mso-bidi-font-style:normal'>Maximum likelihood coefficient of admixture</i>
(AML) computed for each triplet of populations.</span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:35.85pt;margin-bottom:.0001pt;text-indent:-33.0pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US>Note the short code
name for each type of summary statistics (MLP1p, HW, FST2, F<span class=GramE>3,&#8230;</span>).
Such code names of statistics (plus the suffix m and v for mean and variance
over loci, respectively; see section 2.6.3) will be used in most files produced
by the program, including the key files headerRF.txt, <span class=SpellE>reftableRF.bin</span>
and statobsRF.txt.</span></p>

<h2><a name="_Toc62127970"></a><a name=building-the-reference-table><span
style='mso-bookmark:_Toc62127970'><span lang=EN-US style='font-size:12.0pt'>5.4.6
Step 6: Simulate the training set</span></span></a><span lang=EN-US
style='font-size:12.0pt'><o:p></o:p></span></h2>

<p class=FirstParagraph><span lang=EN-US>You can access the &#8220;training set
simulation&#8221; module by clicking on the corresponding + button on the far
right of the box blue title bar &#8220;Training set simulation&#8221;. The
following &#8220;Run&#8221; panel appears.</span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_69" o:spid="_x0000_i1050"
 type="#_x0000_t75" style='width:555.6pt;height:312.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image097.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image098.jpg" v:shapes="Image_x0020_69"><![endif]><span
style='mso-spacerun:yes'>&nbsp;</span></span></p>

<p class=FirstParagraph style='margin-left:36.0pt;text-indent:-18.0pt;
mso-list:l20 level1 lfo24'><![if !supportLists]><span lang=EN-US
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Indicate the required number of
datasets to simulate in the training set: we here change the default value of
100 into 12000 (in order to simulate 2000 datasets for each of the six
scenarios)</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l20 level1 lfo24'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Launch the generation of the
simulated datasets of the training set by clicking on the large blue color
button &#8220;Simulate&#8221;. The number of datasets already simulated is
given in the &#8220;Running simulation&#8221; progress bar below.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l32 level1 lfo25'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>You can stop the generation of
simulated datasets by clicking on the &#8220;STOP&#8221; button. Both a &#8220;Run
succeeded&#8221; message appears when all requested simulations have been
processed.</span></p>

<p class=MsoListParagraph style='text-indent:-18.0pt;mso-list:l32 level1 lfo25'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><u><span lang=EN-US>Do not forget to click on
the &#8220;Save&#8221; button in the &#8220;Project administration&#8221; lower
panel to implement and save a concatenated zip file including various input and
output files</span></u><span lang=EN-US> (see section 7 for a description of
the content of some of those files) on your computer. The name of the
implemented/saved concatenated .zip file is the one given at the start in the
&#8220;Project name&#8221; window (here <span class=SpellE>IndSeq_demonstration_project</span>).</span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_70" o:spid="_x0000_i1049"
 type="#_x0000_t75" style='width:555.6pt;height:312.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image099.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image100.jpg" v:shapes="Image_x0020_70"><![endif]></span></p>

<p class=MsoListParagraph style='text-indent:-18.0pt;mso-list:l32 level1 lfo25'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Warning: The button &#8220;Reset&#8221;
at the very bottom part of the &#8220;Project administration&#8221; panel
restart the pipeline from scratch removing all configurations previously set up
and all data and result files generated.</span></p>

<h2><a name="_Toc62127971"><span lang=EN-US style='font-size:12.0pt'>5.4.7 Step
7: Prior-scenario checking PRE-ANALYSIS (optional but recommended)</span></a><span
lang=EN-US style='font-size:12.0pt'><o:p></o:p></span></h2>

<p class=MsoBodyText><span lang=EN-US>Before performing a full ABC-RF analysis
on a large size training set, we advise using the prior-scenario checking
option on a training set of small size (e.g. including 500 to 1000 simulated
dataset per scenarios). As a matter of fact, this pre-analysis option is a
convenient way to reveal potential mis-specification of models (scenarios)
and/or prior distributions of parameters (and correct it). This action requires
a training set file (`<span class=SpellE>reftableRF.bin</span>`) and a
statobsRF.txt file either generated when clicking on 'Simulate' or uploaded
with an existing project. When clicking on the button <u>&#8220;Run&#8221; of
the &#8220;Prior and scenario checking&#8221; option (see previous/above panel
copy)</u> the program will use the statobsRF.txt and <span class=SpellE>reftableRF.bin</span>
files to generate two specific outputs:</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l12 level1 lfo28'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>A Principal Component Analysis
(PCA) representing on 2-axes plans the simulated dataset from the training set
and the observed dataset as graphical output. Note that this figure is somewhat
similar in its spirit to the graphical output
LDA_training_set_and_obs_dataset.png obtained when running the scenario choice
option in &#8220;Random Forest Analyses&#8221;. The latter output is different,
however, as it corresponds to the projection of the datasets from the training
set linear discriminant analysis (LDA) axes (see section 7.3 for details).</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l12 level1 lfo28'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>A numerical output file, named
pcaloc1_locate.txt, which includes for each summary statistics and for each
scenario, the proportion of simulated data (considering the total training set)
that have a summary statistics value below the value of the observed dataset. A
star indicates proportions lower than 5% or greater than 95% (two stars, &lt;1%
or &gt;1%; three stars, &lt;0.1% or &gt;0.1%). The presence of such star
symbols is a sign of substantial mismatch between the observed dataset and the
simulated datasets. The latter numerical results can help users to
reformulating the compared scenarios and/or the associated prior distributions
in order to achieve some compatibility (see e.g. <span class=SpellE>Cornuet</span>
et al. 2010). If only a few stars are observed at a few summary statistics for
one or several scenarios, one can conclude that prior-scenario conditions are
suitable enough to further process Random Forest analysis, and this potentially
on a training set including more simulated datasets.</span></p>

<h2><a name="_Toc62127972"><span lang=EN-US>5.5 How to generate a new </span></a><span
class=SpellE><span style='mso-bookmark:_Toc62127972'><span lang=EN-US>PoolSeq</span></span></span><span
style='mso-bookmark:_Toc62127972'><span lang=EN-US> SNP training set</span></span></h2>

<p class=MsoBodyText><span lang=EN-US>Follow the same five steps described in
section 5.4 for <span class=SpellE>IndSeq</span> SNPs.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l15 level1 lfo18'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Step 1</span></i><span lang=EN-US>: Defining a new <span
class=SpellE>PoolSeq</span> SNP project </span><span lang=EN-US
style='font-family:Wingdings;mso-fareast-font-family:Wingdings;mso-bidi-font-family:
Wingdings'>&agrave;</span><span lang=EN-US> see section 5.4.1 </span></p>

<p class=MsoBodyText><span lang=EN-US>Click on the &#8220;Launch project&#8221;
button of the home screen. Select SNP as project type and <span class=SpellE>PoolSeq</span>
as sequencing mode </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l15 level1 lfo18'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Step 2</span></i><span lang=EN-US>: Choosing the data file </span><span
lang=EN-US style='font-family:Wingdings;mso-fareast-font-family:Wingdings;
mso-bidi-font-family:Wingdings'>&agrave;</span><span lang=EN-US> see section
5.4.2 and select a dataset file characterized by a <span class=SpellE>PoolSeq</span>
format (format detailed in section 7.1.2).</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l15 level1 lfo18'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Step 3</span></i><span lang=EN-US>: Inform the Historical model </span><span
lang=EN-US style='font-family:Wingdings;mso-fareast-font-family:Wingdings;
mso-bidi-font-family:Wingdings'>&agrave;</span><span lang=EN-US> see section
5.4.3 </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l15 level1 lfo18'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Step 4</span></i><span lang=EN-US>: Inform chromosome type and
number of loci </span><span lang=EN-US style='font-family:Wingdings;mso-fareast-font-family:
Wingdings;mso-bidi-font-family:Wingdings'>&agrave;</span><span lang=EN-US> see
section 5.4.4 </span></p>

<p class=MsoBodyText><span lang=EN-US>Note: The panel for <span class=SpellE>PoolSeq</span>
SNPs is simpler than for <span class=SpellE>IndSeq</span> SNPs because <span
class=SpellE>PoolSeq</span> SNPs are considered as located on <b
style='mso-bidi-font-weight:normal'>autosomal chromosomes only</b>.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l44 level1 lfo21'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Step 5</span></i><span lang=EN-US>: Summary statistics: </span><span
lang=EN-US style='font-family:Wingdings;mso-fareast-font-family:Wingdings;
mso-bidi-font-family:Wingdings'>&agrave;</span><span lang=EN-US> see section
5.4.5</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l44 level1 lfo21'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Step 6</span></i><span lang=EN-US>: Simulate the training set </span><span
lang=EN-US style='font-family:Wingdings;mso-fareast-font-family:Wingdings;
mso-bidi-font-family:Wingdings'>&agrave;</span><span lang=EN-US> see section
5.4.6 </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l44 level1 lfo21'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Step 7</span></i><span lang=EN-US> (optional but recommended):
Prior-scenario checking PRE-ANALYSIS </span><span lang=EN-US style='font-family:
Wingdings;mso-fareast-font-family:Wingdings;mso-bidi-font-family:Wingdings'>&agrave;</span><span
lang=EN-US> see section 5.4.7</span></p>

<h2><a name="_Toc62127973"><span lang=EN-US>5.6 How to generate a new
microsatellite (and/or DNA sequence) training set</span></a></h2>

<p class=MsoBodyText><span lang=EN-US>Follow the same steps described in
section 5.4 for <span class=SpellE>IndSeq</span> SNPs.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l15 level1 lfo18'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Step 1</span></i><span lang=EN-US>: Defining a new or
microsatellites/DNA sequences project </span><span lang=EN-US style='font-family:
Wingdings;mso-fareast-font-family:Wingdings;mso-bidi-font-family:Wingdings'>&agrave;</span><span
lang=EN-US> see section 5.4.1 </span></p>

<p class=MsoBodyText style='margin-left:36.0pt'><span lang=EN-US>Click on the
&#8220;Launch project&#8221; button of the home screen. Select <span
class=SpellE>Microsat</span>/Sequences as project type (not any sequencing mode
is needed)</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l15 level1 lfo18'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Step 2</span></i><span lang=EN-US>: Choosing the data file </span><span
lang=EN-US style='font-family:Wingdings;mso-fareast-font-family:Wingdings;
mso-bidi-font-family:Wingdings'>&agrave;</span><span lang=EN-US> see section
5.4.2 and select a dataset file characterized by a Microsatellite and/or DNA
Sequences format (format detailed in section 7.1.3).</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l15 level1 lfo18'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Step 3</span></i><span lang=EN-US>: Inform the Historical model </span><span
lang=EN-US style='font-family:Wingdings;mso-fareast-font-family:Wingdings;
mso-bidi-font-family:Wingdings'>&agrave;</span><span lang=EN-US> see section
5.4.3 </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l15 level1 lfo18'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><i
style='mso-bidi-font-style:normal'><span lang=EN-US>Step 4</span></i><span
lang=EN-US>: The panel &#8220;Inform chromosome type and number of loci&#8221;
is replaced by the panel &#8220;Inform the genetic model&#8221; </span></b><b
style='mso-bidi-font-weight:normal'><span lang=EN-US style='font-family:Wingdings;
mso-fareast-font-family:Wingdings;mso-bidi-font-family:Wingdings'>&agrave;</span><span
lang=EN-US> see details below in section 5.6.1</span></b></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l44 level1 lfo21'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><i
style='mso-bidi-font-style:normal'><span lang=EN-US>Step 5</span></i><span
lang=EN-US>: Summary statistics: different sets of summary statistics are
computed as compared to SNP summary statistics </span></b><b style='mso-bidi-font-weight:
normal'><span lang=EN-US style='font-family:Wingdings;mso-fareast-font-family:
Wingdings;mso-bidi-font-family:Wingdings'>&agrave;</span><span lang=EN-US> see
details below in section 5.6.2<o:p></o:p></span></b></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l44 level1 lfo21'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Step 6</span></i><span lang=EN-US>: Simulate the training set </span><span
lang=EN-US style='font-family:Wingdings;mso-fareast-font-family:Wingdings;
mso-bidi-font-family:Wingdings'>&agrave;</span><span lang=EN-US> see section
5.4.6 </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l44 level1 lfo21'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Step 7</span></i><span lang=EN-US> (optional but recommended):
Prior-scenario checking PRE-ANALYSIS </span><span lang=EN-US style='font-family:
Wingdings;mso-fareast-font-family:Wingdings;mso-bidi-font-family:Wingdings'>&agrave;</span><span
lang=EN-US> see section 5.4.7</span></p>

<h3 style='margin-left:18.0pt'><a name="_Toc62127974"><span lang=EN-US>5.6.1
Step 4: Inform the genetic model</span></a><span lang=EN-US> </span></h3>

<p class=FirstParagraph style='margin-left:36.0pt'><b style='mso-bidi-font-weight:
normal'><u><span lang=EN-US>For microsatellite markers</span></u><span
lang=EN-US>: Informing the genetic model can be done using the following panels
(implemented here from a toy example including microsatellites + DNA sequences named
TOY_EXAMPLE2_microsatellites_DNAsequence_two_pops_ancient_admixture; see
section 10. TOY EXAMPLES for details about this toy example).<o:p></o:p></span></b></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l44 level1 lfo21'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>First:<span
style='mso-spacerun:yes'>&nbsp; </span>define the number of groups of markers
(a group been defined by different mutational modalities). By default, all
microsatellite loci are grouped together, same for all DNA sequence loci. If
you want to define more than a single group then <span class=help-block>configure
the number of groups (<span class=SpellE>cf</span> button &#8220;Add
group&#8221;) and then assign each locus to one of those groups using the Group
arrow on the right.</span></span></p>

<p class=FirstParagraph><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_16" o:spid="_x0000_i1048"
 type="#_x0000_t75" style='width:442.2pt;height:248.4pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image101.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=590 height=331
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image102.jpg" v:shapes="Image_x0020_16"><![endif]></span><span
style='mso-ansi-language:FR;mso-fareast-language:FR'><o:p></o:p></span></p>

<p class=FirstParagraph style='margin-left:36.0pt;text-indent:-18.0pt;
mso-list:l26 level1 lfo35'><![if !supportLists]><span lang=EN-US
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol;mso-fareast-language:FR'><span style='mso-list:Ignore'>&middot;<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span></span><![endif]><span
lang=EN-US style='mso-fareast-language:FR'>Second: define the motif size and
range of motif variation. </span><span class=help-block><span lang=EN-US>By
default, all microsatellite loci are supposed to be <span class=SpellE>dinucleotidic</span>
(motif = 2) with a range of 40.</span></span><span lang=EN-US style='mso-fareast-language:
FR'><o:p></o:p></span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_17" o:spid="_x0000_i1047"
 type="#_x0000_t75" style='width:440.4pt;height:247.8pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image103.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=587 height=330
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image104.jpg" v:shapes="Image_x0020_17"><![endif]></span><span
style='mso-ansi-language:FR;mso-fareast-language:FR'><o:p></o:p></span></p>

<p class=MsoBodyText style='margin-left:36.0pt'><span lang=EN-US>It is worth
stressing that the values for motif size and allelic range are just default
values and do not necessarily correspond to the actual data. The user who knows
the real values for its data is required to set the correct values at this
stage. If the range is too short to include all values observed in the analyzed
dataset, a message appears in a box asking to enlarge the corresponding allelic
range. Note that the allelic range is measured in number of motifs, so that a
range of 40 for a motif length of 2 bp means that the difference between the
smallest and the longest alleles should not exceed 80 bp. It is worth stressing
that the indicated allelic range (expressed in number of continuous allelic
states) corresponds to a potential range which is usually larger than the range
observed from the analyzed dataset (cf.&nbsp;all possible allelic states have usually
not been sampled). In practice it is difficult to assess the actual
microsatellite constraints on the allelic range; to do that one needs allelic
data from several distantly related populations/sub-species as well as related
species which is rarely the case(see Pollack et al. 1998; Estoup et al. 2002).
We achieved a meta-analysis from numerous primer notes documenting the
microsatellite allelic ranges of many (i.e.&nbsp;&gt; 100) different species
(and related species). We used the corrective statistical treatment on such
data proposed by (Pollack et al. 1998). Our results pointed to a mean
microsatellite allelic range of 40 continuous states (hence the default allelic
range value of 40 mentioned in the program). We also found, however, that range
values greatly varied among species and among loci within species (unpublished
results). We therefore recommend the following pragmatic behavior when
considering the allelic range of your <span class=SpellE>analysed</span>
microsatellite dataset: (<span class=SpellE>i</span>) if the difference in
number of motif of your locus is &lt; 40 motifs in the <span class=SpellE>analysed</span>
dataset then leave the default allelic range value of 40. (ii) if the
difference in number of motif of your locus is &gt;40 motifs in your dataset
then take <span class=SpellE>Max_allele_size</span> </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>-</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:9pt;height:22.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image105.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=12 height=30
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image106.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>Min_allele_size)/motif
size + say 10 additi<span class=SpellE>onal</span> motifs to re-define the
allelic range of the locus in the corresponding DIYABC panel (e.g.&nbsp;(200 nu
</span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:normal'><span
 lang=EN-US style='font-family:"Cambria Math",serif'><m:r>-</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:9pt;height:22.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image105.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=12 height=30
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image106.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>100 <span class=GramE>nu</span>)/2
+ 10 = 50 + 10 = 60 as allelic range).</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l26 level1 lfo35'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Third: adjust the prior
specificities of mutational modalities as you want using the corresponding buttons
in the panel below. Otherwise keep the default values as the latter have been
reported for many eukaryote species (e.g. Estoup et al. 2002; but see <span
class=SpellE>Fraimout</span> et al. 2017).</span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_19" o:spid="_x0000_i1046"
 type="#_x0000_t75" style='width:555.6pt;height:312.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image107.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image108.jpg" v:shapes="Image_x0020_19"><![endif]></span></p>

<p class=FirstParagraph><b style='mso-bidi-font-weight:normal'><u><span
lang=EN-US><o:p><span style='text-decoration:none'>&nbsp;</span></o:p></span></u></b></p>

<p class=FirstParagraph><b style='mso-bidi-font-weight:normal'><u><span
lang=EN-US>For DNA sequence markers</span></u><span lang=EN-US>: Informing the
genetic model can be done using the following panels (implemented here from the
same toy example as above). <o:p></o:p></span></b></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l44 level1 lfo21'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>We need to define at least one
group of DNA sequence loci. </span></p>

<p class=MsoBodyText style='margin-left:36.0pt'><span style='mso-ansi-language:
FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_18"
 o:spid="_x0000_i1045" type="#_x0000_t75" style='width:555.6pt;height:312.6pt;
 visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image109.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image110.jpg" v:shapes="Image_x0020_18"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l44 level1 lfo21'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Suppose that we want all DNA
sequence loci in the same group because we consider that they all have similar
mutational modalities. We then put &#8220;2&#8221; in the button
&#8220;Group&#8221; for all loci (Group 1 is for the previous microsatellite
markers). Otherwise you can associate some loci to additional groups (after
clicking on the button &#8220;Add group&#8221;) if you consider that such loci
have different mutational modalities (for instance here Locus_S_M_16_).</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l44 level1 lfo21'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Adjust the prior specificities
of mutational modalities (for the different DNA sequence groups you have
defined previously) using the corresponding buttons in the panel below.
Otherwise keep the default values as the latter have been reported for many
eukaryote species (e.g. <span class=SpellE>Cornuet</span> et al. 2010).</span></p>

<p class=MsoBodyText style='margin-left:18.0pt'><span style='mso-ansi-language:
FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_21"
 o:spid="_x0000_i1044" type="#_x0000_t75" style='width:510pt;height:286.8pt;
 visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image111.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=680 height=382
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image112.jpg" v:shapes="Image_x0020_21"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt'><span lang=EN-US>The default
values are indicated in the panel. Note that the default mean mutation rate is
not suited to mitochondrial DNA which generally evolves at a faster rate than
nuclear DNA (Haag-<span class=SpellE>Liautard</span> et al., 2008). <span
class=GramE>So</span> we set its value to 10<sup>-8</sup>. For all other
mutation model parameters, one can just keep the default values. <u>See section
2.4.2 &#8220;DNA sequence loci&#8221; for details about the mutation models
proposed for DNA sequences</u>.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l44 level1 lfo21'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol;mso-fareast-language:
FR'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US style='mso-fareast-language:
FR'>Before launching a run to simulate the training set, do not forget to <u>validate
your microsatellite and/or DNA settings by clicking on the blue-color button
&#8220;Validate&#8221;</u>. Indicate the number of simulations in the
corresponding button (here 30000) and click on the button
&#8220;Simulate&#8221; to launch the simulation of the training set.<o:p></o:p></span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_22" o:spid="_x0000_i1043"
 type="#_x0000_t75" style='width:457.8pt;height:256.8pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image113.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=610 height=342
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image114.jpg" v:shapes="Image_x0020_22"><![endif]></span></p>

<p class=MsoBodyText><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<h3><a name="_Toc62127975"><span lang=EN-US>5.6.2 Step 5: Summary statistics</span></a></h3>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US>In the present
version of the program, <u>all</u> available summary statistics are computed as
mentioned in the interface below the &#8220;Summary statistics&#8221; item (see
previous panel-copy) and as detailed when clicking on the button &#8220;?&#8221;
on the right:<span style='color:black;mso-themecolor:text1;background:yellow;
mso-highlight:yellow'><o:p></o:p></span></span></p>

<p class=MsoBodyText><b style='mso-bidi-font-weight:normal'><u><span
lang=EN-US>For microsatellite loci<o:p></o:p></span></u></b></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US>The following set of
summary statistics has been implemented. </span></p>

<p class=DefinitionTerm><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US>Single sample statistics:</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>1. mean number of alleles across loci (NAL)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>2. mean gene diversity across loci (HET)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>3. mean allele size variance across loci (VAR)</span></p>

<p class=MsoNormal><span lang=EN-US>3. mean M index across loci (MGW)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US>Two sample statistics:</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>1. mean number of alleles across loci (two samples) (N2P)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>2. mean gene diversity across loci (two samples) (H2P)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>3. mean allele size variance across loci (two samples) (V2P)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><!--[if gte msEquation 12]><m:oMath><m:sSub><m:sSubPr><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>3.</m:r><m:r>F</m:r></span></i></m:e><m:sub><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>ST</m:r></span></i></m:sub></m:sSub></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:27.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image115.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=37 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image116.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>between two samples
(FST)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>4. mean index of classification (two samples) (LIK)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>5. shared allele distance between two samples (DAS)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US style='mso-fareast-font-family:Cambria;mso-fareast-theme-font:minor-fareast'>6.</span><span
lang=EN-US> distance between two samples (DM2)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US>Three sample statistics:</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>1. Maximum likelihood coefficient of admixture (AML)</span></p>

<p class=MsoBodyText><b style='mso-bidi-font-weight:normal'><u><span
lang=EN-US><o:p><span style='text-decoration:none'>&nbsp;</span></o:p></span></u></b></p>

<p class=MsoBodyText><b style='mso-bidi-font-weight:normal'><u><span
lang=EN-US>For DNA sequence loci<o:p></o:p></span></u></b></p>

<p class=MsoBodyText style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:2.85pt;margin-bottom:.0001pt'><span lang=EN-US>The following set of
summary statistics has been implemented. </span></p>

<p class=MsoBodyText style='margin:0cm;margin-bottom:.0001pt'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US>Single sample statistics:</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>1. number of distinct haplotypes (NHA)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>2. number of segregating sites (NSS)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>3. mean pairwise difference (MPD)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>4. variance of the number of pairwise differences (VPD)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>5. Tajima&#8217;s D statistics (DTA)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>6. Number of private segregating sites (PSS)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>7. Mean of the numbers of the rarest nucleotide at segregating sites
(MNS)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>8. Variance of the numbers of the rarest nucleotide at segregating
sites (VNS)</span></p>

<p class=DefinitionTerm><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm><span lang=EN-US>Two sample statistics:</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>1. number of distinct haplotypes in the pooled sample (NH2)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>2. number of segregating sites in the pooled sample (NS2)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>3. mean of within sample pairwise differences (MP2)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US>4. mean of between sample pairwise differences (MPB)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US style='mso-fareast-font-family:Cambria;mso-fareast-theme-font:minor-fareast'><span
style='mso-spacerun:yes'>&nbsp;</span></span><!--[if gte msEquation 12]><m:oMath><m:sSub><m:sSubPr><span
   style='font-family:"Cambria Math",serif;mso-ascii-font-family:"Cambria Math";
   mso-hansi-font-family:"Cambria Math"'><m:ctrlPr></m:ctrlPr></span></m:sSubPr><m:e><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>5.</m:r><m:r>F</m:r></span></i></m:e><m:sub><i
   style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>ST</m:r></span></i></m:sub></m:sSub></m:oMath><![endif]--><![if !msEquation]><span
style='font-size:11.0pt;line-height:107%;font-family:"Calibri",sans-serif;
mso-ascii-theme-font:minor-latin;mso-fareast-font-family:"Times New Roman";
mso-fareast-theme-font:minor-fareast;mso-hansi-theme-font:minor-latin;
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
position:relative;top:2.5pt;mso-text-raise:-2.5pt;mso-ansi-language:FR;
mso-fareast-language:FR;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:27.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image117.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=37 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image118.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>between two samples
(HST)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=DefinitionTerm style='margin-left:2.85pt'><span lang=EN-US>Three
sample statistics:</span></p>

<p class=Definition style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:60.0pt;margin-bottom:.0001pt;text-indent:-24.0pt;mso-list:l9 level2 lfo10;
tab-stops:list 36.0pt'><![if !supportLists]><span lang=EN-US style='mso-bidi-font-family:
Cambria;mso-bidi-theme-font:minor-latin'><span style='mso-list:Ignore'>1.<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Maximum likelihood coefficient
of admixture (SML)</span></p>

<p class=Definition style='margin-bottom:0cm;margin-bottom:.0001pt'><span
lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-top:10.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:2'><a name="_Toc62127976"><b><span
lang=EN-US style='font-size:14.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'>5.7 How to work from an &#8220;Existing project&#8221;
(for any type of markers)</span></b></a><b><span lang=EN-US style='font-size:
14.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'><o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US>Working from an existing project is
particularly appealing when that one wants to:</span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US>(<span class=SpellE>i</span>) <b
style='mso-bidi-font-weight:normal'>objective 1</b>: simply add more
simulations in the training set of the <span class=SpellE>reftableRF.bin</span>
of an existing project without changing anything in the headerRF.txt. </span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US><span
style='mso-spacerun:yes'>&nbsp;</span>(ii) <b style='mso-bidi-font-weight:normal'>objective
2</b>: use the headerRF.txt content of an existing project as a base to
generate a new headerRF.txt file including different changes for instance in
the scenarios formulated, in the definition of priors, the number of loci to
simulate&#8230;</span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US>Let&#8217;s consider that one wants to work
from the <span class=SpellE>IndSeq_demonstration_project</span> previously
generated (and including a training set with 12000 simulation records) as
described in the above section 5.4.</span></p>

<p class=MsoListParagraph style='margin-top:9.0pt;margin-right:0cm;margin-bottom:
9.0pt;margin-left:36.0pt;mso-add-space:auto;text-indent:-18.0pt;mso-list:l11 level1 lfo34'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Click on the &#8220;Start&#8221;
button in the middle of the Home panel. The following panel appears.</span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span style='mso-ansi-language:FR;mso-fareast-language:FR;
mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_23" o:spid="_x0000_i1042"
 type="#_x0000_t75" style='width:555.6pt;height:312.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image119.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image120.jpg" v:shapes="Image_x0020_23"><![endif]></span></p>

<ul style='margin-top:0cm' type=disc>
 <li class=MsoNormal style='margin-top:9.0pt;margin-bottom:9.0pt;mso-list:l6 level1 lfo32'><span
     lang=EN-US>Enter a project name in the &#8220;Project name&#8221; window
     (here &#8220;<span class=SpellE>Starting_from_existing_project_IndSeq_demonstration_project</span>&#8221;)
     and validate the project name.</span></li>
 <li class=MsoNormal style='margin-top:9.0pt;margin-bottom:9.0pt;mso-list:l6 level1 lfo32'><span
     lang=EN-US>Select the project type (<span class=SpellE>Microsat</span>/sequence
     or SNP) and the sequencing mode (for SNP only; <span class=SpellE>IndSeq</span>
     or <span class=SpellE>PoolSeq</span>). As we will start from a
     pre-existing <span class=SpellE>IndSeq</span> project, select SNP and
     Individual Seq.</span></li>
 <li class=MsoNormal style='margin-top:9.0pt;margin-bottom:9.0pt;mso-list:l6 level1 lfo32'><span
     lang=EN-US>Click on the &#8220;Existing project&#8221; button of <span
     class=GramE>the &#8221;Project</span> type&#8221; item as we want to work
     from an existing project.</span></li>
</ul>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l6 level1 lfo32'><![if !supportLists]><span lang=EN-US style='font-family:Symbol;
mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol;color:black;
mso-themecolor:text1'><span style='mso-list:Ignore'>&middot;<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span></span><![endif]><span
lang=EN-US style='color:black;mso-themecolor:text1'>Use the browser of <span
class=GramE>the &#8221;Project</span> files&#8221; item to select key files in
the Existing project directory (here a directory named &#8220;<span
class=SpellE>IndSeq_demonstration_project</span>&#8221;). Key files are <u>headerRF.txt</u>
(mandatory), <span class=SpellE><u>reftableRF.bin</u></span> (if one simply
wants to add more simulations in the training set of the <span class=SpellE>reftableRF.bin</span>
of an existing project without changing anything in the headerRF.txt cf.
objective 1 above), and <u>statobsRF.txt</u> (optional). If the objective 2 is
concerned then only the file headerRF.txt is needed (avoid loading the <span
class=SpellE>reftableRF.bin</span> file as the latter might not <span
class=GramE>be in agreement</span> with the new headerRF.txt file that you will
generate). </span><span lang=EN-US style='font-family:"Calibri",sans-serif;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin;mso-bidi-font-family:
Cambria;mso-bidi-theme-font:minor-bidi;color:black;mso-themecolor:text1'><o:p></o:p></span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_31" o:spid="_x0000_i1041"
 type="#_x0000_t75" style='width:555.6pt;height:312.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image121.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image122.jpg" v:shapes="Image_x0020_31"><![endif]></span><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:Cambria;mso-bidi-theme-font:
minor-bidi;color:black;mso-themecolor:text1'><o:p></o:p></span></p>

<ul style='margin-top:0cm' type=disc>
 <li class=MsoNormal style='margin-top:9.0pt;margin-bottom:9.0pt;mso-list:l27 level1 lfo26'><span
     lang=EN-US>Browse and choose a data file (an <span class=SpellE>IndSeq</span>
     format data file in this case) in the &#8220;Data file&#8221; window (here
     INDSNP_sim_dataset_4POP_001.snp). A short summary of the specificities of
     the loaded datafile appears: here the total number of loci (30000), the
     minimum allele frequency chosen for simulation (MAF=0.05), the sex ratio
     indicated by the user (NM=1NF), the total number of individuals and
     populations, and finally the locus type and their corresponding numbers
     (A= autosomal, M=<span class=GramE>mitochondrial,&#8230;</span>; see
     section 7.1 for details about data file format).</span></li>
 <li class=MsoNormal style='margin-top:9.0pt;margin-bottom:9.0pt;mso-list:l27 level1 lfo26'><span
     lang=EN-US>A &#8220;Project set up is ok&#8221; message appears at the
     bottom of the panel if all items have gone correctly. You can then go the
     next steps by clicking on the large blue color &#8220;Training set
     simulation&#8221; &#8220;+&#8221; button&#8221;. The following panel
     appears.</span></li>
</ul>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:18.0pt'><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_40" o:spid="_x0000_i1040"
 type="#_x0000_t75" style='width:483.6pt;height:272.4pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image123.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=645 height=363
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image124.jpg" v:shapes="Image_x0020_40"><![endif]></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:36.0pt'><span lang=EN-US>The panel displays all information
included in the headerRF.txt file (historical models, priors and conditions,
number of SNP loci to simulate). Warning: such information cannot be changed. </span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:36.0pt'><span lang=EN-US>To change information in the headerRF.txt
(cf. <b style='mso-bidi-font-weight:normal'>objective 2</b>) click on the
&#8220;Edit configuration&#8221; button located at the top of the panel and changes
for instance the scenarios formulated, the definition of priors, the number of
loci to simulate etc. In this case a new <span class=SpellE>reftableRF.bin</span>
file will be generated from scratch following the instruction included in the
newly generated headerRF.txt file. </span></p>

<p class=MsoListParagraphCxSpFirst style='margin-top:9.0pt;margin-right:0cm;
margin-bottom:9.0pt;margin-left:36.0pt;mso-add-space:auto;text-indent:-18.0pt;
mso-list:l27 level1 lfo26'><![if !supportLists]><span lang=EN-US
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Following <b style='mso-bidi-font-weight:
normal'>objective 1</b>, you can launch new simulations that will be added to
those already present in the pre-existing training set (i.e. <span
class=SpellE>reftableRF.bin</span> file). As we have already 12000 simulations
in the initial <span class=SpellE>reftableRF.bin</span> file we put the number
of <span class=GramE>simulation</span> to a number &gt; 12000 (here 24000).
Then click on the large blue color button &#8220;Simulate&#8221;.</span></p>

<p class=MsoListParagraphCxSpMiddle style='margin-top:9.0pt;margin-right:0cm;
margin-bottom:9.0pt;margin-left:36.0pt;mso-add-space:auto'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoListParagraphCxSpLast style='margin-top:9.0pt;margin-right:0cm;
margin-bottom:9.0pt;margin-left:36.0pt;mso-add-space:auto'><span
style='mso-ansi-language:FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape
 id="Image_x0020_41" o:spid="_x0000_i1039" type="#_x0000_t75" style='width:435pt;
 height:244.8pt;visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image125.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=580 height=326
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image126.jpg" v:shapes="Image_x0020_41"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt'><span lang=EN-US>You can stop
the generation of simulated datasets by clicking on the &#8220;STOP&#8221;
button. A &#8220;<span class=help-block>Run succeeded</span>&#8221; message
appears when all requested simulations have been processed.</span></p>

<p class=MsoListParagraph style='text-indent:-18.0pt;mso-list:l32 level1 lfo25'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><u><span lang=EN-US>Do not forget to click on
the &#8220;Save&#8221; button in the &#8220;Project administration&#8221; lower
panel to implement and save a concatenated zip file including various input and
output files</span></u><span lang=EN-US> (see section 7 for a description of
the content of some of those files) on your computer. The name of the
implemented/saved concatenated .zip file is the one given at the start in the
&#8220;Project name&#8221; window (here <span class=SpellE>Starting_from_existing_project_IndSeq_demonstration_project</span>).</span></p>

<p class=MsoNormal style='margin-left:18.0pt'><span style='mso-ansi-language:
FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_44"
 o:spid="_x0000_i1038" type="#_x0000_t75" style='width:501pt;height:282pt;
 visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image127.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=668 height=376
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image128.jpg" v:shapes="Image_x0020_44"><![endif]></span></p>

<p class=MsoListParagraphCxSpFirst style='text-indent:-18.0pt;mso-list:l34 level1 lfo38'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>You might also want to launch
the optional but recommended option &#8220;Prior and scenario checking&#8221;.
For details see section 5.4.7.</span></p>

<p class=MsoListParagraphCxSpLast style='text-indent:-18.0pt;mso-list:l34 level1 lfo38'><a
name="_Toc62127977"><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Warning: The button
&#8220;Reset&#8221; at the very bottom part of the &#8220;Project
administration&#8221; panel restart the pipeline from scratch removing all
configurations previously set up and all data and result files generated.</span></a></p>

<h1><span style='mso-bookmark:_Toc62127977'><span lang=EN-US style='font-family:
"Cambria",serif;mso-ascii-theme-font:minor-latin;mso-hansi-theme-font:minor-latin'>6.
PERFORMING RANDOM FOREST ANALYSES</span></span><span lang=EN-US
style='font-family:"Cambria",serif;mso-ascii-theme-font:minor-latin;mso-hansi-theme-font:
minor-latin'><o:p></o:p></span></h1>

<h2><a name="_Toc62127978"><span lang=EN-US>6.1 Files needed</span></a></h2>

<p class=MsoBodyText><span lang=EN-US>Four files are needed at this stage: </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l28 level1 lfo22'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>the file <b style='mso-bidi-font-weight:
normal'>headerRF.txt</b> which describes the full settings of the generation of
the training set</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l28 level1 lfo22'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>the training set file <span
class=SpellE><b style='mso-bidi-font-weight:normal'>reftableRF.bin</b></span>
(generated from the headerRF.txt file), </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l28 level1 lfo22'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol;color:black;
mso-themecolor:text1'><span style='mso-list:Ignore'>&middot;<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span></span><![endif]><span
lang=EN-US>the file <b style='mso-bidi-font-weight:normal'>statobsRF.txt</b>
which corresponds to the values of the summary statistics computed for the </span><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:Cambria;mso-bidi-theme-font:
minor-bidi;color:black;mso-themecolor:text1'>observed dataset and which was
generated when producing the headerRF.txt and <span class=SpellE>reftableRF.bin</span>
files.<o:p></o:p></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l28 level1 lfo22'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><u><span lang=EN-US>Warning</span></u><span
lang=EN-US>: although not necessary <i style='mso-bidi-font-style:normal'>per
se</i> for Random Forest analyses as the later only need the statobsRF.txt file
as data point to analyze, the <b style='mso-bidi-font-weight:normal'>observed (here
<span class=SpellE>IndSeq</span> SNP) data file</b> has to be present.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l28 level1 lfo22'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol;color:black;
mso-themecolor:text1'><span style='mso-list:Ignore'>&middot;<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span></span><![endif]><b
style='mso-bidi-font-weight:normal'><span lang=EN-US style='color:black;
mso-themecolor:text1'>It is worth stressing that these files are: (<span
class=SpellE>i</span>) already available following an in-course (and
terminated) training set simulation or (ii) they can be selected and loaded
independently by clicking on the &#8220;Existing project&#8221; button and
using the &#8221;Project files&#8221; and &#8220;Data file&#8221; browsers</span></b><span
lang=EN-US style='color:black;mso-themecolor:text1'>.</span><span lang=EN-US
style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:Cambria;mso-bidi-theme-font:
minor-bidi;color:black;mso-themecolor:text1'><o:p></o:p></span></p>

<p class=MsoBodyText><span lang=EN-US style='mso-fareast-language:FR'><span
style='mso-spacerun:yes'>&nbsp;</span></span><span style='mso-ansi-language:
FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_71"
 o:spid="_x0000_i1037" type="#_x0000_t75" style='width:555.6pt;height:312.6pt;
 visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image129.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image130.jpg" v:shapes="Image_x0020_71"><![endif]></span><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:Cambria;mso-bidi-theme-font:
minor-bidi;color:black;mso-themecolor:text1'><o:p></o:p></span></p>

<h2><a name="_Toc62127979"></a><a name=performing-analyses><span
style='mso-bookmark:_Toc62127979'><span lang=EN-US>6.2. Scenario choice
analysis</span></span></a></h2>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Click on the large orange color
&#8220;Random Forest Analyses&#8221; button (&#8220;+&#8221; on the right). </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Click on the button
&#8220;Check input&#8221;: useful information about the training set (cf. <span
class=SpellE>reftableRF.bin</span> file) content is provided (<span
class=SpellE>nbr</span> of scenarios, total <span class=SpellE>nbr</span> of
simulated datasets, <span class=SpellE>nbr</span> of simulated dataset
available per scenario, <span class=SpellE>nbr</span> of parameters, <span
class=SpellE>nbr</span> of summary statistics). </span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_73" o:spid="_x0000_i1036"
 type="#_x0000_t75" style='width:547.2pt;height:307.8pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image131.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=730 height=410
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image132.jpg" v:shapes="Image_x0020_73"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Provide an Analysis
(sub-project) name (here &#8220;<span class=SpellE>Scenario_choice_demonstration</span>&#8221;)
and validate it. </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Activate the &#8220;Scenario
choice&#8221; button in the mode setting (button activated by default). The
observed dataset that will be analyzed is the one described in the file
&#8220;statobsRF.txt&#8221;.</span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_76" o:spid="_x0000_i1035"
 type="#_x0000_t75" style='width:547.2pt;height:307.8pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image133.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=730 height=410
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image134.jpg" v:shapes="Image_x0020_76"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Modify the RF analysis settings
in the following frames or keep default settings</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Indicate the number of samples
(i.e. simulated datasets) in the training set to consider for analyses (default
= 0 = total number of simulated datasets available in the training set). This
number should be lower or equal to the total number of simulated datasets
available in the training set. See section 4.2 for useful insights about this
number.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Indicate the number of noise
variables to add (default = 5). Noise variables (corresponding to values
randomly drawn into uniform distributions bounded between 0 and 1) allows
evaluating which components of the feature vector (i.e. summary statistics) are
informative (see section 7.3 and Collin et al. 2020 for details).</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Enable/disable the addition of
linear combinations axes (LDA for model choice) (default = enable)</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>You may 'group' your models in
several groups. For instance, if you have six models, labeled from 1 to 6, you
can specify '1,2,3;4,5,6' to make 2 groups of 3. Leave blank to not group
scenarios and analyzed them separately (see Collin et al. 2020). (default =
blank). Note that if you write &#8216;1;3;5&#8217; then the RF scenario choice
analysis will concern only the scenarios 1, 3 and 5 of the training set.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Indicate the number of tress in
the forest (default = 500). See section 4.3 for useful insights about this
number.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Click on the button
&#8220;RUN&#8221; to launch the RF analysis for scenario choice </span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_75" o:spid="_x0000_i1034"
 type="#_x0000_t75" style='width:555.6pt;height:312.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image135.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image136.jpg" v:shapes="Image_x0020_75"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Follow the progress bar. You
can stop the RF analysis by clicking on the &#8220;STOP&#8221; button. A
&#8220;RF run succeeded&#8221; label appears when the RF analysis has been
entirely processed.</span></p>

<p class=MsoListParagraphCxSpFirst style='text-indent:-18.0pt;mso-list:l41 level1 lfo23'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Do not forget to click on the
&#8220;Save&#8221; button in the &#8220;Project administration&#8221; panel to
implement and save a concatenated zip file including various input and output
files (see section 7.3 for a description of the content of some of those files)
on your computer. The name of the implemented/saved concatenated zip file is
the one given at the start of the analysis (i.e. Analysis (sub-project) name;
here &#8220;Scenario_choice_demonstration.zip&#8221;).</span></p>

<p class=MsoListParagraphCxSpLast style='text-indent:-18.0pt;mso-list:l41 level1 lfo23'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Warning: The button
&#8220;Reset&#8221; in the &#8220;Project administration&#8221; panel restart
the pipeline from scratch removing all configurations previously set up and all
data and result files generated.</span></p>

<h2><a name="_Toc62127980"><span lang=EN-US>6.3. Parameter estimation analysis</span></a></h2>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Click on the large orange color
&#8220;Random Forest Analyses&#8221; button (&#8220;+&#8221; on the right). </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Activate the &#8220;Parameter
estimation&#8221; button in the mode setting. </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Click on the button
&#8220;Check input&#8221;: Useful information about the training set (cf. <span
class=SpellE>reftableRF.bin</span> file) content is provided (<span
class=SpellE>nbr</span> of scenarios, total <span class=SpellE>nbr</span> of
simulated datasets, <span class=SpellE>nbr</span> of simulated dataset
available per scenario, <span class=SpellE>nbr</span> of parameters, <span
class=SpellE>nbr</span> of summary statistics). </span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Provide an Analysis
(sub-project) name (here &#8220;<span class=SpellE>Parameter_estimation_ra_demonstration</span>&#8221;)
and validate it. </span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_78" o:spid="_x0000_i1033"
 type="#_x0000_t75" style='width:555.6pt;height:312.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image137.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image138.jpg" v:shapes="Image_x0020_78"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Activate the &#8220;Parameter
choice&#8221; button in the mode setting. The observed dataset that will be
analyzed is the one described in the file &#8220;statobsRF.txt&#8221;.</span></p>

<p class=MsoBodyText><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_79" o:spid="_x0000_i1032"
 type="#_x0000_t75" style='width:555.6pt;height:312.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image139.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image140.jpg" v:shapes="Image_x0020_79"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Indicate the Scenario ID number
for parameter estimation (default = 1)</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Indicate the name of the
parameter to estimate (here the admixture rate &#8220;ra&#8221;): a list of
parameter names available for estimation according to the header and training
set files is provided. Give a single name or a combination of names as
explained in the interface. One analysis has to be processed for each parameter
of interest.</span></p>

<p class=MsoListParagraph style='text-indent:-18.0pt;mso-list:l41 level1 lfo23'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Indicate the number of
simulated datasets in the training set to consider for analyses (default = 0 =
total number of simulated datasets available in the training set). <u>Warning: this
number should be lower or equal to the total number of simulated datasets
available in the training set for the scenario of interest.</u> Here we put
this number to 2000 as only 2028 simulated datasets are available for scenario
3 in the <span class=SpellE>reftableRF.bin</span> file previously used for
scenario choice and also loaded in the present illustration. See section 4.2
for useful insights about the number of simulated datasets in the training set
to consider for parameter estimation. In practice, it might be useful to
generate a (second) training set including only the selected scenario to
process parameter estimation from a larger number of simulated datasets (e.g.
10000 as in Collin et al. 2020).</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Indicate the number of noise
variables to add (default = 5). Noise variables (corresponding to values
randomly drawn into uniform distributions bounded between 0 and 1) allows
evaluating which components of the feature vector (i.e. summary statistics) are
informative (see section 7.3 and Collin et al. 2020 for details).</span></p>

<p class=MsoListParagraph><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_81" o:spid="_x0000_i1031"
 type="#_x0000_t75" style='width:555.6pt;height:312.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image141.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image142.jpg" v:shapes="Image_x0020_81"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Enable/Disable the addition of linear
combinations axes (PLS for parameter estimation) (default = enable)</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Indicate the threshold of
percentage of explained variance by the PLS components you will keep in the
analysis (default = 0.95)</span></p>

<p class=MsoListParagraph style='text-indent:-18.0pt;mso-list:l41 level1 lfo23'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Indicate the number of
out-of-bag (<span class=SpellE>oob</span>) testing samples (default = 10 but a
minimum of 1000 <span class=SpellE>oob</span> testing samples is often
necessary to obtained stable estimation of accuracy indices; see section
3.4.2). This number should be equal or lower to the number of datasets
available in the training set for the scenario under study.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Indicate the number of trees in
the forest &#8211; here 1000 - (default = 500). See section 4.3 for useful
insights about this number.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Click on the button &#8220;Run&#8221;
to launch the parameter estimation analyses focusing on the single parameter
(or parameter combination) indicated above.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt'><span style='mso-ansi-language:
FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_82"
 o:spid="_x0000_i1030" type="#_x0000_t75" style='width:555.6pt;height:312.6pt;
 visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image143.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=741 height=417
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image144.jpg" v:shapes="Image_x0020_82"><![endif]></span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l41 level1 lfo23'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Follow the progress bar. You
can stop the RF analysis by clicking on the &#8220;STOP&#8221; button. A
&#8220;RF run succeeded&#8221; label appears when the RF analysis has been
entirely processed.</span></p>

<p class=MsoListParagraphCxSpFirst style='text-indent:-18.0pt;mso-list:l41 level1 lfo23'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Do not forget to click on the
&#8220;Save&#8221; button in the &#8220;Project administration&#8221; panel to
implement and save a concatenated zip file including various input and output
files (see section 7.3 for a description of the content of some of those files)
on your computer. The name of the implemented/saved concatenated zip file is
the one given at the start of the analysis (i.e. Analysis (sub-project) name;
here &#8220;Parameter_estimation_ra_demonstration.zip&#8221;).</span></p>

<p class=MsoListParagraphCxSpLast style='text-indent:-18.0pt;mso-list:l41 level1 lfo23'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Warning: The button
&#8220;Reset&#8221; in the &#8220;Project administration&#8221; panel restart
the pipeline from scratch removing all configurations previously set up and all
data and result files generated.</span></p>

<h1><a name="_Toc62127981"><span lang=EN-US style='font-family:"Cambria",serif;
mso-ascii-theme-font:minor-latin;mso-hansi-theme-font:minor-latin'>7. KEY FILES</span></a><span
lang=EN-US style='font-family:"Cambria",serif;mso-ascii-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin'><o:p></o:p></span></h1>

<p class=FirstParagraph><span lang=EN-US>The program uses and produces various
files which we will describe now.</span></p>

<h2><a name="_Toc62127982"><span lang=EN-US>7.1 Data files</span></a><span
lang=EN-US> </span></h2>

<p class=MsoNormal style='margin-top:10.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:2'><a name="_Toc62127983"><b><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'>7.1.1 </span></b></a><span class=SpellE><span
style='mso-bookmark:_Toc62127983'><b><span lang=EN-US style='font-family:"Calibri",sans-serif;
mso-ascii-theme-font:major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'>IndSeq</span></b></span></span><span
style='mso-bookmark:_Toc62127983'><b><span lang=EN-US style='font-family:"Calibri",sans-serif;
mso-ascii-theme-font:major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'> SNP data</span></b></span><b><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'><o:p></o:p></span></b></p>

<p class=FirstParagraph><span lang=EN-US>The data file format includes:</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>A first line (headline)
providing the sex-ratio as above (e.g. </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>NM</m:r><m:r>=1.0</m:r><m:r>NF</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:79.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image145.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=106 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image146.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>), the required MAF (minimum allele frequency criterion; e.g.
MAF=0.05 or MAF=hudson&gt;), and any text that can be used as a title. The sex
ratio of the analyzed species is noted under the form &lt;NM=<span
class=SpellE>rNF</span>&gt;, in which r is the ratio of the number of females
per male (e.g. &lt;NM=2.5NF&gt; means that the number of males is 2.5 times the
number of females; for a balanced sex ratio one should write &lt;NM=1.0NF&gt;).
The MAF is computed pooling all genes genotyped over all studied population
samples. For instance, the specification of a MAF equal to 5% (i.e. MAF=0.05)
will automatically select a subset of m loci characterized by a minimum allele
frequency 5% among the <i style='mso-bidi-font-style:normal'>l</i> locus of the
observed dataset. In agreement with this, only <i style='mso-bidi-font-style:
normal'>m</i> locus with a MAF&gt;5%. Writing &lt;MAF = Hudson&gt; (or omitting
to write any instruction with respect to the MAF) will bring the program to use
the standard Hudson&#8217;s algorithm without further selection; see also the
above section 2.4.2.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>A second line starting with the
three keywords </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>IND SEX POP</span></span><span
lang=EN-US>, separated by at least one space, followed by as many letters as
SNP loci, the letter giving the location of the locus as above <i
style='mso-bidi-font-style:normal'>A</i> for autosomal diploid loci, <i
style='mso-bidi-font-style:normal'>H</i> for autosomal haploid loci, <i
style='mso-bidi-font-style:normal'>X</i> for X-linked (or haplo-diploid) loci,<i
style='mso-bidi-font-style:normal'> Y</i> for Y-linked loci and <i
style='mso-bidi-font-style:normal'>M</i> for mitochondrial loci. Letters are
separated by a single space.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>As many lines as there are
genotyped individuals, with the code-name of the individual, a letter (M or F;
or 9 if the sex of the individual is unknown) indicating its sex, and a
code-name for its population. <b style='mso-bidi-font-weight:normal'><u>A
genotype is defined by the values 0, 1 or 2 which correspond to the number of
the (arbitrarily chosen) reference allele composing the genotype</u></b><u> </u>at
each SNP locus. For instance, in the case autosomal diploid SNP loci (denoted
A), we have: 2 = homozygous genotype for the reference allele (cf. 2 copies of
the reference allele), 1 = heterozygous genotype (cf. 1 copy of the reference
allele), and 0 = homozygous genotype for the variant allele, with 0 copy of the
reference allele. <u>It is worth noting that for autosomal haploid loci
(denoted H), as well as for mitochondrial loci (denoted M) and Y-linked loci
(denoted Y), the SNP genotypes will be 0 or 1, which is coherent with our logic
of defining a genotype according to the number of the reference allele
composing the genotype.<o:p></o:p></u></span></p>

<p class=MsoListParagraph style='margin-left:24.0pt;mso-add-space:auto;
text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Missing SNP genotypes are noted
9 for all type of SNP loci.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Only a subset of the SNP loci
included in the data file can be considered (selected) in the simulations and
hence in subsequent analyses, without modifying the original dataset. For
instance, one can choose to select in the corresponding panel the SNP loci 1 to
1000 of a data file including a total of say 10000 loci (see the above section
5.4.4). This allows running faster simulations and processing independent replicate
analyses of sets of 1000 SNP loci by considering loci 1 to 1000 and then 1001
to 2000, and so on, in separate analyses.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Following Hudson&#8217;s (2002)
criterion, only polymorphic SNP loci (over the entire dataset) are considered.
Monomorphic SNP loci (over the entire dataset) are automatically filtered by
the program. It is preferable, however, that the user removes himself all
monomorphic loci from his/her (observed) dataset before submitting it to DIYABC
Random Forest.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Before running any simulation,
DIYABC Random Forest provides a text file including the set of SNP loci
selected from the observed dataset (e.g. polymorphic loci 1 to 1000 with a
MAF=0.05). This file is named &#8220;UserDataFileName.bin.txt&#8221;.</span></p>

<p class=FirstParagraph><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>EXAMPLE<o:p></o:p></span></b></p>

<p class=MsoBodyText><span lang=EN-US>In our example below, the species is
diploid, has an unbalanced sex ration the sex-ratio as above (i.e.
&lt;NM=1.5NF&gt;) and two of its populations were genotyped at 23 SNP loci: 20
autosomal loci, 1 X-linked locus, 1 Y-linked locus and 1 mitochondrial locus. The
first line provides the title which includes the species sex-ratio and the MAF
(minimum allele frequency). The second line indicates: individual name in
column 1, individual sex in column 2 (</span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>M</span></span><span
lang=EN-US> for male, </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>F</span></span><span
lang=EN-US> for female, </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>9</span></span><span
lang=EN-US> or any other letter if unknown), population name in column 3 and
one column per SNP locus (letter </span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>A</span></span><span
lang=EN-US> for an autosomal locus, </span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>X</span></span><span
lang=EN-US> for an X-linked locus, </span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>Y</span></span><span
lang=EN-US> for a Y-linked locus and </span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>M</span></span><span
lang=EN-US> for a mitochondrial locus). Columns are separated by one or more
spaces. SNP genotypes are coded </span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>0</span></span><span
lang=EN-US>, </span><span class=VerbatimChar><span lang=EN-US style='font-size:
11.0pt;mso-bidi-font-size:12.0pt'>1</span></span><span lang=EN-US> or </span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>2</span></span><span lang=EN-US> (</span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>9</span></span><span
lang=EN-US> for missing data) according to the number of reference alleles at
the corresponding locus. Note that the sex has no influence on simulations for
autosomal, mitochondrial or haploid loci (any sex can be hence declared). For
individuals with an unknown sex (denoted </span><span class=VerbatimChar><span
lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>9</span></span><span
lang=EN-US>, see </span><span class=VerbatimChar><span lang=EN-US
style='font-size:11.0pt;mso-bidi-font-size:12.0pt'>IND P1_2, P1_3 and P2_15),</span></span><span
lang=EN-US> data for autosomal (as well as mitochondrial and haploid) loci will
be <span class=GramE>taken into account</span> and simulated. On the other
hand, the genotypes of X-linked and Y-linked loci for the </span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>same IND P1_2, P1_3 and P2_15</span></span><span lang=EN-US> with unknown
sex cannot be safely determined and are hence noted </span><span
class=VerbatimChar><span lang=EN-US style='font-size:11.0pt;mso-bidi-font-size:
12.0pt'>9</span></span><span lang=EN-US> for missing data (i.e. they are not
simulated).</span></p>

<p class=CaptionedFigure><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image14" o:spid="_x0000_i1029"
 type="#_x0000_t75" alt="image" style='width:348pt;height:313.2pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image147.png"
  o:title="image"/>
</v:shape><![endif]--><![if !vml]><img border=0 width=464 height=418
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image148.jpg" alt=image
v:shapes="Image14"><![endif]></span></p>

<p class=MsoNormal style='margin-top:10.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:2'><a name="_Toc62127984"><b><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'>7.1.2 </span></b></a><span class=SpellE><span
style='mso-bookmark:_Toc62127984'><b><span lang=EN-US style='font-family:"Calibri",sans-serif;
mso-ascii-theme-font:major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'>PoolSeq</span></b></span></span><span
style='mso-bookmark:_Toc62127984'><b><span lang=EN-US style='font-family:"Calibri",sans-serif;
mso-ascii-theme-font:major-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'> SNP data</span></b></span><b><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'><o:p></o:p></span></b></p>

<p class=FirstParagraph><span lang=EN-US>The data file format includes:</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>A first line (headline)
providing the sex-ratio of the analyzed species (e.g.&lt;NM=1.0NF&gt;), the required
MRC (minimum read count criterion; e.g. &lt;MRC=5&gt;), and any text that can
be used as a title. The sex ratio is noted under the form &lt;NM=<span
class=SpellE>rNF</span>&gt;, in which r is the ratio of the number of females
per male (e.g. &lt;NM=2.5NF&gt; means that the number of males is 2.5 times the
number of females; for a balanced sex ratio one should write &lt;NM=1.0NF&gt;).
The MRC is the number of sequence reads of the minor allele frequency allele
when pooling the reads <span class=SpellE>over all</span> population samples.
The specification of a MRC equal for instance to 5 (as in the present example)
will automatically select a subset of <i style='mso-bidi-font-style:normal'>m</i>
<span class=SpellE>PoolSeq</span> loci characterized by more than five reads
over all studied pools among the l loci of the observed dataset. For instance,
the first locus of the list (cf. third line of the file) will not be selected.
In agreement with this, only <i style='mso-bidi-font-style:normal'>m</i> loci
with more than five reads will be retained in a simulated dataset. We advise
using MRC values of 2, 3, 4 or 5.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Remember that (in contrast to <span
class=SpellE>IndSeq</span> SNP) only <span class=SpellE>PoolSeq</span> SNPs
located on autosomal chromosomes of diploid individuals can be considered by
the program.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>A second line provides the
haploid sample size of each population pool.<span
style='mso-spacerun:yes'>&nbsp; </span></span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The following lines correspond
to the <span class=SpellE>PoolSeq</span> SNP genotypes (given in read counts).
Each line represents a SNP and each pair of columns points to a population pool
</span><span class=tlid-translation><span lang=EN style='mso-ansi-language:
EN'>in the same order as in the second line</span></span><span lang=EN-US>. For
each SNP-pool combination, the number of read counts is indicated for the first
and second allele in the first and second column, respectively. </span></p>

<p class=MsoBodyText style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:
l0 level1 lfo12'><![if !supportLists]><span lang=EN-US style='font-family:Symbol;
mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Warning: no missing data (i.e.
&#8220;0 0&#8221; data; cf. no read count for any allele) is allowed in the <span
class=SpellE>PoolSeq</span> data file and this for any pools in the data file.</span></p>

<p class=MsoListParagraph style='margin-left:24.0pt;mso-add-space:auto;
text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>We advise to put only SNP loci
with a coverage &gt; 10 reads for each analyzed pool of the dataset in order to
ensure descent allele frequency estimation within each pool. SNP loci with an
abnormally high coverage should be also removed as those loci can be duplicated
loci.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l0 level1 lfo12'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Only a subset of the SNP loci
included in the data file can be considered (selected) in the simulations and
hence in subsequent analyses, without modifying the original dataset. For
instance, one can choose to select in the corresponding panel the SNP loci 1 to
1000 of a data file including a total of say 10000 loci (see the above section
5.4.4). This allows running faster simulations and processing independent
replicate analyses of sets of 1000 SNP loci by considering loci 1 to 1000 and
then 1001 to 2000, and so on, in separate analyses.</span></p>

<p class=FirstParagraph><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>EXAMPLE<o:p></o:p></span></b></p>

<p class=MsoBodyText><span lang=EN-US>In our example below, the species is
diploid and was genotyped using the <span class=SpellE>PoolSeq</span>
technology at several thousands of autosomal SNPs in four population pools; <u>only
the first 13 SNPs are shown here </u>for sake of concision.<span
style='mso-spacerun:yes'>&nbsp; </span>The species sex-ratio is balanced and
hence equal 1 and the MRC (minimum read count) is 5. Each population pool is
composed of 100 diploid individuals (haploid sample size = 200).<span
style='mso-spacerun:yes'>&nbsp; </span>The second line provides the haploid
sample size of each population pool. It has to start with the explicit
terminology POOL POP_NAME:HAPLOID_SAMPLE_SIZE. Here the haploid sample size is
200 for each &#8220;<span class=SpellE>POPx</span>&#8221;. Note that you can
write any sample name, e.g. MONTPELLIER, instead of <span class=SpellE>POPx</span>.<span
style='mso-spacerun:yes'>&nbsp; </span>Columns are separated by one or more
spaces. Each line represents a SNP (hence the 13 lines) and each pair of
columns points to a population pool (hence the 4 pairs of <span class=GramE>column</span>).
For each SNP-pool combination, the number of read counts is indicated for the
first and second allele in the first and second column, respectively. For
instance, we have 2 reads of the first allele and 114 reads of the second
allele for the first SNP genotyped in the POOL-POP3. Note that because the
MRC=5 several SNP of the present example data file, e.g. the first SNP for
which only two read counts are present over all pools, will not be selected
when generating simulated datasets.</span></p>

<p class=MsoBodyText><span style='mso-ansi-language:FR;mso-fareast-language:
FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Picture_x0020_4" o:spid="_x0000_i1028"
 type="#_x0000_t75" style='width:469.8pt;height:313.2pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image149.emz"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=626 height=418
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image150.gif" v:shapes="Picture_x0020_4"><![endif]></span></p>

<p class=MsoNormal style='margin-top:10.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:2'><a name="_Toc62127985"><b><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'>7.1.3 Microsatellite and DNA sequence data</span></b></a><b><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;
mso-fareast-font-family:Calibri;mso-fareast-theme-font:major-fareast;
mso-hansi-theme-font:major-latin;mso-bidi-theme-font:major-bidi;color:#4F81BD;
mso-themecolor:accent1'><o:p></o:p></span></b></p>

<p class=FirstParagraph><span lang=EN-US>The data file format is an extended
format of the file used in the classical population genetics program <span
class=SpellE>Genepop</span> (<span class=SpellE>Rousset</span> et al. 1995).
The additional features are:</span></p>

<p class=MsoListParagraphCxSpFirst style='text-indent:-18.0pt;mso-list:l21 level1 lfo29'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>In the title line appears the
sex ratio noted </span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>NM</m:r><m:r>=</m:r><m:r>rNF</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:69.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image151.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=93 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image152.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>, in which </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>r</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image021.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=8 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image022.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>is the ratio of the
number of females per male (</span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>e</m:r><m:r>.</m:r><m:r>g</m:r><m:r>.</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:20.4pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image153.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=27 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image154.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span></span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>NM</m:r><m:r>=2.5</m:r><m:r>NF</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:79.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image155.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=106 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image156.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>means that the number of
males is 2.5 times the number of females; for a balanced sex ratio one should
write </span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>NM</m:r><m:r>=1.0</m:r><m:r>NF</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:79.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image145.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=106 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image146.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>). Since the title is generally only copied, this addition should
not interfere with other programs using Genepop datafiles. <span class=GramE>Also</span>
if there is no such sex ratio addition, DIYABC will consider by default that
NM=1.0NF.</span></p>

<p class=MsoListParagraphCxSpMiddle style='text-indent:-18.0pt;mso-list:l21 level1 lfo29'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>After the locus name, there is
an indication for the category of the locus which is </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>A</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:19.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image157.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=26 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image158.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>for autosomal diploid
loci, </span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>H</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:21.6pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image159.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=29 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image160.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>for autosomal haploid
loci, </span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>X</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:20.4pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image161.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=27 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image162.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>for X-linked (or
haplo-diploid) loci, </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>Y</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:19.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image163.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=26 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image164.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>for Y-linked loci and </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>M</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:22.8pt;height:13.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image165.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=30 height=18
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image166.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>for mitochondrial loci.
If no category is noted, DIYABC will consider the locus as autosomal diploid or
autosomal haploid depending on the corresponding genotype of the first typed
individual.</span></p>

<p class=MsoListParagraphCxSpMiddle style='text-indent:-18.0pt;mso-list:l21 level1 lfo29'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Genotypes of microsatellite
loci are noted with <b style='mso-bidi-font-weight:normal'>six digit numbers</b>
if diploid (e.g.&nbsp;190188 for a heterozygous genotype with one allele 190
and one allele 188, and 200200 for a homozygous genotype with two alleles 200)
and by <b style='mso-bidi-font-weight:normal'>three digit numbers</b>
(e.g.&nbsp;190) if haploid. <u>The three digit numbers correspond to the length
in nucleotides of the corresponding PCR products</u>.</span></p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:35.7pt;mso-add-space:
auto;text-indent:-17.85pt;mso-list:l21 level1 lfo29'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Sequence locus are noted
between &lt; and &gt;. In addition, each sequence alleles/haplotypes <span
class=GramE>is</span> noted between brackets. For instance, a haploid sequence
locus will be noted &lt;[GTCTA]&gt; and a diploid sequence locus &lt;[GTCTA][GTCTT]&gt;.
Sequences may contain undetermined nucleotides which will be denoted &quot;N
&quot; or &quot; - &quot;. Note that all sequence alleles/haplotypes have to be
of same length. The length of shorter sequence allele/haplotypes needs to be
adjusted to the larger sequence allele/haplotype by adding &quot;N &quot; or
&quot; - &quot; symbols at the end of the sequences. It is worth stressing
that, at a given locus, only the portion of the sequence shared by all
individuals of the dataset will be used for computing summary statistics. We
therefore advise removing locus-individual sequence data with too many
&#8220;N&#8221; and replace them by missing data. Finally, remember that this
version of the program does not consider insertion-deletion mutations, mainly
because there does not seem to be much consensus on this topic.</span></p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:35.7pt;mso-add-space:
auto;text-indent:-17.85pt;mso-list:l21 level1 lfo29'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Missing microsatellite
genotypes are noted 000 if haploid or 000000 if diploid.</span></p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:35.7pt;mso-add-space:
auto;text-indent:-17.85pt;mso-list:l21 level1 lfo29'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Missing sequence
alleles/haplotypes are noted <span class=GramE>&lt;[</span> ]&gt; if haploid or
&lt;[ ][ ]&gt; if diploid.</span></p>

<p class=MsoListParagraphCxSpLast style='margin-left:35.7pt;mso-add-space:auto;
text-indent:-17.85pt;mso-list:l21 level1 lfo29'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>Missing data are <span
class=GramE>taken into account</span> in the following way. For each appearance
of a missing genotype in the observed dataset, the program records the
individual and the locus. When simulating datasets, the program replaces the
simulated genotype (obtained through the coalescence process algorithm) by the
missing data code at all corresponding locations. All summary statistics are
thus computed with the same missing data as for the observed dataset. <i
style='mso-bidi-font-style:normal'>WARNING</i>: data files with virtually any
amount of missing data can be analyzed by DIYABC Random Forest. <u>However, for
each locus a minimum of one genotyped individual per population is required.</u>
This is because summary statistics cannot be computed at a given locus in a
given population if only missing data are present.</span></p>

<p class=MsoBodyText><b style='mso-bidi-font-weight:normal'><span lang=EN-US>EXAMPLES<o:p></o:p></span></b></p>

<p class=MsoBodyText><u><span lang=EN-US>Example 1</span></u><span lang=EN-US>:<span
style='mso-spacerun:yes'>&nbsp; </span>the example dataset below includes two
population samples, each of 12 diploid individuals (8 females and 4 males in
the first sample and 5 females and 7 males in the second sample). As deduced
from the letter between<span style='mso-spacerun:yes'>&nbsp; </span>and<span
style='mso-spacerun:yes'>&nbsp; </span>on the locus name lines (see page 25),
these individuals have been genotyped at 3 microsatellite loci (1 autosomal </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>A</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:19.8pt;height:22.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image167.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=26 height=30
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image168.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>, 1 X-linked </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>X</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:20.4pt;height:22.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image169.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=27 height=30
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image170.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>and 1 Y-linked </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>Y</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:19.8pt;height:22.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image171.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=26 height=30
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image172.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>) and 3 DNA sequence loci (1 autosomal. 1 X-linked and 1
mitochondrial </span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>M</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:22.8pt;height:22.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image173.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=30 height=30
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image174.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>). The species sex-ratio, given in the title line, is of three males
for one female (</span><!--[if gte msEquation 12]><m:oMath><i style='mso-bidi-font-style:
 normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>NM</m:r><m:r>=3</m:r><m:r>NF</m:r><m:r>&gt;</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:70.2pt;height:22.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image175.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=94 height=30
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image176.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US>) or in other words, the number of males equals three times the
number of females.</span></p>

<p class=MsoBodyText><a name="_Toc62034937"><span style='mso-ansi-language:
FR;mso-fareast-language:FR;mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image16"
 o:spid="_x0000_i1027" type="#_x0000_t75" alt="image" style='width:387.6pt;
 height:301.8pt;visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image177.png"
  o:title="image"/>
</v:shape><![endif]--><![if !vml]><img border=0 width=517 height=402
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image178.jpg" alt=image
v:shapes="Image16"><![endif]></span></a></p>

<p class=MsoBodyText><u><span lang=EN-US>Example 2</span></u><span lang=EN-US>:
in this second example, the species is haploid. Individuals have been genotyped
at three autosomal microsatellite loci and one mitochondrial DNA sequence
locus. The species being haploid (deduced from the presence of autosomal
haploid loci), no indication of the sex-ratio appears in the title line.</span></p>

<p class=MsoNormal><span style='mso-ansi-language:FR;mso-fareast-language:FR;
mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image26" o:spid="_x0000_i1026"
 type="#_x0000_t75" alt="image" style='width:353.4pt;height:196.8pt;
 visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image179.png"
  o:title="image"/>
</v:shape><![endif]--><![if !vml]><img border=0 width=471 height=262
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image180.jpg" alt=image
v:shapes="Image26"><![endif]></span><b><span lang=EN-US style='font-size:16.0pt;
font-family:"Calibri",sans-serif;mso-ascii-theme-font:major-latin;mso-fareast-font-family:
Calibri;mso-fareast-theme-font:major-fareast;mso-hansi-theme-font:major-latin;
mso-bidi-theme-font:major-bidi;color:#4F81BD;mso-themecolor:accent1'><o:p></o:p></span></b></p>

<h2><a name="_Toc62127986"><span lang=EN-US>7.2 Training set file(s)</span></a><a
name=reference-table-files></a></h2>

<p class=FirstParagraph><span lang=EN-US>Training set files are binary files
which include two successive parts:</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l40 level1 lfo11'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The first file is a header
(named <b style='mso-bidi-font-weight:normal'>headerRF.txt</b>) which contains
information necessary to read the second part (file), such as the number of
scenarios, or the number of parameters of each scenario.</span></p>

<p class=MsoNormal style='margin-left:24.0pt;text-indent:-24.0pt;mso-list:l40 level1 lfo11'><![if !supportLists]><span
lang=EN-US style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The second file (named <span
class=SpellE><b style='mso-bidi-font-weight:normal'>reftableRF.bin</b></span>)
corresponds to the training set per se, as this file contains simulated dataset
records, each record containing on the same line the scenario number, the
parameter values and summary statistics values.</span></p>

<p class=FirstParagraph><span lang=EN-US>Each time a training set (<span
class=SpellE>reftableRF.bin</span>) is created or increased (each time the
training set &#8220;Run&#8221; button is pressed), two text files are created
in the project directory:</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l22 level1 lfo33'><![if !supportLists]><span lang=EN-US style='font-family:
Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>statobsRF.txt</span></b><span lang=EN-US>: this text file contains
the values of summary statistics for the observed data set.</span></p>

<p class=FirstParagraph style='margin-left:36.0pt;text-indent:-18.0pt;
mso-list:l22 level1 lfo33'><![if !supportLists]><span lang=EN-US
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span class=VerbatimChar><b style='mso-bidi-font-weight:
normal'><span lang=EN-US style='mso-ansi-font-size:12.0pt;font-family:"Cambria",serif;
mso-ascii-theme-font:minor-latin;mso-hansi-theme-font:minor-latin'>first_records_of_the_reference_table_X.txt</span></b></span><span
lang=EN-US> in which <span class=VerbatimChar><span style='mso-ansi-font-size:
12.0pt;font-family:"Cambria",serif;mso-ascii-theme-font:minor-latin;mso-hansi-theme-font:
minor-latin'>X</span></span> is an integer number starting at 0 and increasing
each time the training set &#8220;Run&#8221; button is pressed. This file
provides a text version of the first </span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>n</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:7.2pt;height:22.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image181.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=10 height=30
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image182.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>newly created records of
the training set reftableRF.bin (</span><!--[if gte msEquation 12]><m:oMath><i
 style='mso-bidi-font-style:normal'><span lang=EN-US style='font-family:"Cambria Math",serif'><m:r>n</m:r></span></i></m:oMath><![endif]--><![if !msEquation]><span
lang=EN-US style='font-size:12.0pt;font-family:"Calibri",sans-serif;mso-ascii-theme-font:
minor-latin;mso-fareast-font-family:Calibri;mso-fareast-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;mso-bidi-font-family:"Times New Roman";
mso-bidi-theme-font:minor-bidi;position:relative;top:2.5pt;mso-text-raise:-2.5pt;
mso-ansi-language:EN-US;mso-fareast-language:EN-US;mso-bidi-language:AR-SA'><!--[if gte vml 1]><v:shape
 id="_x0000_i1025" type="#_x0000_t75" style='width:7.2pt;height:22.8pt'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image181.png"
  o:title="" chromakey="white"/>
</v:shape><![endif]--><![if !vml]><img width=10 height=30
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image182.gif" v:shapes="_x0000_i1025"><![endif]></span><![endif]><span
lang=EN-US><span style='mso-spacerun:yes'>&nbsp;</span>being equal to the <i
style='mso-bidi-font-style:normal'>Particle loop size</i>, see section 3.7.3).</span></p>

<h2><a name="_Toc62127987"><span lang=EN-US>7.3 Output files produced by a
Random Forest analysis</span></a><a name=output-files></a></h2>

<p class=MsoNormal><span lang=EN-US>DIYABC Random Forest provides various
numerical and graphical outputs. The integration of the various graphical
outputs is managed with the ggplot2 R package (Wickham 2016), allowing user to
create and export high-quality graphics related to the analyses. Non-graphical
outputs are saved in text files. Graphs are saved as images, and user can
choose the image format among <span class=SpellE>png</span>, jpg, etc. Once a
Random Forest analysis is terminated, do not forget to click on the
&#8220;Save&#8221; button in the &#8220;Project housekeeping&#8221; panel to
implement and save a concatenated .zip file including various input and output
files (see section 7 for a description of the content of some of those files)
on your computer. The name of the implemented/saved concatenated .zip file is
the one given at the start in the &#8220;Project name&#8221; window.</span></p>

<p class=FirstParagraph><span lang=EN-US>We now describe all the files produced
by each Random Forest analysis.</span></p>

<p class=MsoBodyText><b style='mso-bidi-font-weight:normal'><span lang=EN-US>abcranger_call.log</span></b><span
lang=EN-US>: log files with votes and posterior probability of the selected (i.e.
best) scenario (i.e. model) for a model choice analysis and with the main
outputs of parameter estimation (including predictions and accuracy of
estimation indices) for a parameter estimation analysis.</span></p>

<p class=MsoBodyText><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText><u><span lang=EN-US><o:p><span style='text-decoration:
 none'>&nbsp;</span></o:p></span></u></p>

<p class=MsoBodyText><u><span lang=EN-US>Scenario choice analysis<o:p></o:p></span></u></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l5 level1 lfo30'><![if !supportLists]><span lang=EN-US style='font-family:Symbol;
mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Numerical outputs<o:p></o:p></span></i></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>modelchoice_out.settings</span></b></span><span lang=EN-US>: this
file summarizes the settings associated to the RF analysis processed for
scenario choice.</span></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>modelchoice_out.confusion</span></b></span><span lang=EN-US>: mean
misclassification error rate and confusion matrix (i.e. the contingency table
of the true and predicted classes for each example in the training set). These
indices are computed using the out-of-bag (a.k.a. out-of-bootstrap or <span
class=SpellE>oob</span>) training data as free test dataset.</span></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>modelchoice_out.predictions</span></b></span><span lang=EN-US>: votes,
prediction of the selected scenario and posterior probability of the selected
scenario (i.e. model).</span></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>modelchoice_<span class=GramE>out.ooberror</span></span></b></span><span
lang=EN-US>: Evolution of prediction power relatively to the number of trees in
the forest. Line number is the number of trees. Power is the global (prior)
error rate for scenario choice with the mean taken as point estimate (computed
using <span class=SpellE>oob</span>). See the file
modelchoice_out_graph_error_versus_ntrees.png for a graphical representation.</span></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>modelchoice_out.importance</span></b></span><span lang=EN-US>:
variables importance (sorted) which correspond to the contributions of the
statistics to the Random Forest when choosing among scenarios. The variable
importance of each statistics is computed as the mean decrease of impurity
across the trees, where the impurity measure is the Gini index. Note that the
scale is irrelevant: only the relative values matter. The variable importance
was computed for each of the summary statistics provided by DIYABC Random
Forest, plus the LDA axes (denoted LDA) if they have been added to the feature
vector. The higher the variable importance the more informative is the
statistic. Variable importance of noise variables (denoted <span class=SpellE>NOISEx</span>
and corresponding to values randomly drawn into uniform distributions bounded
between 0 and 1) are also provided. Such noise variable can be added to the
feature vector processed by RF in order to evaluate the threshold of variable
importance values below which components of the vector are not informative
anymore. More details about summary statistics including their code names can
be found in section 2.6.3. Population index(s) are also indicated at the end of
each statistic. See Collin et al. 2020 for examples. See the file
modelchoice_out_graph_variable_importance.png for a graphical representation.</span></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>modelchoice_out.lda</span></b></span><b style='mso-bidi-font-weight:
normal'><span lang=EN-US>:</span></b><span lang=EN-US><span
style='mso-spacerun:yes'>&nbsp; </span>coordinates for the projection of the
datasets from the training set and from the observed dataset on the first two
linear discriminant analysis (LDA) axes when analyzing more than two scenarios
(models) and on a single LDA axis when analyzing a pair of scenarios (or two
groups of scenarios); See the file modelchoice_out_graph_lda.png for a
graphical representation.</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l5 level1 lfo30'><![if !supportLists]><span lang=EN-US style='font-family:Symbol;
mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Graphical outputs</span></i></p>

<p class=MsoBodyText><b style='mso-bidi-font-weight:normal'><span lang=EN-US>modelchoice_out_graph_lda.png:</span></b><span
lang=EN-US> Projection of the datasets from the training set on the first two
linear discriminant analysis (LDA) axes when analyzing more than two scenarios
(models) and on a single LDA axis when analyzing a pair of scenarios (or two
groups of scenarios). The location of the observed dataset in the LDA
projection is indicated by a star symbol or a vertical line. Graphical
representation obtained from the text file <span class=SpellE>modelchoice_out.lda</span>;
see also Figure 2 in Collin et al. 2020.</span></p>

<p class=MsoBodyText><b style='mso-bidi-font-weight:normal'><span lang=EN-US>modelchoice_out_graph_error_versus_ntrees.png:</span></b><span
lang=EN-US> Evolution of prediction power relatively to the number of trees in
the forest. Graphical representation obtained from the text file <span
class=SpellE>modelchoice_<span class=GramE>out.ooberror</span></span> ; see
also Figure S1 in Collin et al. 2020.</span></p>

<p class=MsoBodyText><b style='mso-bidi-font-weight:normal'><span lang=EN-US>modelchoice_out_graph_variable_importance.png:</span></b><span
lang=EN-US> Contributions of the 50 most informative statistics (including LDA
axes if computed) to the Random Forest when choosing among scenarios. Graphical
representation obtained from the text file <span class=SpellE>modelchoice_out.importance</span>;
see also Figure 3 in Collin et al. 2020.</span></p>

<p class=MsoBodyText><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

<p class=MsoBodyText><u><span lang=EN-US>Parameter estimation analysis<o:p></o:p></span></u></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l5 level1 lfo30'><![if !supportLists]><span lang=EN-US style='font-family:Symbol;
mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Numerical outputs<o:p></o:p></span></i></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>estim_param_out.settings</span></b></span><span lang=EN-US>: this
file summarizes the settings associated to the RF analysis processed to
estimate a given parameter</span></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>estim_param_out.predictions</span></b></span><span lang=EN-US>:
expectation, median, quantiles 0.05, quantile 0.95, and variance for prediction
of parameter estimation</span></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>estim_param_<span class=GramE>out.ooberror</span></span></b></span><span
lang=EN-US>: Evolution of prediction power relatively to the number of trees in
the forest. Line number is the number of trees. Power is the global (prior)
mean squared error with the mean taken as point estimate (computed using <span
class=SpellE>oob</span>). See the file
estim_param_out_graph_error_versus_ntrees.png for a graphical representation.</span></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>estim_param_out.oobstats</span></b></span><span lang=EN-US>: this
file contains various accuracy indices of parameter estimation (computed using <span
class=SpellE>oob</span>). Both global (prior) error indices and Local
(posterior) error indices are provided (see Collin et al. 2020 for details).</span></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>estim_param_out.importance</span></b></span><span lang=EN-US>:
variables importance (sorted) which correspond to the contributions of the
statistics to the Random Forest when processing parameter estimation. The
variable importance of each statistics is computed as the mean decrease of
impurity across the trees, where the impurity measure is the residual sum of
squares. Note that the scale is irrelevant: only the relative values matter.
The variable importance was computed for each of the summary statistics
provided by DIYABC Random Forest, plus the PLS axes (denoted Comp.) if they
have been added to the feature vector. The higher the variable importance the
more informative is the statistic. Variable importance of noise variables
(denoted <span class=SpellE>NOISEx</span> and corresponding to values randomly
drawn into uniform distributions bounded between 0 and 1) are also provided.
Such noise variable can be added to the feature vector processed by RF in order
to evaluate the threshold of variable importance values below which components
of the vector are not informative anymore. More details about summary
statistics including their code names can be found in section 2.6.3. Population
index(s) are also indicated at the end of each statistic. See Collin et al.
2020 for examples. See the file estim_param_out_graph_variable_importance.png
for a graphical representation.</span></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>estim_param_out.predweights</span></b></span><span lang=EN-US>:
pairs of parameter value - weight value inferred during prediction. Such
pairwise values can be used to do density plot of parameter estimation. See the
file estim_param_out_graph_density_plot.png for a graphical representation.</span></p>

<p class=MsoBodyText><i style='mso-bidi-font-style:normal'><span lang=EN-US>If
a Partial Least Squares Regression analysis (PLS) is processed on the summary
statistics then two additional output files are provided.<o:p></o:p></span></i></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>estim_param_out.plsvar</span></b></span><span lang=EN-US>: variance
explained by number of the <i style='mso-bidi-font-style:normal'>c</i> selected
PLS components (from component 1 to <i style='mso-bidi-font-style:normal'>c,</i>
with c the number of PLS axes providing a given fraction of the maximum amount
of variance explained by all PLS axes - 95% by default-).</span></p>

<p class=MsoBodyText><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
lang=EN-US>estim_param_out.plsweights</span></b></span><span lang=EN-US>:
weight of each variable (i.e. summary statistics) for the first PLS component
(sorted by absolute value)</span></p>

<p class=MsoBodyText style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:
l5 level1 lfo30'><![if !supportLists]><span lang=EN-US style='font-family:Symbol;
mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>&middot;<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><i style='mso-bidi-font-style:normal'><span
lang=EN-US>Graphical outputs</span></i></p>

<p class=MsoBodyText><b style='mso-bidi-font-weight:normal'><span lang=EN-US>estim_param_out_graph_density_plot.png</span></b><span
lang=EN-US>: density plot graphical representation for the estimated parameter
(obtained from the text file <span class=SpellE>estim_param_out.predweights</span>).</span></p>

<p class=MsoBodyText><b style='mso-bidi-font-weight:normal'><span lang=EN-US>estim_param_out_graph_error_versus_ntrees.png:</span></b><span
lang=EN-US> Evolution of prediction power relatively to the number of trees in
the forest. Graphical representation obtained from the text file <span
class=SpellE>estim_param_out.ooberror</span>; see also Figure S1 in Collin et
al. 2020.</span></p>

<p class=MsoBodyText><b style='mso-bidi-font-weight:normal'><span lang=EN-US>estim_param_out_graph_variable_importance.png</span></b><span
lang=EN-US>: Contributions of the 50 most informative statistics (including PLS
components &#8211; denoted as Comp - if computed) to the Random Forest when
choosing among scenarios. Graphical representation obtained from the text file <span
class=SpellE>estim_param_out.ooberror</span>; see also Figure 3 in Collin et
al. 2020.</span></p>

<h2><a name="_Toc62127988"><span lang=EN-US>7.4 Other files</span></a></h2>

<p class=FirstParagraph><span lang=EN-US>The program writes several additional
files in the saved project directory:</span></p>

<p class=DefinitionTerm style='line-height:150%'><span lang=EN-US>diyabc_run_call.log:</span><span
lang=EN-US style='font-weight:normal'> this text file contains information
about the first steps of the training set production, including the reading of
the observed data file and the interpretation of the headerRF.txt file.<o:p></o:p></span></p>

<p class=DefinitionTerm style='line-height:150%'><span lang=EN-US>reftable.log:
</span><span lang=EN-US style='font-weight:normal'>this file contains information
about the progress of computations (achieved number of records)<o:p></o:p></span></p>

<p class=Definition style='line-height:150%'><b style='mso-bidi-font-weight:
normal'><span lang=EN-US>RNG_state_0000.bin</span></b><span lang=EN-US>: this
binary file contains the current state of the random generator.</span></p>

<p class=Definition style='line-height:150%'><b style='mso-bidi-font-weight:
normal'><span lang=EN-US>diyabc_seed_init_call.log: </span></b><span
lang=EN-US>this text file contains information about the initialization of the
random generator</span></p>

<p class=Definition style='line-height:150%'><b style='mso-bidi-font-weight:
normal'><span lang=EN-US>Files produced with <span class=SpellE>IndSeq</span>
analysis</span></b><span lang=EN-US> <br>
&nbsp;&nbsp;&nbsp; - &lt;<span class=SpellE>observed_dataset_name</span>&gt;.bin
and &lt;<span class=SpellE>observed_dataset_name</span>&gt;.txt: observed
dataset file in bin and txt format, respectively <br>
&nbsp;&nbsp;&nbsp; - maf.txt: file in which the minimum allele frequency value
(<span class=SpellE>maf</span>) has been recorded</span></p>

<p class=MsoNormal style='margin-top:24.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:1'><a name="_Toc62127989"><b><span
lang=EN-US style='font-size:16.0pt;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'>8. GENERATE SYNTHETIC DATAFILES</span></b></a><b><span
lang=EN-US style='font-size:16.0pt;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'> </span></b></p>

<p class=MsoNormal style='margin-top:10.0pt'><span lang=EN-US>An (auxiliary)
module named &#8220;<b style='mso-bidi-font-weight:normal'>Synthetic data file
generation</b>&#8221; (based on the DIYABC simulation engine) is available in
DIYABC Random Forest v1.0. You can use this module to generate datafile(s) for
various types of genetic markers corresponding to synthetic &#8220;ground
truth&#8221; raw data (not summarized through statistics) under a given
historical scenario and a set of parameter values <b style='mso-bidi-font-weight:
normal'>fixed by the user</b>. Such <b style='mso-bidi-font-weight:normal'>pseudo-observed
dataset</b> does not refer to any (actual) observed dataset. <b
style='mso-bidi-font-weight:normal'>The formats of the generated datafiles are
similar to those of the observed input datafiles read by DIYABC Random Forest
v1.0 (see sections 7.1.1, 7.1.2 and 7.1.3).</b></span></p>

<p class=MsoNormal><span lang=EN-US>To access to the panels allowing generating
files corresponding to full in silico datasets (also called pseudo-observed
datasets) click on the &#8220;Synthetic data file generation&#8221; button in
the middle of the main panel of the GUI or on the &#8220;Synthetic data file
generation&#8221; button in the upper-left part of the same panel.</span></p>

<p class=MsoNormal><span style='mso-ansi-language:FR;mso-fareast-language:FR;
mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Image_x0020_128" o:spid="_x0000_i1025"
 type="#_x0000_t75" style='width:502.8pt;height:282.6pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="DiyabcRandomForestUserManual21-01-2021_fichiers/image183.png"
  o:title=""/>
</v:shape><![endif]--><![if !vml]><img border=0 width=670 height=377
src="DiyabcRandomForestUserManual21-01-2021_fichiers/image184.jpg" v:shapes="Image_x0020_128"><![endif]></span></p>

<p class=MsoNormal><span lang=EN-US>The panels following such clicking are very
similar to those detailed in the sections 5.4 (<span class=SpellE>IndSeq</span>
data), 5.5 (<span class=SpellE>PoolSeq</span> data) and 5.6 (microsatellite
and/or DNA sequence data). <u>Their self-meaning content allows defining a
scenario, fixing prior values for historical (and genetic) parameters, and the
number (and type) of loci to simulate</u>. Note that those panels are completed
by two (also self-meaning) additional panels in which you indicate the sample
sizes for each population (with the proportion of males and females in each
sample), the sex ratio of the studied species, and the number of datafiles that
you want to simulate.</span></p>

<p class=MsoNormal style='margin-top:24.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:1'><a name="_Toc62127990"><b><span
lang=EN-US style='font-size:16.0pt;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'>9. USING DIYABC RANDOM FOREST ON A
COMPUTER SERVER AND EXAMPLES OF COMMAND LINES</span></b></a><a name=cluster></a></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US>The computing effort is considerably reduced
for ABC Random Forest, as the method requires a substantially smaller training
set compared to ABC methods (e.g., a few thousands of simulated datasets versus
hundreds of thousands of simulations per scenario for most ABC approaches; Blum
&amp; Fran&ccedil;ois, 2010; <span class=SpellE>Fraimout</span> et al., 2017; <span
class=SpellE>Pudlo</span> et al., 2016; <span class=SpellE>Raynal</span> et
al., 2019). Given the ever-increasing dimensionality of modern genetic data
generated using NGS technologies, this is a particularly appealing property of
this method. Your personal computer or any local server should hence represent
sufficient computer resources to run most of your DIYABC Random Forest analyses
in a descent time. However, in some cases (for instance when very large SNP
datasets have to be simulated), it might be useful to take advantage of <u>a
compute server (single node of a cluster, or even a powerful workstation),
especially to generate the training set (i.e. the file <span class=SpellE>reftableRF.bin</span>)</u>.</span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US>To do this (i.e. generate the <span
class=SpellE>reftableRF.bin</span> training set), go to the DIYABC Random
Forest main site <a href="https://diyabc.github.io/">https://diyabc.github.io/</a>
at the <i style='mso-bidi-font-style:normal'>Command line tools</i> section to
find</span><span lang=EN-US style='font-family:"Times New Roman",serif;
color:black;mso-themecolor:text1'> </span><span lang=EN-US>the <span
class=SpellE>diyabc</span> v3.0 binary (executable) file corresponding to your
Operating System</span><span lang=EN-US style='font-family:"Times New Roman",serif;
color:black;mso-themecolor:text1'>. Compilation and command line instructions
are also available </span><span lang=EN-US>at</span><span lang=EN-US
style='font-family:"Times New Roman",serif;color:black;mso-themecolor:text1'> </span><span
lang=EN-US><a href="https://github.com/diyabc/diyabc"><span style='font-family:
"Times New Roman",serif'>the <span class=SpellE>github</span> repository for <span
class=SpellE>diyabc</span> RF</span></a>. </span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US>Similarly, compilation, command line
instructions as well as the sources and various binary versions of the program <span
class=SpellE>abcranger</span>, which is the Random Forest inference engine of
the program DIYABC Random Forest, are also available at <a
href="https://github.com/diyabc/abcranger">the <span class=SpellE>github</span>
repository for <span class=SpellE>abcranger</span></a>. However, this might be
less crucial as</span><span lang=EN-US style='font-family:"Times New Roman",serif'>
Random Forest treatments with <span class=SpellE>abcranger</span> (using a
given training set) are much more rapid than the generation of the training set
with <span class=SpellE>diyabc</span> v3.0 (e.g. Collin et al. 2020).<o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><b style='mso-bidi-font-weight:normal'><span lang=EN-US>EXAMPLES
OF COMMAND LINES <o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US style='font-family:Wingdings;mso-ascii-font-family:
Cambria;mso-ascii-theme-font:minor-latin;mso-hansi-font-family:Cambria;
mso-hansi-theme-font:minor-latin;mso-char-type:symbol;mso-symbol-font-family:
Wingdings'><span style='mso-char-type:symbol;mso-symbol-font-family:Wingdings'>&agrave;</span></span><span
lang=EN-US>Warning: for more details and options see instructions in <a
href="https://diyabc.github.io/cli/"><span class=SpellE>Commande</span> line
tools - DIYABC RF</a> </span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:36.0pt;text-indent:-18.0pt;mso-list:l2 level1 lfo44'><![if !supportLists]><span
lang=EN-US style='font-family:Wingdings;mso-fareast-font-family:Wingdings;
mso-bidi-font-family:Wingdings'><span style='mso-list:Ignore'>&Oslash;<span
style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal'><span lang=EN-US>Simulation of a training
set using the (<span class=SpellE>linux</span>) executable file <span
class=SpellE>diyabc_vXXX</span> (where XXX is the version)<o:p></o:p></span></i></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US>Put in a given directory the following files:</span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:36.0pt;text-indent:-18.0pt;mso-list:l43 level1 lfo20'><![if !supportLists]><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-fareast-font-family:
Calibri'><span style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The file corresponding to your
(observed) dataset</span></p>

<p class=MsoNormalCxSpMiddle style='margin-left:36.0pt;mso-add-space:auto;
text-indent:-18.0pt;mso-list:l43 level1 lfo20'><![if !supportLists]><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-fareast-font-family:
Calibri'><span style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The header file (<u>that you
will have to name <b style='mso-bidi-font-weight:normal'>header.txt</b></u> as
a headerRF.txt file will be automatically produced) </span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:36.0pt;text-indent:-18.0pt;mso-list:l43 level1 lfo20'><![if !supportLists]><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-fareast-font-family:
Calibri'><span style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The file RNG_state_0000.bin,
which you should generate at first in a terminal window with:</span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:36.0pt'><span class=GramE><span style='font-family:"Courier New";
mso-ansi-language:FR'>./</span></span><span class=SpellE><span
style='font-family:"Courier New";mso-ansi-language:FR'>diyabc_vXXX</span></span><span
style='font-family:"Courier New";mso-ansi-language:FR'> -p ./ -n
&quot;t:8&quot;<o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:36.0pt'><span lang=EN-US>where you can replace 8 by the number of
computing cores of your server.</span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US>You can then launch the simulation in a
terminal window with:</span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span class=GramE><span lang=EN-US style='font-family:"Courier New"'>./</span></span><span
class=SpellE><span lang=EN-US style='font-family:"Courier New"'>diyabc_vXXX</span></span><span
lang=EN-US style='font-family:"Courier New"'> -p ./ -R ALL -r 12000 -g 50 -m -t
8<o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US>By doing so you will simulate of a training
set including 12000 datasets with <u>all</u> summary statistics available in <span
class=SpellE>diyabcrf</span> (plus LDA axes added to the feature vector) and
using a loop size of 50 simulated datasets and a total of 8 cores (you can
specify at most the number of cores specified by the previously generated <span
class=SpellE>RNG_state</span>_***.bin file<span class=GramE>) .</span></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:36.0pt;text-indent:-18.0pt;mso-list:l2 level1 lfo44'><![if !supportLists]><span
lang=EN-US style='font-family:Wingdings;mso-fareast-font-family:Wingdings;
mso-bidi-font-family:Wingdings'><span style='mso-list:Ignore'>&Oslash;<span
style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal'><span lang=EN-US>Random Forest analyses</span></i><span
lang=EN-US> <i style='mso-bidi-font-style:normal'>using the (<span
class=SpellE>linux</span>) executable file <span class=SpellE>abcranger_vXXX</span><o:p></o:p></i></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US>Put in a given directory the following files:</span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:36.0pt;text-indent:-18.0pt;mso-list:l43 level1 lfo20'><![if !supportLists]><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-fareast-font-family:
Calibri'><span style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The file headerRF.txt
(previously generated by <span class=SpellE>diyabc_vXXX</span>) </span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:36.0pt;text-indent:-18.0pt;mso-list:l43 level1 lfo20'><![if !supportLists]><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-fareast-font-family:
Calibri'><span style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The file <span class=SpellE>reftableRF.bin</span>
(previously generated by <span class=SpellE>diyabc_vXXX</span>) corresponding
to the training set</span><span lang=EN-US style='font-family:"Times New Roman",serif'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:36.0pt;text-indent:-18.0pt;mso-list:l43 level1 lfo20'><![if !supportLists]><span
lang=EN-US style='font-family:"Calibri",sans-serif;mso-fareast-font-family:
Calibri'><span style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span lang=EN-US>The file statobsRF.txt
(previously generated by <span class=SpellE>diyabc_vXXX</span>) which contains
the values of summary statistics for the observed data set.</span><span
lang=EN-US style='font-family:"Times New Roman",serif'><o:p></o:p></span></p>

<ul style='margin-top:0cm' type=disc>
 <li class=MsoNormal style='margin-top:9.0pt;margin-bottom:9.0pt;mso-list:l25 level1 lfo45'><span
     lang=EN-US style='font-family:"Times New Roman",serif'>Model (scenario)
     choice using RF<o:p></o:p></span></li>
</ul>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US style='font-family:"Times New Roman",serif'>You
can then launch the analysis in a terminal window with:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span class=GramE><span lang=EN-US style='font-family:"Courier New"'>./</span></span><span
class=SpellE><span lang=EN-US style='font-family:"Courier New"'>abcranger_vXXX</span></span><span
lang=EN-US style='font-family:"Courier New"'> -n 12000 -t 1000 -j 8 -o <span
class=SpellE>modelchoice_analysis</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US style='font-family:"Times New Roman",serif'>By
doing so you will process a model choice analysis on 12000 datasets of the
training set with a forest including 1000 trees, with error metrics computed
from 12000 &#8220;out-of-bag (<span class=SpellE>oob</span>) datasets, and
using a total of 8 cores. All output files names will have &#8220;<span
class=SpellE>modelchoice_analysis</span>&#8221; as prefix.<o:p></o:p></span></p>

<ul style='margin-top:0cm' type=disc>
 <li class=MsoNormal style='margin-top:9.0pt;margin-bottom:9.0pt;mso-list:l25 level1 lfo45'><span
     lang=EN-US style='font-family:"Times New Roman",serif'>Parameter
     estimation (t) using RF<o:p></o:p></span></li>
</ul>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US style='font-family:"Times New Roman",serif'>You
can launch the analysis in a terminal window with:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span class=GramE><span lang=EN-US style='font-family:"Courier New"'>./</span></span><span
class=SpellE><span lang=EN-US style='font-family:"Courier New"'>abcranger_vXXX</span></span><span
lang=EN-US style='font-family:"Courier New"'> -n 2000 -t 1000 -j 8 --parameter
t --<span class=SpellE>chosenscen</span> 1 --<span class=SpellE>plsmaxvar</span>
0.95 --noob 10000 -o <span class=SpellE>estim_param_t_with_PLS</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><span lang=EN-US style='font-family:"Times New Roman",serif'>By
doing so you will process an estimation of the parameter t from 2000 datasets
(if available in the training set) with a forest including 1000 trees, with a
number of PLS components explaining 95% of variance added to the feature
vector, with accuracy metrics computed from 10000 out-of-bag (<span
class=SpellE>oob</span>) datasets, and using a total of 8 cores. All output
files names will have &#8220;<span class=SpellE>estim_param_t_with_PLS</span>&#8221;
as prefix.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm'><i style='mso-bidi-font-style:normal'><span lang=EN-US
style='font-family:"Times New Roman",serif'>Note: The pipeline and command
lines used to generate and analyze the example datasets in Collin et al. (2020)
can be consulted at </span></i><span lang=EN-US><a
href="https://github.com/diyabc/MER_publication_materials/tree/main/MER_2021_DATASET_EXAMPLES"><span
style='font-family:"Times New Roman",serif'>https://github.com/diyabc/MER_publication_materials/tree/main/MER_2021_DATASET_EXAMPLES</span></a></span><span
lang=EN-US style='font-family:"Times New Roman",serif'> <o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:9.0pt;margin-right:0cm;margin-bottom:9.0pt;
margin-left:0cm;text-indent:36.0pt'><span lang=EN-US>It is worth stressing that
<span class=SpellE>abcranger</span> is not limited to population genetics
applications as you can use the program as a Random Forest inference engine
independently from the <span class=SpellE>diyabc</span> simulator. However, for
the moment, the binary standalone used by the DIYABC Random Forest interface
handles only outputs produced by the <span class=SpellE>diyabc</span>
simulator. A python wrapper (and example notebooks) is available at the <a
href="https://github.com/diyabc/abcranger">notebook directory in the <span
class=SpellE>github</span> repository</a></span><span lang=EN-US
style='font-family:"Times New Roman",serif;color:black;mso-themecolor:text1'> </span><span
lang=EN-US>and a R wrapper will be soon provided at the same site.</span></p>

<p class=MsoNormal style='margin-top:24.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:1'><a name="_Toc62127991"><b><span
lang=EN-US style='font-size:16.0pt;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'>10. TOY EXAMPLES</span></b></a><b><span
lang=EN-US style='font-size:16.0pt;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'><o:p></o:p></span></b></p>

<p class=Definition style='margin-top:10.0pt;line-height:150%'><span
lang=EN-US>Five toy examples to process run-tests with DIYABC-RF (i.e. for a
quick generation of a training set and rapid Random Forest analyses) can be
downloaded from <span style='color:#0066FF'>https://diyabc.github.io/toy_examples</span>/.
Each toy example illustrates crucial aspects such as the formats of different
types of observed datasets, the formalization of different types of scenarios
and the using of various types of molecular markers. Each toy example includes
two files: <b style='mso-bidi-font-weight:normal'>an observed dataset file and
a &#8220;ready to use&#8221; headerRF.txt file</b>. You can launch directly the
ABC-RF standalone by choosing the &#8220;Existing project&#8221; option and
loading these two files.</span></p>

<p class=Definition style='line-height:150%'><b style='mso-bidi-font-weight:
normal'><span lang=EN-US>TOY_EXAMPLE1_microsatellites_one_pop_bottlenecked<o:p></o:p></span></b></p>

<p class=Definition style='line-height:150%'><span lang=EN-US>The
pseudo-observed dataset includes a single population bottlenecked at time <span
class=SpellE>tbn</span> in the past with 4 samples (20 <span class=SpellE>inds</span>
per sample) taken at different times. Note that NM=1.0NF, <span class=SpellE>Npresent</span>=
200, <span class=SpellE>tbn</span>=300, <span class=SpellE>Npass</span>=20000. The
dataset includes 45 autosomal microsatellite loci and 5 mitochondrial microsatellite
loci with 2 groups of mutation parameters. </span></p>

<p class=Definition style='line-height:150%'><span lang=EN-US>The headerRF.txt
file formalizes two scenarios: one with and one without bottleneck events.</span></p>

<p class=Definition style='line-height:150%'><b style='mso-bidi-font-weight:
normal'><span lang=EN-US>TOY_EXAMPLE2_microsatellites_DNAsequence_two_pops_ancient_admixture<o:p></o:p></span></b></p>

<p class=Definition style='line-height:150%'><span lang=EN-US>The
pseudo-observed dataset includes 2 populations with ancestral admixture between
2 unsampled pops at time ta=5000 (ra=0.5) that diverged at time t2=6000 with
N1=5000 and t1=1000 for 10 autosomal microsatellites 5 autosomal DNA sequences
and 5 mitochondrial DNA sequences with 3 groups of mutation parameters. </span></p>

<p class=Definition style='line-height:150%'><span lang=EN-US>The headerRF.txt
file formalizes two scenarios: one with and one without an ancestral admixture
event.</span></p>

<p class=Definition style='line-height:150%'><b style='mso-bidi-font-weight:
normal'><span lang=EN-US>TOY_EXAMPLE3_SNP_INDSEQ_scenario3_MER_light_configuration<o:p></o:p></span></b></p>

<p class=Definition style='line-height:150%'><span lang=EN-US>This toy example
is similar to the <span class=SpellE>IndSeq</span> SNP illustration of the
paper Collin et al. (2020) based on the pseudo-observed dataset generated from
scenario 3, except that only 100 SNP (instead of 5000) are requested for
simulation in the headerRF.txt file formalizing 6 scenarios to speed-up simulations.
</span></p>

<p class=Definition style='line-height:150%'><span lang=EN-US>Warning: because
only 100 SNP are simulated, do not expect to obtain the same results as those
described in Collin et al. (2020).</span></p>

<p class=Definition style='line-height:150%'><b style='mso-bidi-font-weight:
normal'><span lang=EN-US>TOY_EXAMPLE4_SNP_POOLSEQ_scenario3_MER_light_configuration<o:p></o:p></span></b></p>

<p class=Definition style='line-height:150%'><span lang=EN-US>This toy example
is similar to the <span class=SpellE>PoolSeq</span> SNP illustration of the
paper Collin et al. (2020) based on the pseudo-observed dataset generated from
scenario 3, except that only 100 SNP (instead of 5000) are requested for
simulation in the headerRF.txt file formalizing 6 scenarios to speed-up
simulations.</span></p>

<p class=Definition style='line-height:150%'><span lang=EN-US>Warning: because
only 100 SNP are simulated, do not expect to obtain the same results as those
described in Collin et al. (2020).</span></p>

<p class=Definition style='line-height:150%'><b style='mso-bidi-font-weight:
normal'><span lang=EN-US>TOY_EXAMPLE5_SNP_INDSEQ_several_groups_of_loci_admixture<o:p></o:p></span></b></p>

<p class=Definition style='line-height:150%'><span lang=EN-US>The <span
class=SpellE>IndSeq</span> SNP pseudo-observed dataset includes 3 populations
with pop 3 admixed between pops 1 and 2 at time ta=100 (r=0.4) and pops 1 and 2
diverging at time <span class=SpellE>ts</span>=10000 (N1=20000, N2=5000,
N3=10000, N4=10000). Note that NM=0.428571NF and MAF=Hudson. The dataset
includes SNP loci of different types: A=700 loci, X=100, Y=100 and M=100.</span></p>

<p class=Definition style='line-height:150%'><span lang=EN-US>The headerRF.txt
file formalizes three scenarios: one with and two without admixture event. The
headerRF.txt file requests only A=70 loci, X=10, Y=10 and M=10 to speed-up
simulations.</span></p>

<p class=MsoNormal style='margin-top:24.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:1'><a name="_Toc62127992"><b><span
lang=EN-US style='font-size:16.0pt;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'>11. REFERENCES CITED</span></b></a><b><span
lang=EN-US style='font-size:16.0pt;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'><o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-bottom:0cm;margin-bottom:.0001pt;mso-pagination:
widow-orphan lines-together;page-break-after:avoid;mso-outline-level:1'><b><span
lang=EN-US style='font-size:16.0pt;mso-fareast-font-family:Calibri;mso-fareast-theme-font:
major-fareast;mso-bidi-font-family:Calibri;mso-bidi-theme-font:major-bidi;
color:#4F81BD;mso-themecolor:accent1'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Beaumont
M. 2010. Approximate Bayesian computation in evolution and ecology. <i
style='mso-bidi-font-style:normal'>Annual Review of Ecology, Evolution, and
Systematics</i>, 41, 379&#8211;406.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Breiman</span></span><span lang=EN-US>, L., 2001.
Random forests. <i style='mso-bidi-font-style:normal'>Machine Learning</i>,
45(1), 5&#8211;32.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Blum
MGB, Fran&ccedil;ois O. 2010.Non-linear regression models for Approximate
Bayesian Computation. <i style='mso-bidi-font-style:normal'>Statistics and
computing</i>. 20(1):63&#8211;73. https://doi.10.1007/s11222-009-9116-0</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
style='mso-ansi-language:FR'>Chang W, Cheng J, Allaire JJ, Xie Y, McPherson J
,2019. </span><span lang=EN-US>Shiny: Web Application Framework for R. R
package version 1.4.0. https://CRAN.R-project.org/package=shiny</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Chapuis</span></span><span lang=EN-US> M-P, R. <span
class=SpellE>Raynal</span>, L, <span class=SpellE>Plantamp</span>, C, <span
class=SpellE>Meynard</span>, CN, <span class=SpellE>Blondin</span>, L. Marin
J-M, Estoup A. 2020. A young age of subspecific divergence in the desert locust
<i style='mso-bidi-font-style:normal'>Schistocerca <span class=SpellE>gregaria</span></i>,
inferred by ABC Random Forest. <i style='mso-bidi-font-style:normal'>Molecular
Ecology</i> 29(23), 4542-4558. https//doi:10.1111/mec.15663. Previous version
reviewed and recommended by Peer Community in Evolutionary Biology, <span
class=SpellE>bioRxiv</span>, 671867, 10.24072/pci.evolbiol.100091</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
style='mso-ansi-language:FR'>Choisy M, P. Franck, Cornuet J-M. 2004. </span><span
lang=EN-US>Estimating admixture proportions with microsatellites: comparison of
methods based on simulated data. <i style='mso-bidi-font-style:normal'>Molecular.
Ecology</i> 13, 955-968.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Chakraborty
R, L <span class=SpellE>Jin</span>, 1993. A unified approach to study
hypervariable polymorphisms: statistical considerations of determining relatedness
and population distances. <i style='mso-bidi-font-style:normal'>EXS</i>. <i
style='mso-bidi-font-style:normal'>67</i>, 153?175.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Collin
F-D, <span class=SpellE>Raynal</span> L, Durif G, Gautier M, Vitalis R, Marin
J-M, Estoup A. 2020. DIYABC Random Forest v1.0: extending approximate Bayesian
computation with supervised machine learning to infer demographic history from
genetic polymorphisms. </span><span class=SpellE><i style='mso-bidi-font-style:
normal'><span style='mso-ansi-language:FR'>Molecular</span></i></span><i
style='mso-bidi-font-style:normal'><span style='mso-ansi-language:FR'> <span
class=SpellE>Ecology</span> <span class=SpellE>Resources</span></span></i><span
style='mso-ansi-language:FR'>. <span class=GramE>https://doi:</span>
10.22541/au.159480722.26357192<o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
style='mso-ansi-language:FR'>Cornuet J.M, <span class=SpellE>Ravign&eacute;</span>
V, Estoup A. 2010. </span><span lang=EN-US>Inference on population history and
model checking using DNA sequence and microsatellite data with the <span
class=SpellE>sofware</span> DIYABC (v1.0). <i style='mso-bidi-font-style:normal'>BMC
Bioinformatics</i> 11,401.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Cornuet</span></span><span lang=EN-US> J.M., F.
Santos, M.A.&nbsp;Beaumont, C.P. Robert, J.M. Marin, D.J. Balding, T. <span
class=SpellE>Guillemaud</span>, Estoup A. 2008. <span class=SpellE>Infering</span>
population history with DIYABC: a user-friendly approach to Approximate
Bayesian Computations. </span><span class=SpellE><i style='mso-bidi-font-style:
normal'><span style='mso-ansi-language:FR'>Bioinformatics</span></i></span><span
style='mso-ansi-language:FR'>, 24 (23), 2713-2719.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
style='mso-ansi-language:FR'>Cornuet J-M, <span class=SpellE>Pudlo</span> P, <span
class=SpellE>Veyssier</span> J, <span class=SpellE>Dehne</span>-Garcia A,
Gautier M, <span class=SpellE>Leblois</span> R, Marin J-M, Estoup A. 2014. </span><span
lang=EN-US>DIYABC v2.0: a software to make approximate Bayesian computation
inferences about population history using single nucleotide polymorphism, DNA
sequence and microsatellite data. <i style='mso-bidi-font-style:normal'>Bioinformatics</i>.
Vol. 30, no. 8, p11871189, <span class=SpellE>doi</span>:
10.1093/bioinformatics/btt763.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Jarne</span></span><span lang=EN-US> P, <span
class=SpellE>Cornuet</span> J-M. 2002. Homoplasy and mutation model at
microsatellite loci and their consequences for population genetics analysis. <i
style='mso-bidi-font-style:normal'>Molecular Ecology.</i>, 11, 1591-1604.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Estoup
A, <span class=SpellE>Raynal</span> L, <span class=SpellE>Verdu</span> P, Marin
J-M. 2018. Model choice using Approximate Bayesian Computation and Random Forests:
analyzes based on model grouping to make inferences about the genetic history
of Pygmy human populations. </span><i style='mso-bidi-font-style:normal'><span
style='mso-ansi-language:FR'>Journal de la Soci&eacute;t&eacute;
Fran&ccedil;aise de Statistiques</span></i><span style='mso-ansi-language:FR'>,
159(3), 167-190.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
style='mso-ansi-language:FR'>Excoffier L, Estoup A, Cornuet J-M. 2005. </span><span
lang=EN-US>Bayesian analysis of an admixture model with mutations and
arbitrarily linked markers. <i style='mso-bidi-font-style:normal'>Genetics</i>
169, 1727-1738.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Fraimout</span></span><span lang=EN-US> A, <span
class=SpellE>Debat</span> V, <span class=SpellE>Fellous</span> S, <span
class=SpellE>Hufbauer</span> RA, <span class=SpellE>Foucaud</span> J, <span
class=SpellE>Pudlo</span> P, Estoup A. 2017. Deciphering the routes of invasion
of Drosophila <span class=SpellE>suzukii</span> by means of ABC Random Forest. Molecular
biology and evolution, 34(4), 980&#8211;996. https:// doi.10.1093/<span
class=SpellE>molbev</span>/msx050</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Garza
JC, Williamson E. 2001. Detection of reduction in population size using data
from microsatellite DNA. <i style='mso-bidi-font-style:normal'>Molecular Ecology</i>
10,305-318.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Gautier
M, <span class=SpellE>Foucaud</span> J, <span class=SpellE>Gharbi</span> K, <span
class=SpellE>Cezard</span> T, Galan M, Loiseau A, &#8230;, Estoup A. 2013. Estimation
of population allele frequencies from next-generation sequencing data:
pool-versus individual-based genotyping. <i style='mso-bidi-font-style:normal'>Molecular
Ecology</i>, 22(14), 3766-3779. https://doi.10.1111/mec.12360 </span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Goldstein
DB, Linares AR, Cavalli-Sforza LL, Feldman MW. 1995. An evaluation of genetic
distances for use with microsatellite loci. <i style='mso-bidi-font-style:normal'>Genetics</i>
139, 463-471.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Hasegawa
M, <span class=SpellE>Kishino</span> H, Yano T. 1985. Dating the human-ape
splitting by a molecular clock of mitochondrial DNA. <i style='mso-bidi-font-style:
normal'>Journal of Molecular Evolution</i> 22:160-174.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
style='mso-ansi-language:FR'>Hivert V, Leblois R, Petit E J, Gautier M,
Vitalis, R, 2018. </span><span lang=EN-US>Measuring genetic differentiation
from pool-seq data. Genetics 210(1): 31-330.
https://doi.10.1534/genetics.118.300900</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Hudson
R, <span class=SpellE>Slatkin</span> M, Maddison WP. 1992. Estimation of levels
of gene flow <span class=SpellE>fom</span> DNA sequence data. <i
style='mso-bidi-font-style:normal'>Genetics</i>, 132, 583-589.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Hudson,
R. 2002. Generating samples under a Wright-Fisher neutral model of genetic
variation. <i style='mso-bidi-font-style:normal'>Bioinformatics</i>, 18:
337-338.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Jukes
TH, Cantor CR, 1969. Evolution of protein molecules. Pp. 21-123 in H. N. Munro,
ed.&nbsp;<i style='mso-bidi-font-style:normal'>Mammalian protein metabolism</i>.
Academic Press, New York.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Kimura
M. 1980. A simple method for estimating evolutionary rate of base substitution
through comparative studies of nucleotide sequences. <i style='mso-bidi-font-style:
normal'>Journal of Molecular Evolution</i> 16:111-120.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Leblois</span></span><span lang=EN-US> R, Gautier
M, <span class=SpellE>Rohfritsch</span> A, <span class=SpellE>Foucaud</span> J,
<span class=SpellE>Burban</span> C, Galan M,&#8230;,<span class=SpellE>Kerdelhu&eacute;</span>
C. 2018. Deciphering the demographic history of allochronic differentiation in
the pine processionary moth <span class=SpellE>Thaumetopoea</span> <span
class=SpellE>pityocampa</span>. <i style='mso-bidi-font-style:normal'>Molecular
Ecology</i>, 27(1), 264-278. doi.10.1111/mec.14411</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Meinshausen</span></span><span lang=EN-US> N 2006.
Quantile regression forests. Journal of Machine Learning Research, 7,
983&#8211;999. https://dl.acm.org/doi/10.5555/1248547.1248582</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Nei</span></span><span lang=EN-US> M 1972.
Genetic distance between populations. <i style='mso-bidi-font-style:normal'>American.
Naturalist.</i> 106:283-292</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Nei</span></span><span lang=EN-US> M 1987. <i
style='mso-bidi-font-style:normal'>Molecular Evolutionary Genetics</i>.
Columbia University Press, New York, 512 pp.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Patterson
N, <span class=SpellE>Moorjani</span> P, Luo Y, Mallick S, <span class=SpellE>Rohland</span>
N, Zhan Y, Reich D. 2012. Ancient admixture in human history. Genetics, 192
,3), 1065&#8211;1093. https://doi.org/10.1534/genetics.112.145037</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
style='mso-ansi-language:FR'>Pascual M., MP Chapuis, F Mestres, J <span
class=SpellE>Balany&aacute;</span>, RB <span class=SpellE>Huey</span>, GW <span
class=SpellE>Gilchrist</span>, <span class=SpellE>L</span> Serra, Estoup A.
2007. </span><span lang=EN-US>Introduction history of <i style='mso-bidi-font-style:
normal'>Drosophila <span class=SpellE>subobscura</span></i> in the New World: a
microsatellite based survey using ABC methods. <i style='mso-bidi-font-style:
normal'>Molecular Ecology</i>, 16, 3069-3083.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Pollock
DD, Bergman A, Feldman MW, Goldstein DB. 1998. Microsatellite behavior with
range constraints: parameter estimation and improved distances for use in
phylogenetic reconstruction. <i style='mso-bidi-font-style:normal'>Theoretical
Population Biology</i>, 53, 256-271.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Pudlo</span></span><span lang=EN-US>, P, Marin,
J.-M, Estoup, A, <span class=SpellE>Cornuet</span> J.-M, Gautier, M, Robert CP.
2016. Reliable ABC model choice via random forests. <i style='mso-bidi-font-style:
normal'>Bioinformatics</i>, 32(6), 859&#8211;866.
https://doi.10.1093/bioinformatics/btv684</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Rannala</span></span><span lang=EN-US>, B,
Mountain JL 1997. Detecting immigration by using <span class=SpellE>multilocus</span>
genotypes. <i style='mso-bidi-font-style:normal'>Pro. Nat. Acad. Sci. USA</i>
94, 9197-9201.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Raymond
M, F. <span class=SpellE>Rousset</span> F 1995. <span class=SpellE>Genepop</span>
(version 1.2), population genetics software for exact tests and ecumenicism. <i
style='mso-bidi-font-style:normal'>Journal of Heredity</i>, 86, 248-249</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span class=GramE><span lang=EN-US>Raynal</span></span></span><span
class=GramE><span lang=EN-US> .</span></span><span lang=EN-US>, Marin J-M, <span
class=SpellE>Pudlo</span> P, <span class=SpellE>Ribatet</span> M, Robert C.P,
Estoup A 2019. ABC random forests for Bayesian parameter inference. <i
style='mso-bidi-font-style:normal'>Bioinformatics</i>, 35(10), 1720&#8211;1728.
https://doi.10.1093/bioinformatics/bty867</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Schl&ouml;tterer</span></span><span lang=EN-US>
C, Tobler R, <span class=SpellE>Kofler</span> R, Nolte V. 2014. Sequencing
pools of individuals - mining genome-wide polymorphism data without big
funding.<i style='mso-bidi-font-style:normal'> Nature Reviews Genetics</i>,
15(11), 749-763. doi.10.1038/nrg3803</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span
class=SpellE><span lang=EN-US>Schrider</span></span><span lang=EN-US> DR, Kern
AD 2018. Supervised machine learning for population genetics: a new paradigm. <i
style='mso-bidi-font-style:normal'>Trends in Genetics</i>, 34(4), 301-312.
https://doi.org/10.1016/j.tig.2017.12.005</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Tajima
F 1989. Statistical method for testing the neutral <span class=SpellE>mutationhypothesis</span>
by DNA polymorphism. <i style='mso-bidi-font-style:normal'>Genetics</i> 123:
585-595</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Tamura
K, <span class=SpellE>Nei</span> M. 1993. Estimation of the number of
nucleotide substitutions in the control region of mitochondrial DNA in humans
and chimpanzees. <i style='mso-bidi-font-style:normal'>Molecular Biology and
Evolution</i> 10:512-526.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Weir
BS, Cockerham CC. 1984. Estimating F-statistics for the analysis of population
structure. <i style='mso-bidi-font-style:normal'>Evolution</i> 38: 1358-1370.</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Weir
BS, <span class=SpellE>Goudet</span> J. 2017. A unified characterization of
population structure and relatedness. Genetics, 206(4), 2085-2103.
https://doi.10.1534/genetics.116.198424</span></p>

<p class=MsoNormal style='margin-top:0cm;margin-right:0cm;margin-bottom:0cm;
margin-left:36.0pt;margin-bottom:.0001pt;text-indent:-36.0pt'><span lang=EN-US>Wickham
H. 2016. <span style='mso-spacerun:yes'>&nbsp;</span>ggplot2: Elegant Graphics
for Data Analysis. Springer-Verlag New York. ISBN 978-0-387-98141-3<a
name=web-site></a></span></p>

<p class=MsoNormal style='margin-top:24.0pt;margin-right:0cm;margin-bottom:
0cm;margin-left:0cm;margin-bottom:.0001pt;mso-pagination:widow-orphan lines-together;
page-break-after:avoid;mso-outline-level:1'><span lang=EN-US><o:p>&nbsp;</o:p></span></p>

</div>

</body>

</html>

